{
  "nbformat": 4,
  "nbformat_minor": 0,
  "metadata": {
    "colab": {
      "provenance": []
    },
    "kernelspec": {
      "name": "python3",
      "display_name": "Python 3"
    },
    "language_info": {
      "name": "python"
    },
    "widgets": {
      "application/vnd.jupyter.widget-state+json": {
        "e4568861fd194aeca79336631cf3e0b7": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "HBoxModel",
          "model_module_version": "1.5.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "HBoxModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "HBoxView",
            "box_style": "",
            "children": [
              "IPY_MODEL_5a435c6667b840059712262571ee1455",
              "IPY_MODEL_e7dc37aeb264456ca3437770c196c1cc",
              "IPY_MODEL_c5e156a53f384cc0adbf9276e4413e91"
            ],
            "layout": "IPY_MODEL_5432ef7b6dc146beac0d889dbf5acb00"
          }
        },
        "5a435c6667b840059712262571ee1455": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "HTMLModel",
          "model_module_version": "1.5.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "HTMLModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "HTMLView",
            "description": "",
            "description_tooltip": null,
            "layout": "IPY_MODEL_1c7550edd596414291dafc0a56b58f84",
            "placeholder": "​",
            "style": "IPY_MODEL_4f11907b4dab49dfab8783c495eb1dce",
            "value": "tokenizer_config.json: 100%"
          }
        },
        "e7dc37aeb264456ca3437770c196c1cc": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "FloatProgressModel",
          "model_module_version": "1.5.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "FloatProgressModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "ProgressView",
            "bar_style": "success",
            "description": "",
            "description_tooltip": null,
            "layout": "IPY_MODEL_3fe4167a91654d82985c693744c898b2",
            "max": 28,
            "min": 0,
            "orientation": "horizontal",
            "style": "IPY_MODEL_42dc65acd9db439aa93a94c28e83d96a",
            "value": 28
          }
        },
        "c5e156a53f384cc0adbf9276e4413e91": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "HTMLModel",
          "model_module_version": "1.5.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "HTMLModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "HTMLView",
            "description": "",
            "description_tooltip": null,
            "layout": "IPY_MODEL_bb8c4d702cfa488fa394f5843de212ae",
            "placeholder": "​",
            "style": "IPY_MODEL_11265d47ffdc44089b4e12d0d22d1c67",
            "value": " 28.0/28.0 [00:00&lt;00:00, 433B/s]"
          }
        },
        "5432ef7b6dc146beac0d889dbf5acb00": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "1.2.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "1c7550edd596414291dafc0a56b58f84": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "1.2.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "4f11907b4dab49dfab8783c495eb1dce": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "DescriptionStyleModel",
          "model_module_version": "1.5.0",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "DescriptionStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "description_width": ""
          }
        },
        "3fe4167a91654d82985c693744c898b2": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "1.2.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "42dc65acd9db439aa93a94c28e83d96a": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "ProgressStyleModel",
          "model_module_version": "1.5.0",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "ProgressStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "bar_color": null,
            "description_width": ""
          }
        },
        "bb8c4d702cfa488fa394f5843de212ae": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "1.2.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "11265d47ffdc44089b4e12d0d22d1c67": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "DescriptionStyleModel",
          "model_module_version": "1.5.0",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "DescriptionStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "description_width": ""
          }
        },
        "a29233c3ed754daeacfbd69e26dd45e5": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "HBoxModel",
          "model_module_version": "1.5.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "HBoxModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "HBoxView",
            "box_style": "",
            "children": [
              "IPY_MODEL_8a3c303fd47745338c08050ac2a1a2bc",
              "IPY_MODEL_5fb65a1046e44b7f901ddf26b5721f15",
              "IPY_MODEL_dd3e556132a84472aa0792e048401a05"
            ],
            "layout": "IPY_MODEL_a8f6235f61a4448dba4fbe4ce13c6fcd"
          }
        },
        "8a3c303fd47745338c08050ac2a1a2bc": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "HTMLModel",
          "model_module_version": "1.5.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "HTMLModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "HTMLView",
            "description": "",
            "description_tooltip": null,
            "layout": "IPY_MODEL_841cfd9e755a447a830dfa732ded6099",
            "placeholder": "​",
            "style": "IPY_MODEL_ddfc3ab031e64c60a058bc4159d1816c",
            "value": "config.json: 100%"
          }
        },
        "5fb65a1046e44b7f901ddf26b5721f15": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "FloatProgressModel",
          "model_module_version": "1.5.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "FloatProgressModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "ProgressView",
            "bar_style": "success",
            "description": "",
            "description_tooltip": null,
            "layout": "IPY_MODEL_40a8080b2c0f48edb2e52048a82b8671",
            "max": 385,
            "min": 0,
            "orientation": "horizontal",
            "style": "IPY_MODEL_dc9a898f9e324ca0a630d03f53595190",
            "value": 385
          }
        },
        "dd3e556132a84472aa0792e048401a05": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "HTMLModel",
          "model_module_version": "1.5.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "HTMLModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "HTMLView",
            "description": "",
            "description_tooltip": null,
            "layout": "IPY_MODEL_326074588d2d4e6c887dcde7c53fa0e3",
            "placeholder": "​",
            "style": "IPY_MODEL_eee90fe279ec445090f12165aea2a25e",
            "value": " 385/385 [00:00&lt;00:00, 6.51kB/s]"
          }
        },
        "a8f6235f61a4448dba4fbe4ce13c6fcd": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "1.2.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "841cfd9e755a447a830dfa732ded6099": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "1.2.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "ddfc3ab031e64c60a058bc4159d1816c": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "DescriptionStyleModel",
          "model_module_version": "1.5.0",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "DescriptionStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "description_width": ""
          }
        },
        "40a8080b2c0f48edb2e52048a82b8671": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "1.2.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "dc9a898f9e324ca0a630d03f53595190": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "ProgressStyleModel",
          "model_module_version": "1.5.0",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "ProgressStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "bar_color": null,
            "description_width": ""
          }
        },
        "326074588d2d4e6c887dcde7c53fa0e3": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "1.2.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "eee90fe279ec445090f12165aea2a25e": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "DescriptionStyleModel",
          "model_module_version": "1.5.0",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "DescriptionStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "description_width": ""
          }
        },
        "f044c3795a02452cba3e1ab8f06ed5cc": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "HBoxModel",
          "model_module_version": "1.5.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "HBoxModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "HBoxView",
            "box_style": "",
            "children": [
              "IPY_MODEL_5d35f048eaba4fa795948c9fecdd4c47",
              "IPY_MODEL_64257029ced24515ac59f615be545cc8",
              "IPY_MODEL_193c62a3424149a6a4bbb331f0c121e0"
            ],
            "layout": "IPY_MODEL_484c8f5f4b26451c81ffb6e7556fce96"
          }
        },
        "5d35f048eaba4fa795948c9fecdd4c47": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "HTMLModel",
          "model_module_version": "1.5.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "HTMLModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "HTMLView",
            "description": "",
            "description_tooltip": null,
            "layout": "IPY_MODEL_76a89adbe6054974bd2ef1c474621405",
            "placeholder": "​",
            "style": "IPY_MODEL_1c1438ed93904d609929ac63f00d426f",
            "value": "vocab.txt: 100%"
          }
        },
        "64257029ced24515ac59f615be545cc8": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "FloatProgressModel",
          "model_module_version": "1.5.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "FloatProgressModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "ProgressView",
            "bar_style": "success",
            "description": "",
            "description_tooltip": null,
            "layout": "IPY_MODEL_ea2fe30719c8434cb6708e351230ddec",
            "max": 226150,
            "min": 0,
            "orientation": "horizontal",
            "style": "IPY_MODEL_3b439c4c41ad462ca6e56a9428c2dde0",
            "value": 226150
          }
        },
        "193c62a3424149a6a4bbb331f0c121e0": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "HTMLModel",
          "model_module_version": "1.5.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "HTMLModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "HTMLView",
            "description": "",
            "description_tooltip": null,
            "layout": "IPY_MODEL_52faa15345de4a17a1ad9e14f02b0997",
            "placeholder": "​",
            "style": "IPY_MODEL_eb5499b9c147408b88e4de3acca1d2ca",
            "value": " 226k/226k [00:00&lt;00:00, 5.89MB/s]"
          }
        },
        "484c8f5f4b26451c81ffb6e7556fce96": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "1.2.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "76a89adbe6054974bd2ef1c474621405": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "1.2.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "1c1438ed93904d609929ac63f00d426f": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "DescriptionStyleModel",
          "model_module_version": "1.5.0",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "DescriptionStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "description_width": ""
          }
        },
        "ea2fe30719c8434cb6708e351230ddec": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "1.2.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "3b439c4c41ad462ca6e56a9428c2dde0": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "ProgressStyleModel",
          "model_module_version": "1.5.0",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "ProgressStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "bar_color": null,
            "description_width": ""
          }
        },
        "52faa15345de4a17a1ad9e14f02b0997": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "1.2.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "eb5499b9c147408b88e4de3acca1d2ca": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "DescriptionStyleModel",
          "model_module_version": "1.5.0",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "DescriptionStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "description_width": ""
          }
        },
        "e5ec05d6cd25439f8daa14e0a69b67b4": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "HBoxModel",
          "model_module_version": "1.5.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "HBoxModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "HBoxView",
            "box_style": "",
            "children": [
              "IPY_MODEL_64147eb6c2604e0a815be7797f85ac18",
              "IPY_MODEL_eb3467d485414a91b3ff855e65675f2d",
              "IPY_MODEL_349ee2cb65784281a2c03fec6a4730e0"
            ],
            "layout": "IPY_MODEL_259d4c7dde8c4ffaad619829e3cc1e41"
          }
        },
        "64147eb6c2604e0a815be7797f85ac18": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "HTMLModel",
          "model_module_version": "1.5.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "HTMLModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "HTMLView",
            "description": "",
            "description_tooltip": null,
            "layout": "IPY_MODEL_9799b290929d416ea33f3b383c1587a8",
            "placeholder": "​",
            "style": "IPY_MODEL_34fba007363a400288303b059b0b75a6",
            "value": "pytorch_model.bin: 100%"
          }
        },
        "eb3467d485414a91b3ff855e65675f2d": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "FloatProgressModel",
          "model_module_version": "1.5.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "FloatProgressModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "ProgressView",
            "bar_style": "success",
            "description": "",
            "description_tooltip": null,
            "layout": "IPY_MODEL_0ea0feb491264a689ae1c2996713031e",
            "max": 440474434,
            "min": 0,
            "orientation": "horizontal",
            "style": "IPY_MODEL_32ee7d7f0685492fa78ab09bf72f82de",
            "value": 440474434
          }
        },
        "349ee2cb65784281a2c03fec6a4730e0": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "HTMLModel",
          "model_module_version": "1.5.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "HTMLModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "HTMLView",
            "description": "",
            "description_tooltip": null,
            "layout": "IPY_MODEL_84751c6490714f2fbb60224b50a2167e",
            "placeholder": "​",
            "style": "IPY_MODEL_50c516e5e3f14590acc32ed18e98f7c0",
            "value": " 440M/440M [00:04&lt;00:00, 108MB/s]"
          }
        },
        "259d4c7dde8c4ffaad619829e3cc1e41": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "1.2.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "9799b290929d416ea33f3b383c1587a8": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "1.2.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "34fba007363a400288303b059b0b75a6": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "DescriptionStyleModel",
          "model_module_version": "1.5.0",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "DescriptionStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "description_width": ""
          }
        },
        "0ea0feb491264a689ae1c2996713031e": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "1.2.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "32ee7d7f0685492fa78ab09bf72f82de": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "ProgressStyleModel",
          "model_module_version": "1.5.0",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "ProgressStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "bar_color": null,
            "description_width": ""
          }
        },
        "84751c6490714f2fbb60224b50a2167e": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "1.2.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "50c516e5e3f14590acc32ed18e98f7c0": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "DescriptionStyleModel",
          "model_module_version": "1.5.0",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "DescriptionStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "description_width": ""
          }
        }
      }
    }
  },
  "cells": [
    {
      "cell_type": "code",
      "source": [
        "import pandas as pd\n",
        "\n",
        "# Specify the file path\n",
        "file_path = '/content/gene_associations (1).tsv.gz'\n",
        "\n",
        "# Read the gzipped TSV file into a DataFrame\n",
        "df = pd.read_csv(file_path, sep='\\t', compression='gzip')\n",
        "\n",
        "# Display the DataFrame\n",
        "df.head()"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 206
        },
        "id": "zi90pALAbi4Q",
        "outputId": "217cee17-e8d9-40a6-fcc4-2a31b5fa2b95"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "   geneId geneSymbol    DSI    DPI           PLI protein_class_name  \\\n",
              "0       1       A1BG  0.700  0.538  4.991700e-09           Receptor   \n",
              "1       2        A2M  0.529  0.769  4.522900e-11   Enzyme modulator   \n",
              "2       3      A2MP1    NaN    NaN           NaN                NaN   \n",
              "3       9       NAT1  0.536  0.846  1.929400e-14             Enzyme   \n",
              "4      10       NAT2  0.451  0.885  3.274400e-06             Enzyme   \n",
              "\n",
              "  protein_class  NofDiseases  NofPmids  \n",
              "0  DTO_05007575           27        20  \n",
              "1  DTO_05007584          147       145  \n",
              "2           NaN            1         1  \n",
              "3  DTO_05007624          133       184  \n",
              "4  DTO_05007624          311       627  "
            ],
            "text/html": [
              "\n",
              "  <div id=\"df-7abf6d92-6111-4f98-b5e2-ae1c48118adc\" class=\"colab-df-container\">\n",
              "    <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>geneId</th>\n",
              "      <th>geneSymbol</th>\n",
              "      <th>DSI</th>\n",
              "      <th>DPI</th>\n",
              "      <th>PLI</th>\n",
              "      <th>protein_class_name</th>\n",
              "      <th>protein_class</th>\n",
              "      <th>NofDiseases</th>\n",
              "      <th>NofPmids</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>1</td>\n",
              "      <td>A1BG</td>\n",
              "      <td>0.700</td>\n",
              "      <td>0.538</td>\n",
              "      <td>4.991700e-09</td>\n",
              "      <td>Receptor</td>\n",
              "      <td>DTO_05007575</td>\n",
              "      <td>27</td>\n",
              "      <td>20</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>2</td>\n",
              "      <td>A2M</td>\n",
              "      <td>0.529</td>\n",
              "      <td>0.769</td>\n",
              "      <td>4.522900e-11</td>\n",
              "      <td>Enzyme modulator</td>\n",
              "      <td>DTO_05007584</td>\n",
              "      <td>147</td>\n",
              "      <td>145</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>3</td>\n",
              "      <td>A2MP1</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>9</td>\n",
              "      <td>NAT1</td>\n",
              "      <td>0.536</td>\n",
              "      <td>0.846</td>\n",
              "      <td>1.929400e-14</td>\n",
              "      <td>Enzyme</td>\n",
              "      <td>DTO_05007624</td>\n",
              "      <td>133</td>\n",
              "      <td>184</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>10</td>\n",
              "      <td>NAT2</td>\n",
              "      <td>0.451</td>\n",
              "      <td>0.885</td>\n",
              "      <td>3.274400e-06</td>\n",
              "      <td>Enzyme</td>\n",
              "      <td>DTO_05007624</td>\n",
              "      <td>311</td>\n",
              "      <td>627</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>\n",
              "    <div class=\"colab-df-buttons\">\n",
              "\n",
              "  <div class=\"colab-df-container\">\n",
              "    <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-7abf6d92-6111-4f98-b5e2-ae1c48118adc')\"\n",
              "            title=\"Convert this dataframe to an interactive table.\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\" viewBox=\"0 -960 960 960\">\n",
              "    <path d=\"M120-120v-720h720v720H120Zm60-500h600v-160H180v160Zm220 220h160v-160H400v160Zm0 220h160v-160H400v160ZM180-400h160v-160H180v160Zm440 0h160v-160H620v160ZM180-180h160v-160H180v160Zm440 0h160v-160H620v160Z\"/>\n",
              "  </svg>\n",
              "    </button>\n",
              "\n",
              "  <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    .colab-df-buttons div {\n",
              "      margin-bottom: 4px;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "    <script>\n",
              "      const buttonEl =\n",
              "        document.querySelector('#df-7abf6d92-6111-4f98-b5e2-ae1c48118adc button.colab-df-convert');\n",
              "      buttonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "      async function convertToInteractive(key) {\n",
              "        const element = document.querySelector('#df-7abf6d92-6111-4f98-b5e2-ae1c48118adc');\n",
              "        const dataTable =\n",
              "          await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                    [key], {});\n",
              "        if (!dataTable) return;\n",
              "\n",
              "        const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "          '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "          + ' to learn more about interactive tables.';\n",
              "        element.innerHTML = '';\n",
              "        dataTable['output_type'] = 'display_data';\n",
              "        await google.colab.output.renderOutput(dataTable, element);\n",
              "        const docLink = document.createElement('div');\n",
              "        docLink.innerHTML = docLinkHtml;\n",
              "        element.appendChild(docLink);\n",
              "      }\n",
              "    </script>\n",
              "  </div>\n",
              "\n",
              "\n",
              "<div id=\"df-4fb83e62-f390-4e7d-abd9-2f5e3cad2342\">\n",
              "  <button class=\"colab-df-quickchart\" onclick=\"quickchart('df-4fb83e62-f390-4e7d-abd9-2f5e3cad2342')\"\n",
              "            title=\"Suggest charts\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "<svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "     width=\"24px\">\n",
              "    <g>\n",
              "        <path d=\"M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zM9 17H7v-7h2v7zm4 0h-2V7h2v10zm4 0h-2v-4h2v4z\"/>\n",
              "    </g>\n",
              "</svg>\n",
              "  </button>\n",
              "\n",
              "<style>\n",
              "  .colab-df-quickchart {\n",
              "      --bg-color: #E8F0FE;\n",
              "      --fill-color: #1967D2;\n",
              "      --hover-bg-color: #E2EBFA;\n",
              "      --hover-fill-color: #174EA6;\n",
              "      --disabled-fill-color: #AAA;\n",
              "      --disabled-bg-color: #DDD;\n",
              "  }\n",
              "\n",
              "  [theme=dark] .colab-df-quickchart {\n",
              "      --bg-color: #3B4455;\n",
              "      --fill-color: #D2E3FC;\n",
              "      --hover-bg-color: #434B5C;\n",
              "      --hover-fill-color: #FFFFFF;\n",
              "      --disabled-bg-color: #3B4455;\n",
              "      --disabled-fill-color: #666;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart {\n",
              "    background-color: var(--bg-color);\n",
              "    border: none;\n",
              "    border-radius: 50%;\n",
              "    cursor: pointer;\n",
              "    display: none;\n",
              "    fill: var(--fill-color);\n",
              "    height: 32px;\n",
              "    padding: 0;\n",
              "    width: 32px;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart:hover {\n",
              "    background-color: var(--hover-bg-color);\n",
              "    box-shadow: 0 1px 2px rgba(60, 64, 67, 0.3), 0 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "    fill: var(--button-hover-fill-color);\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart-complete:disabled,\n",
              "  .colab-df-quickchart-complete:disabled:hover {\n",
              "    background-color: var(--disabled-bg-color);\n",
              "    fill: var(--disabled-fill-color);\n",
              "    box-shadow: none;\n",
              "  }\n",
              "\n",
              "  .colab-df-spinner {\n",
              "    border: 2px solid var(--fill-color);\n",
              "    border-color: transparent;\n",
              "    border-bottom-color: var(--fill-color);\n",
              "    animation:\n",
              "      spin 1s steps(1) infinite;\n",
              "  }\n",
              "\n",
              "  @keyframes spin {\n",
              "    0% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "      border-left-color: var(--fill-color);\n",
              "    }\n",
              "    20% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    30% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    40% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    60% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    80% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "    90% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "  }\n",
              "</style>\n",
              "\n",
              "  <script>\n",
              "    async function quickchart(key) {\n",
              "      const quickchartButtonEl =\n",
              "        document.querySelector('#' + key + ' button');\n",
              "      quickchartButtonEl.disabled = true;  // To prevent multiple clicks.\n",
              "      quickchartButtonEl.classList.add('colab-df-spinner');\n",
              "      try {\n",
              "        const charts = await google.colab.kernel.invokeFunction(\n",
              "            'suggestCharts', [key], {});\n",
              "      } catch (error) {\n",
              "        console.error('Error during call to suggestCharts:', error);\n",
              "      }\n",
              "      quickchartButtonEl.classList.remove('colab-df-spinner');\n",
              "      quickchartButtonEl.classList.add('colab-df-quickchart-complete');\n",
              "    }\n",
              "    (() => {\n",
              "      let quickchartButtonEl =\n",
              "        document.querySelector('#df-4fb83e62-f390-4e7d-abd9-2f5e3cad2342 button');\n",
              "      quickchartButtonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "    })();\n",
              "  </script>\n",
              "</div>\n",
              "    </div>\n",
              "  </div>\n"
            ],
            "application/vnd.google.colaboratory.intrinsic+json": {
              "type": "dataframe",
              "variable_name": "df",
              "summary": "{\n  \"name\": \"df\",\n  \"rows\": 21671,\n  \"fields\": [\n    {\n      \"column\": \"geneId\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 32521109,\n        \"min\": 1,\n        \"max\": 115891964,\n        \"num_unique_values\": 21671,\n        \"samples\": [\n          23283,\n          100113407,\n          100133941\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"geneSymbol\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 21666,\n        \"samples\": [\n          \"IL15\",\n          \"DSN1\",\n          \"RCC1L\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"DSI\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 0.15989681464887168,\n        \"min\": 0.231,\n        \"max\": 1.0,\n        \"num_unique_values\": 321,\n        \"samples\": [\n          0.524,\n          0.531,\n          0.425\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"DPI\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 0.2602428833968141,\n        \"min\": 0.038,\n        \"max\": 0.962,\n        \"num_unique_values\": 25,\n        \"samples\": [\n          0.731,\n          0.077,\n          0.538\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"PLI\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 0.3947507154345081,\n        \"min\": 1.8605e-164,\n        \"max\": 1.0,\n        \"num_unique_values\": 14848,\n        \"samples\": [\n          0.00024242,\n          0.51421,\n          3.758e-07\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"protein_class_name\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 21,\n        \"samples\": [\n          \"Receptor\",\n          \"Chaperone\",\n          \"Cell adhesion\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"protein_class\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 21,\n        \"samples\": [\n          \"DTO_05007575\",\n          \"DTO_05007613\",\n          \"DTO_05007620\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"NofDiseases\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 107,\n        \"min\": 1,\n        \"max\": 2724,\n        \"num_unique_values\": 634,\n        \"samples\": [\n          369,\n          73,\n          128\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"NofPmids\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 341,\n        \"min\": 0,\n        \"max\": 20063,\n        \"num_unique_values\": 895,\n        \"samples\": [\n          427,\n          1258,\n          181\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    }\n  ]\n}"
            }
          },
          "metadata": {},
          "execution_count": 6
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!pip install requests\n",
        "!pip install beautifulsoup4\n",
        "import requests\n",
        "from bs4 import BeautifulSoup"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "2ZJotOQPtrBO",
        "outputId": "5352e213-7b67-43e4-fd12-43773eacc2f1"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Requirement already satisfied: requests in /usr/local/lib/python3.10/dist-packages (2.31.0)\n",
            "Requirement already satisfied: charset-normalizer<4,>=2 in /usr/local/lib/python3.10/dist-packages (from requests) (3.3.2)\n",
            "Requirement already satisfied: idna<4,>=2.5 in /usr/local/lib/python3.10/dist-packages (from requests) (3.7)\n",
            "Requirement already satisfied: urllib3<3,>=1.21.1 in /usr/local/lib/python3.10/dist-packages (from requests) (2.0.7)\n",
            "Requirement already satisfied: certifi>=2017.4.17 in /usr/local/lib/python3.10/dist-packages (from requests) (2024.2.2)\n",
            "Requirement already satisfied: beautifulsoup4 in /usr/local/lib/python3.10/dist-packages (4.12.3)\n",
            "Requirement already satisfied: soupsieve>1.2 in /usr/local/lib/python3.10/dist-packages (from beautifulsoup4) (2.5)\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "import pandas as pd\n",
        "project_= pd.read_csv(\"/content/genes_disease.csv\")\n",
        "df = project_.dropna(subset=['Disease'])\n",
        "df.head()"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 206
        },
        "id": "LqtyC4bIuY8N",
        "outputId": "a0c8007f-faf3-4322-bf63-1bc5a1ed10eb"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "    Unnamed: 0     Gene       Disease\n",
              "13          13    prss1     ORPHA:676\n",
              "25          25    esrrb   ORPHA:90636\n",
              "50          50   csnk2b  ORPHA:178469\n",
              "58          58  csnk2a1  ORPHA:528084\n",
              "76          76   gabrb2  ORPHA:442835"
            ],
            "text/html": [
              "\n",
              "  <div id=\"df-be822f7d-a011-4fba-8a2c-8f31fca9557a\" class=\"colab-df-container\">\n",
              "    <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>Unnamed: 0</th>\n",
              "      <th>Gene</th>\n",
              "      <th>Disease</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>13</th>\n",
              "      <td>13</td>\n",
              "      <td>prss1</td>\n",
              "      <td>ORPHA:676</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>25</th>\n",
              "      <td>25</td>\n",
              "      <td>esrrb</td>\n",
              "      <td>ORPHA:90636</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>50</th>\n",
              "      <td>50</td>\n",
              "      <td>csnk2b</td>\n",
              "      <td>ORPHA:178469</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>58</th>\n",
              "      <td>58</td>\n",
              "      <td>csnk2a1</td>\n",
              "      <td>ORPHA:528084</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>76</th>\n",
              "      <td>76</td>\n",
              "      <td>gabrb2</td>\n",
              "      <td>ORPHA:442835</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>\n",
              "    <div class=\"colab-df-buttons\">\n",
              "\n",
              "  <div class=\"colab-df-container\">\n",
              "    <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-be822f7d-a011-4fba-8a2c-8f31fca9557a')\"\n",
              "            title=\"Convert this dataframe to an interactive table.\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\" viewBox=\"0 -960 960 960\">\n",
              "    <path d=\"M120-120v-720h720v720H120Zm60-500h600v-160H180v160Zm220 220h160v-160H400v160Zm0 220h160v-160H400v160ZM180-400h160v-160H180v160Zm440 0h160v-160H620v160ZM180-180h160v-160H180v160Zm440 0h160v-160H620v160Z\"/>\n",
              "  </svg>\n",
              "    </button>\n",
              "\n",
              "  <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    .colab-df-buttons div {\n",
              "      margin-bottom: 4px;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "    <script>\n",
              "      const buttonEl =\n",
              "        document.querySelector('#df-be822f7d-a011-4fba-8a2c-8f31fca9557a button.colab-df-convert');\n",
              "      buttonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "      async function convertToInteractive(key) {\n",
              "        const element = document.querySelector('#df-be822f7d-a011-4fba-8a2c-8f31fca9557a');\n",
              "        const dataTable =\n",
              "          await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                    [key], {});\n",
              "        if (!dataTable) return;\n",
              "\n",
              "        const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "          '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "          + ' to learn more about interactive tables.';\n",
              "        element.innerHTML = '';\n",
              "        dataTable['output_type'] = 'display_data';\n",
              "        await google.colab.output.renderOutput(dataTable, element);\n",
              "        const docLink = document.createElement('div');\n",
              "        docLink.innerHTML = docLinkHtml;\n",
              "        element.appendChild(docLink);\n",
              "      }\n",
              "    </script>\n",
              "  </div>\n",
              "\n",
              "\n",
              "<div id=\"df-fd26fc91-d8e2-4f6a-b53d-8da6b8583fc3\">\n",
              "  <button class=\"colab-df-quickchart\" onclick=\"quickchart('df-fd26fc91-d8e2-4f6a-b53d-8da6b8583fc3')\"\n",
              "            title=\"Suggest charts\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "<svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "     width=\"24px\">\n",
              "    <g>\n",
              "        <path d=\"M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zM9 17H7v-7h2v7zm4 0h-2V7h2v10zm4 0h-2v-4h2v4z\"/>\n",
              "    </g>\n",
              "</svg>\n",
              "  </button>\n",
              "\n",
              "<style>\n",
              "  .colab-df-quickchart {\n",
              "      --bg-color: #E8F0FE;\n",
              "      --fill-color: #1967D2;\n",
              "      --hover-bg-color: #E2EBFA;\n",
              "      --hover-fill-color: #174EA6;\n",
              "      --disabled-fill-color: #AAA;\n",
              "      --disabled-bg-color: #DDD;\n",
              "  }\n",
              "\n",
              "  [theme=dark] .colab-df-quickchart {\n",
              "      --bg-color: #3B4455;\n",
              "      --fill-color: #D2E3FC;\n",
              "      --hover-bg-color: #434B5C;\n",
              "      --hover-fill-color: #FFFFFF;\n",
              "      --disabled-bg-color: #3B4455;\n",
              "      --disabled-fill-color: #666;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart {\n",
              "    background-color: var(--bg-color);\n",
              "    border: none;\n",
              "    border-radius: 50%;\n",
              "    cursor: pointer;\n",
              "    display: none;\n",
              "    fill: var(--fill-color);\n",
              "    height: 32px;\n",
              "    padding: 0;\n",
              "    width: 32px;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart:hover {\n",
              "    background-color: var(--hover-bg-color);\n",
              "    box-shadow: 0 1px 2px rgba(60, 64, 67, 0.3), 0 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "    fill: var(--button-hover-fill-color);\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart-complete:disabled,\n",
              "  .colab-df-quickchart-complete:disabled:hover {\n",
              "    background-color: var(--disabled-bg-color);\n",
              "    fill: var(--disabled-fill-color);\n",
              "    box-shadow: none;\n",
              "  }\n",
              "\n",
              "  .colab-df-spinner {\n",
              "    border: 2px solid var(--fill-color);\n",
              "    border-color: transparent;\n",
              "    border-bottom-color: var(--fill-color);\n",
              "    animation:\n",
              "      spin 1s steps(1) infinite;\n",
              "  }\n",
              "\n",
              "  @keyframes spin {\n",
              "    0% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "      border-left-color: var(--fill-color);\n",
              "    }\n",
              "    20% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    30% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    40% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    60% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    80% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "    90% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "  }\n",
              "</style>\n",
              "\n",
              "  <script>\n",
              "    async function quickchart(key) {\n",
              "      const quickchartButtonEl =\n",
              "        document.querySelector('#' + key + ' button');\n",
              "      quickchartButtonEl.disabled = true;  // To prevent multiple clicks.\n",
              "      quickchartButtonEl.classList.add('colab-df-spinner');\n",
              "      try {\n",
              "        const charts = await google.colab.kernel.invokeFunction(\n",
              "            'suggestCharts', [key], {});\n",
              "      } catch (error) {\n",
              "        console.error('Error during call to suggestCharts:', error);\n",
              "      }\n",
              "      quickchartButtonEl.classList.remove('colab-df-spinner');\n",
              "      quickchartButtonEl.classList.add('colab-df-quickchart-complete');\n",
              "    }\n",
              "    (() => {\n",
              "      let quickchartButtonEl =\n",
              "        document.querySelector('#df-fd26fc91-d8e2-4f6a-b53d-8da6b8583fc3 button');\n",
              "      quickchartButtonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "    })();\n",
              "  </script>\n",
              "</div>\n",
              "    </div>\n",
              "  </div>\n"
            ],
            "application/vnd.google.colaboratory.intrinsic+json": {
              "type": "dataframe",
              "variable_name": "df",
              "summary": "{\n  \"name\": \"df\",\n  \"rows\": 4007,\n  \"fields\": [\n    {\n      \"column\": \"Unnamed: 0\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 16520,\n        \"min\": 13,\n        \"max\": 57299,\n        \"num_unique_values\": 4007,\n        \"samples\": [\n          18827,\n          32880,\n          51103\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"Gene\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 2090,\n        \"samples\": [\n          \"mif\",\n          \"adamts13\",\n          \"gabra2\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"Disease\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 1504,\n        \"samples\": [\n          \"ORPHA:401920\",\n          \"ORPHA:391389\",\n          \"ORPHA:103909\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    }\n  ]\n}"
            }
          },
          "metadata": {},
          "execution_count": 127
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "Orpha:  https://www.orpha.net/en/disease/detail/746"
      ],
      "metadata": {
        "id": "X6_kErptfjIT"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "def get_info(num):\n",
        "  # URL of the page to scrape\n",
        "  url = \"https://www.orpha.net/en/disease/detail/\"+str(num)\n",
        "  # Send a GET request to the URL\n",
        "  response = requests.get(url)\n",
        "  # Parse the HTML content\n",
        "  soup = BeautifulSoup(response.text, 'html.parser')\n",
        "  # Find the title tag and extract its text\n",
        "  title_tag = soup.find('title')\n",
        "  title = title_tag.text if title_tag else None\n",
        "  ##########################################################\n",
        "  # Find the <strong> tag with the text \"Synonym(s):\"\n",
        "  synonyms=None\n",
        "  synonym_tag = soup.find('strong', string=\"Synonym(s): \")\n",
        "  # If the <strong> tag is found\n",
        "  if synonym_tag:\n",
        "        # Find the <ul> tag following the <strong> tag\n",
        "        ul_tag = synonym_tag.find_next('ul')\n",
        "        # Find all <li> tags within the <ul> tag\n",
        "        synonym_tags = ul_tag.find_all('li')\n",
        "        # Extract and return the synonyms\n",
        "        synonyms = [tag.get_text(strip=True) for tag in synonym_tags]\n",
        "        synonyms= \", \".join(synonyms)\n",
        "  ##########################################################\n",
        "  # Find the <strong> tag with the class \"d-block mb-3 service-color\" containing the text \"Disease definition\"\n",
        "  Dis_def=None\n",
        "  definition_strong = soup.find('strong', class_=\"d-block mb-3 service-color\", string=\"Disease definition\")\n",
        "  # If the <strong> tag is found\n",
        "  if definition_strong:\n",
        "        # Get the next <p> tag which contains the disease definition\n",
        "        definition_p = definition_strong.find_next('p')\n",
        "        # If <p> tag exists\n",
        "        if definition_p:\n",
        "            # Extract and return the disease definition\n",
        "            Dis_def= definition_p.get_text(strip=True)\n",
        "    # If disease definition is not found\n",
        "  ##########################################################\n",
        "  Clinical_des=None\n",
        "  # Find the <div> tag with the class \"fs-5 orpha-blue\" containing the text \"Clinical description\"\n",
        "  clinical_div = soup.find('div', class_=\"fs-5 orpha-blue\", string=\"Clinical description\")\n",
        "  # If the <div> tag is found\n",
        "  if clinical_div:\n",
        "        # Get the next <p> tag which contains the clinical description\n",
        "        clinical_p = clinical_div.find_next('p')\n",
        "        # If <p> tag exists\n",
        "        if clinical_p:\n",
        "            # Extract and return the clinical description\n",
        "            Clinical_des= clinical_p.get_text(strip=True)\n",
        "  return title, synonyms, Dis_def, Clinical_des"
      ],
      "metadata": {
        "id": "bDjNbWMndkN9"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "'''\n",
        "import requests\n",
        "from bs4 import BeautifulSoup\n",
        "def get_name(num):\n",
        "  # URL of the page to scrape\n",
        "  url = \"https://www.orpha.net/en/disease/detail/\"+str(num)\n",
        "  # Send a GET request to the URL\n",
        "  response = requests.get(url)\n",
        "  # Parse the HTML content\n",
        "  soup = BeautifulSoup(response.text, 'html.parser')\n",
        "  # Find the title tag and extract its text\n",
        "  title_tag = soup.find('title')\n",
        "  title = title_tag.text if title_tag else \"Title not found\"\n",
        "  # Print the extracted title\n",
        "  return title\n",
        "  #print(\"Title:\", title)\n",
        "def find_dis(y):\n",
        "  if y[:6] =='ORPHA:':\n",
        "      return get_name(y[6:])\n",
        "\n",
        "df['disease_name']= df.apply(lambda x: find_dis(x['Disease']), axis=1)\n",
        "df.head()\n",
        "'''\n",
        "import requests\n",
        "from bs4 import BeautifulSoup\n",
        "from concurrent.futures import ThreadPoolExecutor\n",
        "\n",
        "def get_name(num):\n",
        "    # URL of the page to scrape\n",
        "    url = \"https://www.orpha.net/en/disease/detail/\" + str(num)\n",
        "    # Send a GET request to the URL\n",
        "    response = requests.get(url)\n",
        "    # Parse the HTML content\n",
        "    soup = BeautifulSoup(response.text, 'html.parser')\n",
        "    # Find the title tag and extract its text\n",
        "    title_tag = soup.find('title')\n",
        "    title = title_tag.text if title_tag else \"Title not found\"\n",
        "    # Print the extracted title\n",
        "    return title\n",
        "\n",
        "def find_dis(y):\n",
        "    if y[:6] == 'ORPHA:':\n",
        "        return get_info(y[6:])\n",
        "    else:\n",
        "      return None, None, None, None\n",
        "# Assuming df is your pandas DataFrame\n",
        "# Modify df in-place, so we don't need to return a new DataFrame\n",
        "# We'll use a ThreadPoolExecutor to make asynchronous requests\n",
        "with ThreadPoolExecutor(max_workers=10) as executor:\n",
        "    # Apply the find_dis function to each row in parallel\n",
        "    df['disease_name'], df['Syn'], df['disease_des/def'], df['Clinicalfeatures/Clinical_des']  =zip(*list(executor.map(find_dis, df['Disease'])))\n",
        " #list(executor.map(find_dis, df['Disease']))\n",
        "\n",
        "\n",
        "\n",
        "# Print the modified DataFrame\n",
        "df.head()\n"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 938
        },
        "id": "EUc9YJuY__MZ",
        "outputId": "f80eface-7558-4133-e2c5-3b59b6377d6b"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "<ipython-input-129-627ce9703448>:51: SettingWithCopyWarning: \n",
            "A value is trying to be set on a copy of a slice from a DataFrame.\n",
            "Try using .loc[row_indexer,col_indexer] = value instead\n",
            "\n",
            "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
            "  df['disease_name'], df['Syn'], df['disease_des/def'], df['Clinicalfeatures/Clinical_des']  =zip(*list(executor.map(find_dis, df['Disease'])))\n",
            "<ipython-input-129-627ce9703448>:51: SettingWithCopyWarning: \n",
            "A value is trying to be set on a copy of a slice from a DataFrame.\n",
            "Try using .loc[row_indexer,col_indexer] = value instead\n",
            "\n",
            "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
            "  df['disease_name'], df['Syn'], df['disease_des/def'], df['Clinicalfeatures/Clinical_des']  =zip(*list(executor.map(find_dis, df['Disease'])))\n",
            "<ipython-input-129-627ce9703448>:51: SettingWithCopyWarning: \n",
            "A value is trying to be set on a copy of a slice from a DataFrame.\n",
            "Try using .loc[row_indexer,col_indexer] = value instead\n",
            "\n",
            "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
            "  df['disease_name'], df['Syn'], df['disease_des/def'], df['Clinicalfeatures/Clinical_des']  =zip(*list(executor.map(find_dis, df['Disease'])))\n",
            "<ipython-input-129-627ce9703448>:51: SettingWithCopyWarning: \n",
            "A value is trying to be set on a copy of a slice from a DataFrame.\n",
            "Try using .loc[row_indexer,col_indexer] = value instead\n",
            "\n",
            "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
            "  df['disease_name'], df['Syn'], df['disease_des/def'], df['Clinicalfeatures/Clinical_des']  =zip(*list(executor.map(find_dis, df['Disease'])))\n"
          ]
        },
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "    Unnamed: 0     Gene       Disease  \\\n",
              "13          13    prss1     ORPHA:676   \n",
              "25          25    esrrb   ORPHA:90636   \n",
              "50          50   csnk2b  ORPHA:178469   \n",
              "58          58  csnk2a1  ORPHA:528084   \n",
              "76          76   gabrb2  ORPHA:442835   \n",
              "\n",
              "                                         disease_name  \\\n",
              "13          Orphanet: Hereditary chronic pancreatitis   \n",
              "25  Orphanet: Rare autosomal recessive non-syndrom...   \n",
              "50  Orphanet: Autosomal dominant non-syndromic int...   \n",
              "58  Orphanet: Non-specific syndromic intellectual ...   \n",
              "76  Orphanet: Non-specific early-onset epileptic e...   \n",
              "\n",
              "                                                  Syn  \\\n",
              "13                                               None   \n",
              "25  Autosomal recessive isolated neurosensory deaf...   \n",
              "50                                               None   \n",
              "58                Complex neurodevelopmental disorder   \n",
              "76  Non-specific EOEE, Undetermined EOEE, Undeterm...   \n",
              "\n",
              "                                      disease_des/def  \\\n",
              "13  A rare gastroenterologic disease characterized...   \n",
              "25                                               None   \n",
              "50                                               None   \n",
              "58  A rare genetic intellectual disability charact...   \n",
              "76  A rare infantile epilepsy syndrome characteriz...   \n",
              "\n",
              "                        Clinicalfeatures/Clinical_des  \n",
              "13  Onset of HCP is typically early in life, durin...  \n",
              "25                                               None  \n",
              "50                                               None  \n",
              "58                                               None  \n",
              "76                                               None  "
            ],
            "text/html": [
              "\n",
              "  <div id=\"df-99236ce4-556c-4198-a19a-9096a79b9d9a\" class=\"colab-df-container\">\n",
              "    <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>Unnamed: 0</th>\n",
              "      <th>Gene</th>\n",
              "      <th>Disease</th>\n",
              "      <th>disease_name</th>\n",
              "      <th>Syn</th>\n",
              "      <th>disease_des/def</th>\n",
              "      <th>Clinicalfeatures/Clinical_des</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>13</th>\n",
              "      <td>13</td>\n",
              "      <td>prss1</td>\n",
              "      <td>ORPHA:676</td>\n",
              "      <td>Orphanet: Hereditary chronic pancreatitis</td>\n",
              "      <td>None</td>\n",
              "      <td>A rare gastroenterologic disease characterized...</td>\n",
              "      <td>Onset of HCP is typically early in life, durin...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>25</th>\n",
              "      <td>25</td>\n",
              "      <td>esrrb</td>\n",
              "      <td>ORPHA:90636</td>\n",
              "      <td>Orphanet: Rare autosomal recessive non-syndrom...</td>\n",
              "      <td>Autosomal recessive isolated neurosensory deaf...</td>\n",
              "      <td>None</td>\n",
              "      <td>None</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>50</th>\n",
              "      <td>50</td>\n",
              "      <td>csnk2b</td>\n",
              "      <td>ORPHA:178469</td>\n",
              "      <td>Orphanet: Autosomal dominant non-syndromic int...</td>\n",
              "      <td>None</td>\n",
              "      <td>None</td>\n",
              "      <td>None</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>58</th>\n",
              "      <td>58</td>\n",
              "      <td>csnk2a1</td>\n",
              "      <td>ORPHA:528084</td>\n",
              "      <td>Orphanet: Non-specific syndromic intellectual ...</td>\n",
              "      <td>Complex neurodevelopmental disorder</td>\n",
              "      <td>A rare genetic intellectual disability charact...</td>\n",
              "      <td>None</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>76</th>\n",
              "      <td>76</td>\n",
              "      <td>gabrb2</td>\n",
              "      <td>ORPHA:442835</td>\n",
              "      <td>Orphanet: Non-specific early-onset epileptic e...</td>\n",
              "      <td>Non-specific EOEE, Undetermined EOEE, Undeterm...</td>\n",
              "      <td>A rare infantile epilepsy syndrome characteriz...</td>\n",
              "      <td>None</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>\n",
              "    <div class=\"colab-df-buttons\">\n",
              "\n",
              "  <div class=\"colab-df-container\">\n",
              "    <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-99236ce4-556c-4198-a19a-9096a79b9d9a')\"\n",
              "            title=\"Convert this dataframe to an interactive table.\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\" viewBox=\"0 -960 960 960\">\n",
              "    <path d=\"M120-120v-720h720v720H120Zm60-500h600v-160H180v160Zm220 220h160v-160H400v160Zm0 220h160v-160H400v160ZM180-400h160v-160H180v160Zm440 0h160v-160H620v160ZM180-180h160v-160H180v160Zm440 0h160v-160H620v160Z\"/>\n",
              "  </svg>\n",
              "    </button>\n",
              "\n",
              "  <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    .colab-df-buttons div {\n",
              "      margin-bottom: 4px;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "    <script>\n",
              "      const buttonEl =\n",
              "        document.querySelector('#df-99236ce4-556c-4198-a19a-9096a79b9d9a button.colab-df-convert');\n",
              "      buttonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "      async function convertToInteractive(key) {\n",
              "        const element = document.querySelector('#df-99236ce4-556c-4198-a19a-9096a79b9d9a');\n",
              "        const dataTable =\n",
              "          await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                    [key], {});\n",
              "        if (!dataTable) return;\n",
              "\n",
              "        const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "          '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "          + ' to learn more about interactive tables.';\n",
              "        element.innerHTML = '';\n",
              "        dataTable['output_type'] = 'display_data';\n",
              "        await google.colab.output.renderOutput(dataTable, element);\n",
              "        const docLink = document.createElement('div');\n",
              "        docLink.innerHTML = docLinkHtml;\n",
              "        element.appendChild(docLink);\n",
              "      }\n",
              "    </script>\n",
              "  </div>\n",
              "\n",
              "\n",
              "<div id=\"df-68c099ac-a454-4a20-a6a3-ecd363a3612e\">\n",
              "  <button class=\"colab-df-quickchart\" onclick=\"quickchart('df-68c099ac-a454-4a20-a6a3-ecd363a3612e')\"\n",
              "            title=\"Suggest charts\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "<svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "     width=\"24px\">\n",
              "    <g>\n",
              "        <path d=\"M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zM9 17H7v-7h2v7zm4 0h-2V7h2v10zm4 0h-2v-4h2v4z\"/>\n",
              "    </g>\n",
              "</svg>\n",
              "  </button>\n",
              "\n",
              "<style>\n",
              "  .colab-df-quickchart {\n",
              "      --bg-color: #E8F0FE;\n",
              "      --fill-color: #1967D2;\n",
              "      --hover-bg-color: #E2EBFA;\n",
              "      --hover-fill-color: #174EA6;\n",
              "      --disabled-fill-color: #AAA;\n",
              "      --disabled-bg-color: #DDD;\n",
              "  }\n",
              "\n",
              "  [theme=dark] .colab-df-quickchart {\n",
              "      --bg-color: #3B4455;\n",
              "      --fill-color: #D2E3FC;\n",
              "      --hover-bg-color: #434B5C;\n",
              "      --hover-fill-color: #FFFFFF;\n",
              "      --disabled-bg-color: #3B4455;\n",
              "      --disabled-fill-color: #666;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart {\n",
              "    background-color: var(--bg-color);\n",
              "    border: none;\n",
              "    border-radius: 50%;\n",
              "    cursor: pointer;\n",
              "    display: none;\n",
              "    fill: var(--fill-color);\n",
              "    height: 32px;\n",
              "    padding: 0;\n",
              "    width: 32px;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart:hover {\n",
              "    background-color: var(--hover-bg-color);\n",
              "    box-shadow: 0 1px 2px rgba(60, 64, 67, 0.3), 0 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "    fill: var(--button-hover-fill-color);\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart-complete:disabled,\n",
              "  .colab-df-quickchart-complete:disabled:hover {\n",
              "    background-color: var(--disabled-bg-color);\n",
              "    fill: var(--disabled-fill-color);\n",
              "    box-shadow: none;\n",
              "  }\n",
              "\n",
              "  .colab-df-spinner {\n",
              "    border: 2px solid var(--fill-color);\n",
              "    border-color: transparent;\n",
              "    border-bottom-color: var(--fill-color);\n",
              "    animation:\n",
              "      spin 1s steps(1) infinite;\n",
              "  }\n",
              "\n",
              "  @keyframes spin {\n",
              "    0% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "      border-left-color: var(--fill-color);\n",
              "    }\n",
              "    20% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    30% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    40% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    60% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    80% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "    90% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "  }\n",
              "</style>\n",
              "\n",
              "  <script>\n",
              "    async function quickchart(key) {\n",
              "      const quickchartButtonEl =\n",
              "        document.querySelector('#' + key + ' button');\n",
              "      quickchartButtonEl.disabled = true;  // To prevent multiple clicks.\n",
              "      quickchartButtonEl.classList.add('colab-df-spinner');\n",
              "      try {\n",
              "        const charts = await google.colab.kernel.invokeFunction(\n",
              "            'suggestCharts', [key], {});\n",
              "      } catch (error) {\n",
              "        console.error('Error during call to suggestCharts:', error);\n",
              "      }\n",
              "      quickchartButtonEl.classList.remove('colab-df-spinner');\n",
              "      quickchartButtonEl.classList.add('colab-df-quickchart-complete');\n",
              "    }\n",
              "    (() => {\n",
              "      let quickchartButtonEl =\n",
              "        document.querySelector('#df-68c099ac-a454-4a20-a6a3-ecd363a3612e button');\n",
              "      quickchartButtonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "    })();\n",
              "  </script>\n",
              "</div>\n",
              "    </div>\n",
              "  </div>\n"
            ],
            "application/vnd.google.colaboratory.intrinsic+json": {
              "type": "dataframe",
              "variable_name": "df",
              "summary": "{\n  \"name\": \"df\",\n  \"rows\": 4007,\n  \"fields\": [\n    {\n      \"column\": \"Unnamed: 0\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 16520,\n        \"min\": 13,\n        \"max\": 57299,\n        \"num_unique_values\": 4007,\n        \"samples\": [\n          18827,\n          32880,\n          51103\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"Gene\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 2090,\n        \"samples\": [\n          \"mif\",\n          \"adamts13\",\n          \"gabra2\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"Disease\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 1504,\n        \"samples\": [\n          \"ORPHA:401920\",\n          \"ORPHA:391389\",\n          \"ORPHA:103909\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"disease_name\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 1194,\n        \"samples\": [\n          \"Orphanet: Lethal brain and heart developmental defects\",\n          \"Orphanet: Early-onset seizures-distal limb anomalies-facial dysmorphism-global developmental delay syndrome\",\n          \"Orphanet: Hypocalcemic vitamin D-resistant rickets\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"Syn\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 942,\n        \"samples\": [\n          \"CABV syndrome, CANVAS, Cerebellar ataxia with bilateral vestibulopathy syndrome\",\n          \"3-methylglutaconic aciduria-epilepsy-spasticity-severe intellectual disability syndrome, MGA9\",\n          \"Nanophthalmia\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"disease_des/def\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 1039,\n        \"samples\": [\n          \"A rare ectodermal dysplasia syndrome characterized by the association of lower eyelid ectropion, upper eyelid distichiasis, euryblepharon, bilateral cleft lip and palate, and conical teeth.\",\n          \"A rare genetic disorder caused by deletions in the long arm of chromosome 11 (11q) and mainly characterized by craniofacial dysmorphism, congenital heart disease, intellectual disability, Paris Trousseau bleeding disorder, structural kidney defects and immunodeficiency.\",\n          \"A rare, autosomal recessive autoinflammatory disorder characterized by early-onset erythematous popular/nodular skin eruptions, recurrent fever, possible joint contractures, lipodystrophy, erythematous inflammatory skin changes, joint and muscle involvement (joint contractures, arthralgia, muscle weakness), and hepatosplenomegaly.\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"Clinicalfeatures/Clinical_des\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 475,\n        \"samples\": [\n          \"Clinical features develop perinatally and during infancy. Patients tend to be born preterm. Hyperkeratosis is usually present at birth and progresses to a generalized ichthyosis, particularly prominent on flexural areas, the nape of the neck, the trunk and the extremities. Pruritus is a prominent feature. Erythematous dermatitis is often present at birth, and then tends to fade with increasing age. Neurological signs appear during the first two years of life and consist of delay in reaching motor milestones due to spastic diplegia or, much less commonly, spastic tetraplegia. Approximately one-half of patients are non-ambulatory. Seizures occur in about 40% of cases. Intellectual deficit varies from mild to severe, although rare patients with normal intellect have been reported. Delayed speech and dysarthria are common. Ophthalmologic involvement is often present and is characterized by retinal crystalline inclusions (so-called glistening white dots) surrounding the fovea. Photophobia and myopia are common.\",\n          \"Most patients display a severe phenotype that presents in infancy, usually from the neonatal period up until 12 months of age. The disease manifests as hypoketotic hypoglycemia, metabolic acidosis, hypotonia, liver involvement with hepatic encephalopathy, cardiomyopathy and arrhythmias. Clinical presentation is frequently preceded by fasting and/or intercurrent illness and often presents with hypoketotic hypoglycemia. Chronic peripheral neuropathy and pigmentary retinopathy develop over time in many surviving patients. Rarer presentations of LCHADD are sudden cardiac arrest or sudden infant death. HELLP syndrome (see this term) often occurs in pregnant women carrying a fetus affected with LCHADD.\",\n          \"The median age at diagnosis is 60-65 years. The disease is extremely rare in children. Follicular lymphoma is located primarily in the lymph nodes, but can also involve the spleen, bone marrow, peripheral blood and Waldeyer's ring. The skin and central nervous system are affected in rare cases. Symptoms appear at an advanced stage of the disease and can include fever, night sweats and weight loss. At diagnosis, patients usually present with adenopathy and, in 50% of cases, splenomegaly.\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    }\n  ]\n}"
            }
          },
          "metadata": {},
          "execution_count": 129
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "df2= df.dropna(subset=['disease_name'])\n",
        "df_unique_1 = df2.drop_duplicates(subset=['disease_name'])\n",
        "df_unique_1"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 961
        },
        "id": "m1lkwoqZCvuU",
        "outputId": "344be54b-849e-4b81-8b13-d1a88820090c"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "       Unnamed: 0     Gene       Disease  \\\n",
              "13             13    prss1     ORPHA:676   \n",
              "25             25    esrrb   ORPHA:90636   \n",
              "50             50   csnk2b  ORPHA:178469   \n",
              "58             58  csnk2a1  ORPHA:528084   \n",
              "76             76   gabrb2  ORPHA:442835   \n",
              "...           ...      ...           ...   \n",
              "56605       56605  cyp51a1  ORPHA:521432   \n",
              "56616       56616     g6pd  ORPHA:466026   \n",
              "56682       56682     dhfr  ORPHA:319651   \n",
              "56970       56970      gp6   ORPHA:98885   \n",
              "57224       57224     fzd5   ORPHA:98942   \n",
              "\n",
              "                                            disease_name  \\\n",
              "13             Orphanet: Hereditary chronic pancreatitis   \n",
              "25     Orphanet: Rare autosomal recessive non-syndrom...   \n",
              "50     Orphanet: Autosomal dominant non-syndromic int...   \n",
              "58     Orphanet: Non-specific syndromic intellectual ...   \n",
              "76     Orphanet: Non-specific early-onset epileptic e...   \n",
              "...                                                  ...   \n",
              "56605  Orphanet: Congenital cataract-severe neonatal ...   \n",
              "56616  Orphanet: Class I glucose-6-phosphate dehydrog...   \n",
              "56682  Orphanet: Constitutional megaloblastic anemia ...   \n",
              "56970  Orphanet: Bleeding diathesis due to glycoprote...   \n",
              "57224           Orphanet: Coloboma of choroid and retina   \n",
              "\n",
              "                                                     Syn  \\\n",
              "13                                                  None   \n",
              "25     Autosomal recessive isolated neurosensory deaf...   \n",
              "50                                                  None   \n",
              "58                   Complex neurodevelopmental disorder   \n",
              "76     Non-specific EOEE, Undetermined EOEE, Undeterm...   \n",
              "...                                                  ...   \n",
              "56605                                               None   \n",
              "56616  Class I G6PD deficiency, Severe hemolytic anem...   \n",
              "56682  DHFR deficiency, Dihydrofolate reductase defic...   \n",
              "56970                                               None   \n",
              "57224                                               None   \n",
              "\n",
              "                                         disease_des/def  \\\n",
              "13     A rare gastroenterologic disease characterized...   \n",
              "25                                                  None   \n",
              "50                                                  None   \n",
              "58     A rare genetic intellectual disability charact...   \n",
              "76     A rare infantile epilepsy syndrome characteriz...   \n",
              "...                                                  ...   \n",
              "56605  A rare genetic disease characterized by congen...   \n",
              "56616  A rare constitutional hemolytic anemia due to ...   \n",
              "56682                                               None   \n",
              "56970                                               None   \n",
              "57224  Coloboma of choroid and retina is a rare, gene...   \n",
              "\n",
              "                           Clinicalfeatures/Clinical_des  \n",
              "13     Onset of HCP is typically early in life, durin...  \n",
              "25                                                  None  \n",
              "50                                                  None  \n",
              "58                                                  None  \n",
              "76                                                  None  \n",
              "...                                                  ...  \n",
              "56605                                               None  \n",
              "56616                                               None  \n",
              "56682                                               None  \n",
              "56970                                               None  \n",
              "57224                                               None  \n",
              "\n",
              "[1194 rows x 7 columns]"
            ],
            "text/html": [
              "\n",
              "  <div id=\"df-b6510b7f-f556-47fc-9276-adb434008e63\" class=\"colab-df-container\">\n",
              "    <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>Unnamed: 0</th>\n",
              "      <th>Gene</th>\n",
              "      <th>Disease</th>\n",
              "      <th>disease_name</th>\n",
              "      <th>Syn</th>\n",
              "      <th>disease_des/def</th>\n",
              "      <th>Clinicalfeatures/Clinical_des</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>13</th>\n",
              "      <td>13</td>\n",
              "      <td>prss1</td>\n",
              "      <td>ORPHA:676</td>\n",
              "      <td>Orphanet: Hereditary chronic pancreatitis</td>\n",
              "      <td>None</td>\n",
              "      <td>A rare gastroenterologic disease characterized...</td>\n",
              "      <td>Onset of HCP is typically early in life, durin...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>25</th>\n",
              "      <td>25</td>\n",
              "      <td>esrrb</td>\n",
              "      <td>ORPHA:90636</td>\n",
              "      <td>Orphanet: Rare autosomal recessive non-syndrom...</td>\n",
              "      <td>Autosomal recessive isolated neurosensory deaf...</td>\n",
              "      <td>None</td>\n",
              "      <td>None</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>50</th>\n",
              "      <td>50</td>\n",
              "      <td>csnk2b</td>\n",
              "      <td>ORPHA:178469</td>\n",
              "      <td>Orphanet: Autosomal dominant non-syndromic int...</td>\n",
              "      <td>None</td>\n",
              "      <td>None</td>\n",
              "      <td>None</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>58</th>\n",
              "      <td>58</td>\n",
              "      <td>csnk2a1</td>\n",
              "      <td>ORPHA:528084</td>\n",
              "      <td>Orphanet: Non-specific syndromic intellectual ...</td>\n",
              "      <td>Complex neurodevelopmental disorder</td>\n",
              "      <td>A rare genetic intellectual disability charact...</td>\n",
              "      <td>None</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>76</th>\n",
              "      <td>76</td>\n",
              "      <td>gabrb2</td>\n",
              "      <td>ORPHA:442835</td>\n",
              "      <td>Orphanet: Non-specific early-onset epileptic e...</td>\n",
              "      <td>Non-specific EOEE, Undetermined EOEE, Undeterm...</td>\n",
              "      <td>A rare infantile epilepsy syndrome characteriz...</td>\n",
              "      <td>None</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>...</th>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>56605</th>\n",
              "      <td>56605</td>\n",
              "      <td>cyp51a1</td>\n",
              "      <td>ORPHA:521432</td>\n",
              "      <td>Orphanet: Congenital cataract-severe neonatal ...</td>\n",
              "      <td>None</td>\n",
              "      <td>A rare genetic disease characterized by congen...</td>\n",
              "      <td>None</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>56616</th>\n",
              "      <td>56616</td>\n",
              "      <td>g6pd</td>\n",
              "      <td>ORPHA:466026</td>\n",
              "      <td>Orphanet: Class I glucose-6-phosphate dehydrog...</td>\n",
              "      <td>Class I G6PD deficiency, Severe hemolytic anem...</td>\n",
              "      <td>A rare constitutional hemolytic anemia due to ...</td>\n",
              "      <td>None</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>56682</th>\n",
              "      <td>56682</td>\n",
              "      <td>dhfr</td>\n",
              "      <td>ORPHA:319651</td>\n",
              "      <td>Orphanet: Constitutional megaloblastic anemia ...</td>\n",
              "      <td>DHFR deficiency, Dihydrofolate reductase defic...</td>\n",
              "      <td>None</td>\n",
              "      <td>None</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>56970</th>\n",
              "      <td>56970</td>\n",
              "      <td>gp6</td>\n",
              "      <td>ORPHA:98885</td>\n",
              "      <td>Orphanet: Bleeding diathesis due to glycoprote...</td>\n",
              "      <td>None</td>\n",
              "      <td>None</td>\n",
              "      <td>None</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>57224</th>\n",
              "      <td>57224</td>\n",
              "      <td>fzd5</td>\n",
              "      <td>ORPHA:98942</td>\n",
              "      <td>Orphanet: Coloboma of choroid and retina</td>\n",
              "      <td>None</td>\n",
              "      <td>Coloboma of choroid and retina is a rare, gene...</td>\n",
              "      <td>None</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "<p>1194 rows × 7 columns</p>\n",
              "</div>\n",
              "    <div class=\"colab-df-buttons\">\n",
              "\n",
              "  <div class=\"colab-df-container\">\n",
              "    <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-b6510b7f-f556-47fc-9276-adb434008e63')\"\n",
              "            title=\"Convert this dataframe to an interactive table.\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\" viewBox=\"0 -960 960 960\">\n",
              "    <path d=\"M120-120v-720h720v720H120Zm60-500h600v-160H180v160Zm220 220h160v-160H400v160Zm0 220h160v-160H400v160ZM180-400h160v-160H180v160Zm440 0h160v-160H620v160ZM180-180h160v-160H180v160Zm440 0h160v-160H620v160Z\"/>\n",
              "  </svg>\n",
              "    </button>\n",
              "\n",
              "  <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    .colab-df-buttons div {\n",
              "      margin-bottom: 4px;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "    <script>\n",
              "      const buttonEl =\n",
              "        document.querySelector('#df-b6510b7f-f556-47fc-9276-adb434008e63 button.colab-df-convert');\n",
              "      buttonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "      async function convertToInteractive(key) {\n",
              "        const element = document.querySelector('#df-b6510b7f-f556-47fc-9276-adb434008e63');\n",
              "        const dataTable =\n",
              "          await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                    [key], {});\n",
              "        if (!dataTable) return;\n",
              "\n",
              "        const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "          '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "          + ' to learn more about interactive tables.';\n",
              "        element.innerHTML = '';\n",
              "        dataTable['output_type'] = 'display_data';\n",
              "        await google.colab.output.renderOutput(dataTable, element);\n",
              "        const docLink = document.createElement('div');\n",
              "        docLink.innerHTML = docLinkHtml;\n",
              "        element.appendChild(docLink);\n",
              "      }\n",
              "    </script>\n",
              "  </div>\n",
              "\n",
              "\n",
              "<div id=\"df-b8afa07d-2d04-42a9-8fb1-91e7b9c01939\">\n",
              "  <button class=\"colab-df-quickchart\" onclick=\"quickchart('df-b8afa07d-2d04-42a9-8fb1-91e7b9c01939')\"\n",
              "            title=\"Suggest charts\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "<svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "     width=\"24px\">\n",
              "    <g>\n",
              "        <path d=\"M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zM9 17H7v-7h2v7zm4 0h-2V7h2v10zm4 0h-2v-4h2v4z\"/>\n",
              "    </g>\n",
              "</svg>\n",
              "  </button>\n",
              "\n",
              "<style>\n",
              "  .colab-df-quickchart {\n",
              "      --bg-color: #E8F0FE;\n",
              "      --fill-color: #1967D2;\n",
              "      --hover-bg-color: #E2EBFA;\n",
              "      --hover-fill-color: #174EA6;\n",
              "      --disabled-fill-color: #AAA;\n",
              "      --disabled-bg-color: #DDD;\n",
              "  }\n",
              "\n",
              "  [theme=dark] .colab-df-quickchart {\n",
              "      --bg-color: #3B4455;\n",
              "      --fill-color: #D2E3FC;\n",
              "      --hover-bg-color: #434B5C;\n",
              "      --hover-fill-color: #FFFFFF;\n",
              "      --disabled-bg-color: #3B4455;\n",
              "      --disabled-fill-color: #666;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart {\n",
              "    background-color: var(--bg-color);\n",
              "    border: none;\n",
              "    border-radius: 50%;\n",
              "    cursor: pointer;\n",
              "    display: none;\n",
              "    fill: var(--fill-color);\n",
              "    height: 32px;\n",
              "    padding: 0;\n",
              "    width: 32px;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart:hover {\n",
              "    background-color: var(--hover-bg-color);\n",
              "    box-shadow: 0 1px 2px rgba(60, 64, 67, 0.3), 0 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "    fill: var(--button-hover-fill-color);\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart-complete:disabled,\n",
              "  .colab-df-quickchart-complete:disabled:hover {\n",
              "    background-color: var(--disabled-bg-color);\n",
              "    fill: var(--disabled-fill-color);\n",
              "    box-shadow: none;\n",
              "  }\n",
              "\n",
              "  .colab-df-spinner {\n",
              "    border: 2px solid var(--fill-color);\n",
              "    border-color: transparent;\n",
              "    border-bottom-color: var(--fill-color);\n",
              "    animation:\n",
              "      spin 1s steps(1) infinite;\n",
              "  }\n",
              "\n",
              "  @keyframes spin {\n",
              "    0% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "      border-left-color: var(--fill-color);\n",
              "    }\n",
              "    20% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    30% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    40% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    60% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    80% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "    90% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "  }\n",
              "</style>\n",
              "\n",
              "  <script>\n",
              "    async function quickchart(key) {\n",
              "      const quickchartButtonEl =\n",
              "        document.querySelector('#' + key + ' button');\n",
              "      quickchartButtonEl.disabled = true;  // To prevent multiple clicks.\n",
              "      quickchartButtonEl.classList.add('colab-df-spinner');\n",
              "      try {\n",
              "        const charts = await google.colab.kernel.invokeFunction(\n",
              "            'suggestCharts', [key], {});\n",
              "      } catch (error) {\n",
              "        console.error('Error during call to suggestCharts:', error);\n",
              "      }\n",
              "      quickchartButtonEl.classList.remove('colab-df-spinner');\n",
              "      quickchartButtonEl.classList.add('colab-df-quickchart-complete');\n",
              "    }\n",
              "    (() => {\n",
              "      let quickchartButtonEl =\n",
              "        document.querySelector('#df-b8afa07d-2d04-42a9-8fb1-91e7b9c01939 button');\n",
              "      quickchartButtonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "    })();\n",
              "  </script>\n",
              "</div>\n",
              "    </div>\n",
              "  </div>\n"
            ],
            "application/vnd.google.colaboratory.intrinsic+json": {
              "type": "dataframe",
              "variable_name": "df_unique_1",
              "summary": "{\n  \"name\": \"df_unique_1\",\n  \"rows\": 1194,\n  \"fields\": [\n    {\n      \"column\": \"Unnamed: 0\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 16381,\n        \"min\": 13,\n        \"max\": 57224,\n        \"num_unique_values\": 1194,\n        \"samples\": [\n          16583,\n          20525,\n          22717\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"Gene\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 1194,\n        \"samples\": [\n          \"sirt6\",\n          \"otud6b\",\n          \"vdr\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"Disease\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 1194,\n        \"samples\": [\n          \"ORPHA:580933\",\n          \"ORPHA:505237\",\n          \"ORPHA:93160\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"disease_name\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 1194,\n        \"samples\": [\n          \"Orphanet: Lethal brain and heart developmental defects\",\n          \"Orphanet: Early-onset seizures-distal limb anomalies-facial dysmorphism-global developmental delay syndrome\",\n          \"Orphanet: Hypocalcemic vitamin D-resistant rickets\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"Syn\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 942,\n        \"samples\": [\n          \"CABV syndrome, CANVAS, Cerebellar ataxia with bilateral vestibulopathy syndrome\",\n          \"3-methylglutaconic aciduria-epilepsy-spasticity-severe intellectual disability syndrome, MGA9\",\n          \"Nanophthalmia\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"disease_des/def\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 1039,\n        \"samples\": [\n          \"A rare ectodermal dysplasia syndrome characterized by the association of lower eyelid ectropion, upper eyelid distichiasis, euryblepharon, bilateral cleft lip and palate, and conical teeth.\",\n          \"A rare genetic disorder caused by deletions in the long arm of chromosome 11 (11q) and mainly characterized by craniofacial dysmorphism, congenital heart disease, intellectual disability, Paris Trousseau bleeding disorder, structural kidney defects and immunodeficiency.\",\n          \"A rare, autosomal recessive autoinflammatory disorder characterized by early-onset erythematous popular/nodular skin eruptions, recurrent fever, possible joint contractures, lipodystrophy, erythematous inflammatory skin changes, joint and muscle involvement (joint contractures, arthralgia, muscle weakness), and hepatosplenomegaly.\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"Clinicalfeatures/Clinical_des\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 475,\n        \"samples\": [\n          \"Clinical features develop perinatally and during infancy. Patients tend to be born preterm. Hyperkeratosis is usually present at birth and progresses to a generalized ichthyosis, particularly prominent on flexural areas, the nape of the neck, the trunk and the extremities. Pruritus is a prominent feature. Erythematous dermatitis is often present at birth, and then tends to fade with increasing age. Neurological signs appear during the first two years of life and consist of delay in reaching motor milestones due to spastic diplegia or, much less commonly, spastic tetraplegia. Approximately one-half of patients are non-ambulatory. Seizures occur in about 40% of cases. Intellectual deficit varies from mild to severe, although rare patients with normal intellect have been reported. Delayed speech and dysarthria are common. Ophthalmologic involvement is often present and is characterized by retinal crystalline inclusions (so-called glistening white dots) surrounding the fovea. Photophobia and myopia are common.\",\n          \"Most patients display a severe phenotype that presents in infancy, usually from the neonatal period up until 12 months of age. The disease manifests as hypoketotic hypoglycemia, metabolic acidosis, hypotonia, liver involvement with hepatic encephalopathy, cardiomyopathy and arrhythmias. Clinical presentation is frequently preceded by fasting and/or intercurrent illness and often presents with hypoketotic hypoglycemia. Chronic peripheral neuropathy and pigmentary retinopathy develop over time in many surviving patients. Rarer presentations of LCHADD are sudden cardiac arrest or sudden infant death. HELLP syndrome (see this term) often occurs in pregnant women carrying a fetus affected with LCHADD.\",\n          \"The median age at diagnosis is 60-65 years. The disease is extremely rare in children. Follicular lymphoma is located primarily in the lymph nodes, but can also involve the spleen, bone marrow, peripheral blood and Waldeyer's ring. The skin and central nervous system are affected in rare cases. Symptoms appear at an advanced stage of the disease and can include fever, night sweats and weight loss. At diagnosis, patients usually present with adenopathy and, in 50% of cases, splenomegaly.\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    }\n  ]\n}"
            }
          },
          "metadata": {},
          "execution_count": 131
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "https://www.omim.org/entry/313490"
      ],
      "metadata": {
        "id": "eAMU8-AB8LDE"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "import pandas as pd\n",
        "# Read the contents of omim_info.txt into a Pandas DataFrame\n",
        "omim_info = pd.read_csv('omim_info.txt', sep='\\t',  header=3)\n",
        "# Display the DataFrame\n",
        "display(omim_info.head())"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 206
        },
        "id": "DdrRrZP-755C",
        "outputId": "6966a455-13e7-4ca2-a67a-c9e123c4ea8f"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "                                         # Phenotype  \\\n",
              "0       17,20-lyase deficiency, isolated, 202110 (3)   \n",
              "1  17-alpha-hydroxylase/17,20-lyase deficiency, 2...   \n",
              "2   2,4-dienoyl-CoA reductase deficiency, 616034 (3)   \n",
              "3              2-methylbutyrylglycinuria, 610006 (3)   \n",
              "4                         3-M syndrome 1, 273750 (3)   \n",
              "\n",
              "  Gene/Locus And Other Related Symbols  MIM Number Cyto Location  \n",
              "0              CYP17A1, CYP17, P450C17    609300.0      10q24.32  \n",
              "1              CYP17A1, CYP17, P450C17    609300.0      10q24.32  \n",
              "2                NADK2, C5orf33, DECRD    615787.0        5p13.2  \n",
              "3                        ACADSB, SBCAD    600301.0      10q26.13  \n",
              "4                            CUL7, 3M1    609577.0        6p21.1  "
            ],
            "text/html": [
              "\n",
              "  <div id=\"df-563bbe9b-3bb7-4a3e-90bc-bd4c08265697\" class=\"colab-df-container\">\n",
              "    <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th># Phenotype</th>\n",
              "      <th>Gene/Locus And Other Related Symbols</th>\n",
              "      <th>MIM Number</th>\n",
              "      <th>Cyto Location</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>17,20-lyase deficiency, isolated, 202110 (3)</td>\n",
              "      <td>CYP17A1, CYP17, P450C17</td>\n",
              "      <td>609300.0</td>\n",
              "      <td>10q24.32</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>17-alpha-hydroxylase/17,20-lyase deficiency, 2...</td>\n",
              "      <td>CYP17A1, CYP17, P450C17</td>\n",
              "      <td>609300.0</td>\n",
              "      <td>10q24.32</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>2,4-dienoyl-CoA reductase deficiency, 616034 (3)</td>\n",
              "      <td>NADK2, C5orf33, DECRD</td>\n",
              "      <td>615787.0</td>\n",
              "      <td>5p13.2</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>2-methylbutyrylglycinuria, 610006 (3)</td>\n",
              "      <td>ACADSB, SBCAD</td>\n",
              "      <td>600301.0</td>\n",
              "      <td>10q26.13</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>3-M syndrome 1, 273750 (3)</td>\n",
              "      <td>CUL7, 3M1</td>\n",
              "      <td>609577.0</td>\n",
              "      <td>6p21.1</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>\n",
              "    <div class=\"colab-df-buttons\">\n",
              "\n",
              "  <div class=\"colab-df-container\">\n",
              "    <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-563bbe9b-3bb7-4a3e-90bc-bd4c08265697')\"\n",
              "            title=\"Convert this dataframe to an interactive table.\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\" viewBox=\"0 -960 960 960\">\n",
              "    <path d=\"M120-120v-720h720v720H120Zm60-500h600v-160H180v160Zm220 220h160v-160H400v160Zm0 220h160v-160H400v160ZM180-400h160v-160H180v160Zm440 0h160v-160H620v160ZM180-180h160v-160H180v160Zm440 0h160v-160H620v160Z\"/>\n",
              "  </svg>\n",
              "    </button>\n",
              "\n",
              "  <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    .colab-df-buttons div {\n",
              "      margin-bottom: 4px;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "    <script>\n",
              "      const buttonEl =\n",
              "        document.querySelector('#df-563bbe9b-3bb7-4a3e-90bc-bd4c08265697 button.colab-df-convert');\n",
              "      buttonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "      async function convertToInteractive(key) {\n",
              "        const element = document.querySelector('#df-563bbe9b-3bb7-4a3e-90bc-bd4c08265697');\n",
              "        const dataTable =\n",
              "          await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                    [key], {});\n",
              "        if (!dataTable) return;\n",
              "\n",
              "        const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "          '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "          + ' to learn more about interactive tables.';\n",
              "        element.innerHTML = '';\n",
              "        dataTable['output_type'] = 'display_data';\n",
              "        await google.colab.output.renderOutput(dataTable, element);\n",
              "        const docLink = document.createElement('div');\n",
              "        docLink.innerHTML = docLinkHtml;\n",
              "        element.appendChild(docLink);\n",
              "      }\n",
              "    </script>\n",
              "  </div>\n",
              "\n",
              "\n",
              "<div id=\"df-0a5bf5d1-d1c3-4bd0-a794-f89a8c83f23c\">\n",
              "  <button class=\"colab-df-quickchart\" onclick=\"quickchart('df-0a5bf5d1-d1c3-4bd0-a794-f89a8c83f23c')\"\n",
              "            title=\"Suggest charts\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "<svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "     width=\"24px\">\n",
              "    <g>\n",
              "        <path d=\"M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zM9 17H7v-7h2v7zm4 0h-2V7h2v10zm4 0h-2v-4h2v4z\"/>\n",
              "    </g>\n",
              "</svg>\n",
              "  </button>\n",
              "\n",
              "<style>\n",
              "  .colab-df-quickchart {\n",
              "      --bg-color: #E8F0FE;\n",
              "      --fill-color: #1967D2;\n",
              "      --hover-bg-color: #E2EBFA;\n",
              "      --hover-fill-color: #174EA6;\n",
              "      --disabled-fill-color: #AAA;\n",
              "      --disabled-bg-color: #DDD;\n",
              "  }\n",
              "\n",
              "  [theme=dark] .colab-df-quickchart {\n",
              "      --bg-color: #3B4455;\n",
              "      --fill-color: #D2E3FC;\n",
              "      --hover-bg-color: #434B5C;\n",
              "      --hover-fill-color: #FFFFFF;\n",
              "      --disabled-bg-color: #3B4455;\n",
              "      --disabled-fill-color: #666;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart {\n",
              "    background-color: var(--bg-color);\n",
              "    border: none;\n",
              "    border-radius: 50%;\n",
              "    cursor: pointer;\n",
              "    display: none;\n",
              "    fill: var(--fill-color);\n",
              "    height: 32px;\n",
              "    padding: 0;\n",
              "    width: 32px;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart:hover {\n",
              "    background-color: var(--hover-bg-color);\n",
              "    box-shadow: 0 1px 2px rgba(60, 64, 67, 0.3), 0 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "    fill: var(--button-hover-fill-color);\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart-complete:disabled,\n",
              "  .colab-df-quickchart-complete:disabled:hover {\n",
              "    background-color: var(--disabled-bg-color);\n",
              "    fill: var(--disabled-fill-color);\n",
              "    box-shadow: none;\n",
              "  }\n",
              "\n",
              "  .colab-df-spinner {\n",
              "    border: 2px solid var(--fill-color);\n",
              "    border-color: transparent;\n",
              "    border-bottom-color: var(--fill-color);\n",
              "    animation:\n",
              "      spin 1s steps(1) infinite;\n",
              "  }\n",
              "\n",
              "  @keyframes spin {\n",
              "    0% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "      border-left-color: var(--fill-color);\n",
              "    }\n",
              "    20% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    30% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    40% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    60% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    80% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "    90% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "  }\n",
              "</style>\n",
              "\n",
              "  <script>\n",
              "    async function quickchart(key) {\n",
              "      const quickchartButtonEl =\n",
              "        document.querySelector('#' + key + ' button');\n",
              "      quickchartButtonEl.disabled = true;  // To prevent multiple clicks.\n",
              "      quickchartButtonEl.classList.add('colab-df-spinner');\n",
              "      try {\n",
              "        const charts = await google.colab.kernel.invokeFunction(\n",
              "            'suggestCharts', [key], {});\n",
              "      } catch (error) {\n",
              "        console.error('Error during call to suggestCharts:', error);\n",
              "      }\n",
              "      quickchartButtonEl.classList.remove('colab-df-spinner');\n",
              "      quickchartButtonEl.classList.add('colab-df-quickchart-complete');\n",
              "    }\n",
              "    (() => {\n",
              "      let quickchartButtonEl =\n",
              "        document.querySelector('#df-0a5bf5d1-d1c3-4bd0-a794-f89a8c83f23c button');\n",
              "      quickchartButtonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "    })();\n",
              "  </script>\n",
              "</div>\n",
              "    </div>\n",
              "  </div>\n"
            ],
            "application/vnd.google.colaboratory.intrinsic+json": {
              "type": "dataframe",
              "summary": "{\n  \"name\": \"display(omim_info\",\n  \"rows\": 5,\n  \"fields\": [\n    {\n      \"column\": \"# Phenotype\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 5,\n        \"samples\": [\n          \"17-alpha-hydroxylase/17,20-lyase deficiency, 202110 (3)\",\n          \"3-M syndrome 1, 273750 (3)\",\n          \"2,4-dienoyl-CoA reductase deficiency, 616034 (3)\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"Gene/Locus And Other Related Symbols\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 4,\n        \"samples\": [\n          \"NADK2, C5orf33, DECRD\",\n          \"CUL7, 3M1\",\n          \"CYP17A1, CYP17, P450C17\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"MIM Number\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 5525.867669425318,\n        \"min\": 600301.0,\n        \"max\": 615787.0,\n        \"num_unique_values\": 4,\n        \"samples\": [\n          615787.0,\n          609577.0,\n          609300.0\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"Cyto Location\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 4,\n        \"samples\": [\n          \"5p13.2\",\n          \"6p21.1\",\n          \"10q24.32\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    }\n  ]\n}"
            }
          },
          "metadata": {}
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "import requests\n",
        "from bs4 import BeautifulSoup\n",
        "\n",
        "def get_description_and_clinical_features(num):\n",
        "    # URL of the page to scrape\n",
        "    url = f\"https://api.omim.org/api/entry?mimNumber={num}&apiKey=FNQbzW08SSWjNwMbMFBrhw&include=text\"\n",
        "    # Send a GET request to the URL\n",
        "    response = requests.get(url)\n",
        "    # Parse the HTML content\n",
        "    soup = BeautifulSoup(response.text, 'html.parser')\n",
        "\n",
        "    # Find the description section\n",
        "    description_section = soup.find('textsectionname', string=\"description\")\n",
        "    # If description section is found, get the content\n",
        "    description_content = description_section.find_next_sibling('textsectioncontent').get_text(strip=True) if description_section else None\n",
        "\n",
        "    # Find the clinical features section\n",
        "    clinical_features_section = soup.find('textsectionname', string=\"clinicalFeatures\")\n",
        "    # If clinical features section is found, get the content\n",
        "    clinical_features_content = clinical_features_section.find_next_sibling('textsectioncontent').get_text(strip=True) if clinical_features_section else None\n",
        "\n",
        "    return description_content, clinical_features_content\n",
        "\n",
        "get_description_and_clinical_features(str(620283))\n",
        "\n",
        "\n",
        "def find_dis_omim(number_to_search):\n",
        " result_df = omim_info[omim_info['# Phenotype'].str.contains(number_to_search)]\n",
        " result_df.index=range(result_df.shape[0])\n",
        " description_, _clinical_features=get_description_and_clinical_features(number_to_search)\n",
        " return result_df['# Phenotype'][0] , result_df['Gene/Locus And Other Related Symbols'][0],description_, _clinical_features\n",
        "\n",
        "\n",
        "number_to_search = \"202110\"\n",
        "find_dis_omim(number_to_search)\n",
        "find_dis_omim('620283')\n",
        "\n",
        "def find_dis_from_omim(y):\n",
        "    if y[:5] == 'OMIM:':\n",
        "      try:\n",
        "        return find_dis_omim(str(y[5:]))\n",
        "      except:\n",
        "        pass\n",
        "    return None, None, None, None\n",
        "df['disease_name'], df['Syn'], df['disease_des/def'], df['Clinicalfeatures/Clinical_des'] = zip(*df.apply(lambda x: find_dis_from_omim(x['Disease']), axis=1))\n"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "VSG6kRXmySWZ",
        "outputId": "28f8603e-8939-43c8-dfe6-78376c034355"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "/usr/lib/python3.10/html/parser.py:170: XMLParsedAsHTMLWarning: It looks like you're parsing an XML document using an HTML parser. If this really is an HTML document (maybe it's XHTML?), you can ignore or filter this warning. If it's XML, you should know that using an XML parser will be more reliable. To parse this document as XML, make sure you have the lxml package installed, and pass the keyword argument `features=\"xml\"` into the BeautifulSoup constructor.\n",
            "  k = self.parse_starttag(i)\n",
            "<ipython-input-133-ba5b93bfaa3b>:45: SettingWithCopyWarning: \n",
            "A value is trying to be set on a copy of a slice from a DataFrame.\n",
            "Try using .loc[row_indexer,col_indexer] = value instead\n",
            "\n",
            "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
            "  df['disease_name'], df['Syn'], df['disease_des/def'], df['Clinicalfeatures/Clinical_des'] = zip(*df.apply(lambda x: find_dis_from_omim(x['Disease']), axis=1))\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "df2= df.dropna(subset=['disease_name'])\n",
        "df_unique_2 = df2.drop_duplicates(subset=['disease_name'])\n",
        "df_unique_2"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 805
        },
        "id": "6QUSxl8P82vl",
        "outputId": "1c691b45-bfdc-45d9-e72a-0872bcd78f0b"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "       Unnamed: 0     Gene      Disease  \\\n",
              "197           197    kif4a  OMIM:313490   \n",
              "359           359   epha10  OMIM:620283   \n",
              "657           657  slc14a1  OMIM:111000   \n",
              "674           674   gabrb1  OMIM:617153   \n",
              "707           707      mos  OMIM:620383   \n",
              "...           ...      ...          ...   \n",
              "56117       56117     rora  OMIM:618060   \n",
              "56313       56313    kcnu1  OMIM:620196   \n",
              "56769       56769   map3k8  OMIM:211980   \n",
              "56922       56922     mdh1  OMIM:618959   \n",
              "57287       57287    mpeg1  OMIM:619223   \n",
              "\n",
              "                                            disease_name  \\\n",
              "197    Taurodontism, microdontia, and dens invaginatu...   \n",
              "359         ?Deafness, autosomal dominant 88, 620283 (3)   \n",
              "657                      [Blood group, Kidd], 111000 (3)   \n",
              "674    Developmental and epileptic encephalopathy 45,...   \n",
              "707    Oocyte/zygote/embryo maturation arrest 20, 620...   \n",
              "...                                                  ...   \n",
              "56117  Intellectual developmental disorder with or wi...   \n",
              "56313               Spermatogenic failure 79, 620196 (3)   \n",
              "56769  Adenocarcinoma of lung, response to tyrosine k...   \n",
              "56922  ?Developmental and epileptic encephalopathy 88...   \n",
              "57287                    Immunodeficiency 77, 619223 (3)   \n",
              "\n",
              "                              Syn  \\\n",
              "197    KIF4A, KIF4, XLID100, TMDI   \n",
              "359                EPHA10, DFNA88   \n",
              "657         SLC14A1, JK, UTE, UT1   \n",
              "674                 GABRB1, DEE45   \n",
              "707             MOS, MSV, OZEMA20   \n",
              "...                           ...   \n",
              "56117                RORA, IDDECA   \n",
              "56313         KCNU1, SLO3, SPGF79   \n",
              "56769                EGFR, NISBD2   \n",
              "56922            MDH1, DEE88, KAR   \n",
              "57287          MPEG1, MPG1, IMD77   \n",
              "\n",
              "                                         disease_des/def  \\\n",
              "197    Taurodontism, microdontia, and dens invaginatu...   \n",
              "359    Autosomal dominant deafness-88 (DFNA88) is cha...   \n",
              "657    The Kidd blood group locus encodes a urea tran...   \n",
              "674    Developmental and epileptic encephalopathy-45 ...   \n",
              "707    Oocyte/zygote/embryo maturation arrest-20 (OZE...   \n",
              "...                                                  ...   \n",
              "56117                                               None   \n",
              "56313  Spermatogenic failure-79 (SPGF79) is character...   \n",
              "56769  Lung cancer is the leading cause of cancer dea...   \n",
              "56922  Developmental and epileptic encephalopathy-88 ...   \n",
              "57287  Immunodeficiency-77 (IMD77) is an immunologic ...   \n",
              "\n",
              "                           Clinicalfeatures/Clinical_des  \n",
              "197    {1:Casamassimo et al. (1978)} described a fami...  \n",
              "359    {2:Jiang et al. (2011)} reported a large 4-gen...  \n",
              "657    Although Jk-null red blood cells have reduced ...  \n",
              "674    The {2:Epi4K Consortium and Epilepsy Phenome/G...  \n",
              "707    {3:Zhang et al. (2021)} reported 3 unrelated i...  \n",
              "...                                                  ...  \n",
              "56117  {1:Guissart et al. (2018)} reported 11 unrelat...  \n",
              "56313  {2:Lv et al. (2022)} reported a 29-year-old in...  \n",
              "56769  {23:Joishy et al. (1977)} described identical ...  \n",
              "56922  {1:Broeks et al. (2019)} reported 2 affected f...  \n",
              "57287  {1:McCormack et al. (2017)} identified 4 unrel...  \n",
              "\n",
              "[310 rows x 7 columns]"
            ],
            "text/html": [
              "\n",
              "  <div id=\"df-d25971ef-6dcd-4f8e-a21c-0aaafb4791d7\" class=\"colab-df-container\">\n",
              "    <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>Unnamed: 0</th>\n",
              "      <th>Gene</th>\n",
              "      <th>Disease</th>\n",
              "      <th>disease_name</th>\n",
              "      <th>Syn</th>\n",
              "      <th>disease_des/def</th>\n",
              "      <th>Clinicalfeatures/Clinical_des</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>197</th>\n",
              "      <td>197</td>\n",
              "      <td>kif4a</td>\n",
              "      <td>OMIM:313490</td>\n",
              "      <td>Taurodontism, microdontia, and dens invaginatu...</td>\n",
              "      <td>KIF4A, KIF4, XLID100, TMDI</td>\n",
              "      <td>Taurodontism, microdontia, and dens invaginatu...</td>\n",
              "      <td>{1:Casamassimo et al. (1978)} described a fami...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>359</th>\n",
              "      <td>359</td>\n",
              "      <td>epha10</td>\n",
              "      <td>OMIM:620283</td>\n",
              "      <td>?Deafness, autosomal dominant 88, 620283 (3)</td>\n",
              "      <td>EPHA10, DFNA88</td>\n",
              "      <td>Autosomal dominant deafness-88 (DFNA88) is cha...</td>\n",
              "      <td>{2:Jiang et al. (2011)} reported a large 4-gen...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>657</th>\n",
              "      <td>657</td>\n",
              "      <td>slc14a1</td>\n",
              "      <td>OMIM:111000</td>\n",
              "      <td>[Blood group, Kidd], 111000 (3)</td>\n",
              "      <td>SLC14A1, JK, UTE, UT1</td>\n",
              "      <td>The Kidd blood group locus encodes a urea tran...</td>\n",
              "      <td>Although Jk-null red blood cells have reduced ...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>674</th>\n",
              "      <td>674</td>\n",
              "      <td>gabrb1</td>\n",
              "      <td>OMIM:617153</td>\n",
              "      <td>Developmental and epileptic encephalopathy 45,...</td>\n",
              "      <td>GABRB1, DEE45</td>\n",
              "      <td>Developmental and epileptic encephalopathy-45 ...</td>\n",
              "      <td>The {2:Epi4K Consortium and Epilepsy Phenome/G...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>707</th>\n",
              "      <td>707</td>\n",
              "      <td>mos</td>\n",
              "      <td>OMIM:620383</td>\n",
              "      <td>Oocyte/zygote/embryo maturation arrest 20, 620...</td>\n",
              "      <td>MOS, MSV, OZEMA20</td>\n",
              "      <td>Oocyte/zygote/embryo maturation arrest-20 (OZE...</td>\n",
              "      <td>{3:Zhang et al. (2021)} reported 3 unrelated i...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>...</th>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>56117</th>\n",
              "      <td>56117</td>\n",
              "      <td>rora</td>\n",
              "      <td>OMIM:618060</td>\n",
              "      <td>Intellectual developmental disorder with or wi...</td>\n",
              "      <td>RORA, IDDECA</td>\n",
              "      <td>None</td>\n",
              "      <td>{1:Guissart et al. (2018)} reported 11 unrelat...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>56313</th>\n",
              "      <td>56313</td>\n",
              "      <td>kcnu1</td>\n",
              "      <td>OMIM:620196</td>\n",
              "      <td>Spermatogenic failure 79, 620196 (3)</td>\n",
              "      <td>KCNU1, SLO3, SPGF79</td>\n",
              "      <td>Spermatogenic failure-79 (SPGF79) is character...</td>\n",
              "      <td>{2:Lv et al. (2022)} reported a 29-year-old in...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>56769</th>\n",
              "      <td>56769</td>\n",
              "      <td>map3k8</td>\n",
              "      <td>OMIM:211980</td>\n",
              "      <td>Adenocarcinoma of lung, response to tyrosine k...</td>\n",
              "      <td>EGFR, NISBD2</td>\n",
              "      <td>Lung cancer is the leading cause of cancer dea...</td>\n",
              "      <td>{23:Joishy et al. (1977)} described identical ...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>56922</th>\n",
              "      <td>56922</td>\n",
              "      <td>mdh1</td>\n",
              "      <td>OMIM:618959</td>\n",
              "      <td>?Developmental and epileptic encephalopathy 88...</td>\n",
              "      <td>MDH1, DEE88, KAR</td>\n",
              "      <td>Developmental and epileptic encephalopathy-88 ...</td>\n",
              "      <td>{1:Broeks et al. (2019)} reported 2 affected f...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>57287</th>\n",
              "      <td>57287</td>\n",
              "      <td>mpeg1</td>\n",
              "      <td>OMIM:619223</td>\n",
              "      <td>Immunodeficiency 77, 619223 (3)</td>\n",
              "      <td>MPEG1, MPG1, IMD77</td>\n",
              "      <td>Immunodeficiency-77 (IMD77) is an immunologic ...</td>\n",
              "      <td>{1:McCormack et al. (2017)} identified 4 unrel...</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "<p>310 rows × 7 columns</p>\n",
              "</div>\n",
              "    <div class=\"colab-df-buttons\">\n",
              "\n",
              "  <div class=\"colab-df-container\">\n",
              "    <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-d25971ef-6dcd-4f8e-a21c-0aaafb4791d7')\"\n",
              "            title=\"Convert this dataframe to an interactive table.\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\" viewBox=\"0 -960 960 960\">\n",
              "    <path d=\"M120-120v-720h720v720H120Zm60-500h600v-160H180v160Zm220 220h160v-160H400v160Zm0 220h160v-160H400v160ZM180-400h160v-160H180v160Zm440 0h160v-160H620v160ZM180-180h160v-160H180v160Zm440 0h160v-160H620v160Z\"/>\n",
              "  </svg>\n",
              "    </button>\n",
              "\n",
              "  <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    .colab-df-buttons div {\n",
              "      margin-bottom: 4px;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "    <script>\n",
              "      const buttonEl =\n",
              "        document.querySelector('#df-d25971ef-6dcd-4f8e-a21c-0aaafb4791d7 button.colab-df-convert');\n",
              "      buttonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "      async function convertToInteractive(key) {\n",
              "        const element = document.querySelector('#df-d25971ef-6dcd-4f8e-a21c-0aaafb4791d7');\n",
              "        const dataTable =\n",
              "          await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                    [key], {});\n",
              "        if (!dataTable) return;\n",
              "\n",
              "        const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "          '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "          + ' to learn more about interactive tables.';\n",
              "        element.innerHTML = '';\n",
              "        dataTable['output_type'] = 'display_data';\n",
              "        await google.colab.output.renderOutput(dataTable, element);\n",
              "        const docLink = document.createElement('div');\n",
              "        docLink.innerHTML = docLinkHtml;\n",
              "        element.appendChild(docLink);\n",
              "      }\n",
              "    </script>\n",
              "  </div>\n",
              "\n",
              "\n",
              "<div id=\"df-182dd7b9-1469-4faf-b3a0-889ec205824e\">\n",
              "  <button class=\"colab-df-quickchart\" onclick=\"quickchart('df-182dd7b9-1469-4faf-b3a0-889ec205824e')\"\n",
              "            title=\"Suggest charts\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "<svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "     width=\"24px\">\n",
              "    <g>\n",
              "        <path d=\"M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zM9 17H7v-7h2v7zm4 0h-2V7h2v10zm4 0h-2v-4h2v4z\"/>\n",
              "    </g>\n",
              "</svg>\n",
              "  </button>\n",
              "\n",
              "<style>\n",
              "  .colab-df-quickchart {\n",
              "      --bg-color: #E8F0FE;\n",
              "      --fill-color: #1967D2;\n",
              "      --hover-bg-color: #E2EBFA;\n",
              "      --hover-fill-color: #174EA6;\n",
              "      --disabled-fill-color: #AAA;\n",
              "      --disabled-bg-color: #DDD;\n",
              "  }\n",
              "\n",
              "  [theme=dark] .colab-df-quickchart {\n",
              "      --bg-color: #3B4455;\n",
              "      --fill-color: #D2E3FC;\n",
              "      --hover-bg-color: #434B5C;\n",
              "      --hover-fill-color: #FFFFFF;\n",
              "      --disabled-bg-color: #3B4455;\n",
              "      --disabled-fill-color: #666;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart {\n",
              "    background-color: var(--bg-color);\n",
              "    border: none;\n",
              "    border-radius: 50%;\n",
              "    cursor: pointer;\n",
              "    display: none;\n",
              "    fill: var(--fill-color);\n",
              "    height: 32px;\n",
              "    padding: 0;\n",
              "    width: 32px;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart:hover {\n",
              "    background-color: var(--hover-bg-color);\n",
              "    box-shadow: 0 1px 2px rgba(60, 64, 67, 0.3), 0 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "    fill: var(--button-hover-fill-color);\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart-complete:disabled,\n",
              "  .colab-df-quickchart-complete:disabled:hover {\n",
              "    background-color: var(--disabled-bg-color);\n",
              "    fill: var(--disabled-fill-color);\n",
              "    box-shadow: none;\n",
              "  }\n",
              "\n",
              "  .colab-df-spinner {\n",
              "    border: 2px solid var(--fill-color);\n",
              "    border-color: transparent;\n",
              "    border-bottom-color: var(--fill-color);\n",
              "    animation:\n",
              "      spin 1s steps(1) infinite;\n",
              "  }\n",
              "\n",
              "  @keyframes spin {\n",
              "    0% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "      border-left-color: var(--fill-color);\n",
              "    }\n",
              "    20% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    30% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    40% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    60% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    80% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "    90% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "  }\n",
              "</style>\n",
              "\n",
              "  <script>\n",
              "    async function quickchart(key) {\n",
              "      const quickchartButtonEl =\n",
              "        document.querySelector('#' + key + ' button');\n",
              "      quickchartButtonEl.disabled = true;  // To prevent multiple clicks.\n",
              "      quickchartButtonEl.classList.add('colab-df-spinner');\n",
              "      try {\n",
              "        const charts = await google.colab.kernel.invokeFunction(\n",
              "            'suggestCharts', [key], {});\n",
              "      } catch (error) {\n",
              "        console.error('Error during call to suggestCharts:', error);\n",
              "      }\n",
              "      quickchartButtonEl.classList.remove('colab-df-spinner');\n",
              "      quickchartButtonEl.classList.add('colab-df-quickchart-complete');\n",
              "    }\n",
              "    (() => {\n",
              "      let quickchartButtonEl =\n",
              "        document.querySelector('#df-182dd7b9-1469-4faf-b3a0-889ec205824e button');\n",
              "      quickchartButtonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "    })();\n",
              "  </script>\n",
              "</div>\n",
              "    </div>\n",
              "  </div>\n"
            ],
            "application/vnd.google.colaboratory.intrinsic+json": {
              "type": "dataframe",
              "variable_name": "df_unique_2",
              "summary": "{\n  \"name\": \"df_unique_2\",\n  \"rows\": 310,\n  \"fields\": [\n    {\n      \"column\": \"Unnamed: 0\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 16361,\n        \"min\": 197,\n        \"max\": 57287,\n        \"num_unique_values\": 310,\n        \"samples\": [\n          53219,\n          1383,\n          6679\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"Gene\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 310,\n        \"samples\": [\n          \"scarb1\",\n          \"lgr4\",\n          \"prkag3\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"Disease\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 310,\n        \"samples\": [\n          \"OMIM:610762\",\n          \"OMIM:619613\",\n          \"OMIM:619030\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"disease_name\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 310,\n        \"samples\": [\n          \"[High density lipoprotein cholesterol level QTL6], 610762 (3)\",\n          \"Delayed puberty, self-limited, 619613 (3)\",\n          \"[Skeletal muscle glycogen content and metabolism QTL], 619030 (3)\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"Syn\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 309,\n        \"samples\": [\n          \"SCARB1, CD36L1, CLA1, HDLQTL6\",\n          \"LGR4, GPR48, BNMD17, DPSL\",\n          \"PRKAG3, SMGMQTL\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"disease_des/def\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 237,\n        \"samples\": [\n          \"The antibody defining the very common antigen Yt(a) was the cause of a cross-matching difficulty investigated by {3:Eaton et al. (1956)}. It was presumed to be the result of previous transfusions. Among 1,051 English people, 4 negatives were found. Positives showed 2 grades of strength of reaction; on the assumption that the weaker reactors represented heterozygotes, an estimate of gene frequency simply by counting was possible.\\n\\n{8:Telen et al. (1990)} found that the Yt (Cartwright) red cell antigen resided on an unidentified phosphatidylinositol (PI)-linked protein. {9:Telen and Whitsett (1992)} identified a patient with the hitherto unreported Yt(a-b-) phenotype in whom studies allowed localization of the Yt antigens to the acetylcholinesterase molecule. {9:Telen and Whitsett (1992)} found that binding of antibodies to several membrane proteins including CD55 ({125240}), CD58 ({153420}), and CD59 ({107271}) were normal, whereas 4 monoclonal antibodies to different acetylcholinesterase epitopes reacted only weakly with Yt(a-b-) erythrocytes. In addition, enzymatic assay of acetylcholinesterase activity of Yt(a-b-) erythrocytes demonstrated only 15% of the normal amount of enzyme activity. The use of anti-Yt(a) in radioimmunoprecipitation experiments demonstrated the expected 160 kD of the acetylcholinesterase molecule from normal erythrocyte membrane proteins, but not from Yt(a-b-) erythrocytes. {7:Spring et al. (1992)} obtained similar results. Thus, acetylcholinesterase is the PI-linked protein that represents the Yt antigen.\",\n          \"Autosomal recessive congenital ichthyosis (ARCI) is a heterogeneous group of disorders of keratinization characterized primarily by abnormal skin scaling over the whole body. These disorders are limited to skin, with approximately two-thirds of patients presenting severe symptoms. The main skin phenotypes are lamellar ichthyosis (LI) and nonbullous congenital ichthyosiform erythroderma (NCIE), although phenotypic overlap within the same patient or among patients from the same family can occur (summary by {2:Fischer, 2009}). Neither histopathologic findings nor ultrastructural features clearly distinguish between NCIE and LI. In addition, mutations in several genes have been shown to cause both lamellar and nonbullous ichthyosiform erythrodermal phenotypes ({1:Akiyama et al., 2003}). At the First Ichthyosis Consensus Conference in Soreze in 2009, the term 'autosomal recessive congenital ichthyosis' (ARCI) was designated to encompass LI, NCIE, and harlequin ichthyosis (ARCI4B; {242500}) ({4:Oji et al., 2010}).\\n\\nFor a general phenotypic description and discussion of genetic heterogeneity of autosomal recessive congenital ichthyosis, see ARCI1 ({242300}).\",\n          \"Autosomal recessive spinocerebellar ataxia-30 (SCAR30) is a progressive neurologic disorder characterized by childhood-onset global developmental delay with variably impaired intellectual development, motor dysfunction, and cerebellar ataxia. Affected individuals may also have psychiatric abnormalities, such as obsessive behavior, psychotic episodes, or hallucinations. Brain imaging usually shows cerebellar atrophy, although this may be an age-dependent feature (summary by {3:Langer et al., 2018}).\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"Clinicalfeatures/Clinical_des\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 265,\n        \"samples\": [\n          \"{1:Chilton et al. (2020)} reported 14 unrelated patients, ranging in age from 2 to 29 years, who were ascertained through the GeneMatcher Program after exome sequencing identified de novo heterozygous mutations in the CDC42BPB gene. One patient (individual 4) was a male fetus with cystic hygroma and progressive ventriculomegaly found on prenatal ultrasound; the pregnancy was terminated at 27 weeks' gestation. All except 1 (individual 8) of the other 13 patients had global developmental delay, often with impaired intellectual development, speech delay, and behavioral abnormalities, including ADHD, aggressive behavior, anxiety, and repetitive or stereotypic movements. Eight were diagnosed with autism spectrum disorder, including a 14-year-old boy with Asperger syndrome (individual 2). Several patients had hypotonia or unsteady gait, 3 had seizures, and 1 had EEG abnormalities. One of the patients with seizures (individual 7) was unable to walk at age 12 years and had severe speech delay with only a few words. Four patients had ophthalmologic findings, including ptosis, strabismus, esotropia, amblyopia, refractory errors, and septooptic dysplasia with optic atrophy (1 patient). Brain imaging showed nonspecific abnormalities in 4 patients, including cerebellar hypoplasia, thin or absent corpus callosum, white matter loss, and enlarged ventricles. Brain imaging was normal in other patients tested. About half of individuals had dysmorphic facial features, including macrocephaly, abnormal head shape, broad or high forehead, sparse scalp hair, sparse eyebrows, low-set ears, up- or downslanting palpebral fissures, hypertelorism, high nasal root, flat nasal bridge, broad nasal tip, short philtrum, large mouth, thin upper lip, and widely spaced teeth. A few had distal skeletal anomalies of the hands and feet or pectus abnormalities. Less common features included cryptorchidism, constipation, recurrent otitis media, micropenis, kidney abnormalities, and joint hypermobility. Rare features observed in a single patient each included acquired microcephaly, congenital diaphragmatic hernia with intestinal malrotation, patent foramen ovale, patent ductus arteriosus, and scoliosis.\",\n          \"{1:Csukasi et al. (2018)} reported 2 sibs (R07-429A and R07-429B) from a consanguineous family with spondyloepimetaphyseal dysplasia characterized by rhizomelia and mesomelia with significant anterior bowing of all limbs. Neither sib ever walked, and both were wheelchair bound, unable to bear weight on the lower extremities. Radiographic features included widened/flared metaphyses with irregular ossifications, moth-eaten long bones, fragmentation of the proximal metacarpals, rounded vertebral bodies, and a distinctive transverse gap in the tibiae. The sibs had significant developmental delay with abnormalities on brain MRI, as well as a severe unclassified immunodeficiency, and normal PTH levels with mild hypercalcemia. One sib showed a more severe phenotype, particularly of the immune system, and died of an Epstein-Barr virus-induced small-muscle cancer at 10 years of age. Her sib was alive at 14 years of age. The authors noted similarities between this skeletal dysplasia and metaphyseal chondrodysplasia of the Jansen type (MCDJ; {156400}), but stated that the sibs exhibited unique features that did not match any of the currently classified skeletal disorders.\",\n          \"A critical phenotypic characteristic of human asthma and an important feature of animal models of asthma is airway hyperresponsiveness ({20:Hirshman et al., 1984}; {39:Levitt and Mitzner, 1988}; {40:Levitt and Mitzner, 1989}).\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    }\n  ]\n}"
            }
          },
          "metadata": {},
          "execution_count": 134
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "concatenated_df = pd.concat([df_unique_2, df_unique_1], ignore_index=True)\n",
        "concatenated_df"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 857
        },
        "id": "02xfwZbvG6l5",
        "outputId": "9690359e-8c32-450f-820b-c7f44addb05f"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "      Unnamed: 0     Gene       Disease  \\\n",
              "0            197    kif4a   OMIM:313490   \n",
              "1            359   epha10   OMIM:620283   \n",
              "2            657  slc14a1   OMIM:111000   \n",
              "3            674   gabrb1   OMIM:617153   \n",
              "4            707      mos   OMIM:620383   \n",
              "...          ...      ...           ...   \n",
              "1499       56605  cyp51a1  ORPHA:521432   \n",
              "1500       56616     g6pd  ORPHA:466026   \n",
              "1501       56682     dhfr  ORPHA:319651   \n",
              "1502       56970      gp6   ORPHA:98885   \n",
              "1503       57224     fzd5   ORPHA:98942   \n",
              "\n",
              "                                           disease_name  \\\n",
              "0     Taurodontism, microdontia, and dens invaginatu...   \n",
              "1          ?Deafness, autosomal dominant 88, 620283 (3)   \n",
              "2                       [Blood group, Kidd], 111000 (3)   \n",
              "3     Developmental and epileptic encephalopathy 45,...   \n",
              "4     Oocyte/zygote/embryo maturation arrest 20, 620...   \n",
              "...                                                 ...   \n",
              "1499  Orphanet: Congenital cataract-severe neonatal ...   \n",
              "1500  Orphanet: Class I glucose-6-phosphate dehydrog...   \n",
              "1501  Orphanet: Constitutional megaloblastic anemia ...   \n",
              "1502  Orphanet: Bleeding diathesis due to glycoprote...   \n",
              "1503           Orphanet: Coloboma of choroid and retina   \n",
              "\n",
              "                                                    Syn  \\\n",
              "0                            KIF4A, KIF4, XLID100, TMDI   \n",
              "1                                        EPHA10, DFNA88   \n",
              "2                                 SLC14A1, JK, UTE, UT1   \n",
              "3                                         GABRB1, DEE45   \n",
              "4                                     MOS, MSV, OZEMA20   \n",
              "...                                                 ...   \n",
              "1499                                               None   \n",
              "1500  Class I G6PD deficiency, Severe hemolytic anem...   \n",
              "1501  DHFR deficiency, Dihydrofolate reductase defic...   \n",
              "1502                                               None   \n",
              "1503                                               None   \n",
              "\n",
              "                                        disease_des/def  \\\n",
              "0     Taurodontism, microdontia, and dens invaginatu...   \n",
              "1     Autosomal dominant deafness-88 (DFNA88) is cha...   \n",
              "2     The Kidd blood group locus encodes a urea tran...   \n",
              "3     Developmental and epileptic encephalopathy-45 ...   \n",
              "4     Oocyte/zygote/embryo maturation arrest-20 (OZE...   \n",
              "...                                                 ...   \n",
              "1499  A rare genetic disease characterized by congen...   \n",
              "1500  A rare constitutional hemolytic anemia due to ...   \n",
              "1501                                               None   \n",
              "1502                                               None   \n",
              "1503  Coloboma of choroid and retina is a rare, gene...   \n",
              "\n",
              "                          Clinicalfeatures/Clinical_des  \n",
              "0     {1:Casamassimo et al. (1978)} described a fami...  \n",
              "1     {2:Jiang et al. (2011)} reported a large 4-gen...  \n",
              "2     Although Jk-null red blood cells have reduced ...  \n",
              "3     The {2:Epi4K Consortium and Epilepsy Phenome/G...  \n",
              "4     {3:Zhang et al. (2021)} reported 3 unrelated i...  \n",
              "...                                                 ...  \n",
              "1499                                               None  \n",
              "1500                                               None  \n",
              "1501                                               None  \n",
              "1502                                               None  \n",
              "1503                                               None  \n",
              "\n",
              "[1504 rows x 7 columns]"
            ],
            "text/html": [
              "\n",
              "  <div id=\"df-000f79d5-ddc9-4a5e-aa59-c239a00f46bc\" class=\"colab-df-container\">\n",
              "    <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>Unnamed: 0</th>\n",
              "      <th>Gene</th>\n",
              "      <th>Disease</th>\n",
              "      <th>disease_name</th>\n",
              "      <th>Syn</th>\n",
              "      <th>disease_des/def</th>\n",
              "      <th>Clinicalfeatures/Clinical_des</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>197</td>\n",
              "      <td>kif4a</td>\n",
              "      <td>OMIM:313490</td>\n",
              "      <td>Taurodontism, microdontia, and dens invaginatu...</td>\n",
              "      <td>KIF4A, KIF4, XLID100, TMDI</td>\n",
              "      <td>Taurodontism, microdontia, and dens invaginatu...</td>\n",
              "      <td>{1:Casamassimo et al. (1978)} described a fami...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>359</td>\n",
              "      <td>epha10</td>\n",
              "      <td>OMIM:620283</td>\n",
              "      <td>?Deafness, autosomal dominant 88, 620283 (3)</td>\n",
              "      <td>EPHA10, DFNA88</td>\n",
              "      <td>Autosomal dominant deafness-88 (DFNA88) is cha...</td>\n",
              "      <td>{2:Jiang et al. (2011)} reported a large 4-gen...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>657</td>\n",
              "      <td>slc14a1</td>\n",
              "      <td>OMIM:111000</td>\n",
              "      <td>[Blood group, Kidd], 111000 (3)</td>\n",
              "      <td>SLC14A1, JK, UTE, UT1</td>\n",
              "      <td>The Kidd blood group locus encodes a urea tran...</td>\n",
              "      <td>Although Jk-null red blood cells have reduced ...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>674</td>\n",
              "      <td>gabrb1</td>\n",
              "      <td>OMIM:617153</td>\n",
              "      <td>Developmental and epileptic encephalopathy 45,...</td>\n",
              "      <td>GABRB1, DEE45</td>\n",
              "      <td>Developmental and epileptic encephalopathy-45 ...</td>\n",
              "      <td>The {2:Epi4K Consortium and Epilepsy Phenome/G...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>707</td>\n",
              "      <td>mos</td>\n",
              "      <td>OMIM:620383</td>\n",
              "      <td>Oocyte/zygote/embryo maturation arrest 20, 620...</td>\n",
              "      <td>MOS, MSV, OZEMA20</td>\n",
              "      <td>Oocyte/zygote/embryo maturation arrest-20 (OZE...</td>\n",
              "      <td>{3:Zhang et al. (2021)} reported 3 unrelated i...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>...</th>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1499</th>\n",
              "      <td>56605</td>\n",
              "      <td>cyp51a1</td>\n",
              "      <td>ORPHA:521432</td>\n",
              "      <td>Orphanet: Congenital cataract-severe neonatal ...</td>\n",
              "      <td>None</td>\n",
              "      <td>A rare genetic disease characterized by congen...</td>\n",
              "      <td>None</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1500</th>\n",
              "      <td>56616</td>\n",
              "      <td>g6pd</td>\n",
              "      <td>ORPHA:466026</td>\n",
              "      <td>Orphanet: Class I glucose-6-phosphate dehydrog...</td>\n",
              "      <td>Class I G6PD deficiency, Severe hemolytic anem...</td>\n",
              "      <td>A rare constitutional hemolytic anemia due to ...</td>\n",
              "      <td>None</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1501</th>\n",
              "      <td>56682</td>\n",
              "      <td>dhfr</td>\n",
              "      <td>ORPHA:319651</td>\n",
              "      <td>Orphanet: Constitutional megaloblastic anemia ...</td>\n",
              "      <td>DHFR deficiency, Dihydrofolate reductase defic...</td>\n",
              "      <td>None</td>\n",
              "      <td>None</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1502</th>\n",
              "      <td>56970</td>\n",
              "      <td>gp6</td>\n",
              "      <td>ORPHA:98885</td>\n",
              "      <td>Orphanet: Bleeding diathesis due to glycoprote...</td>\n",
              "      <td>None</td>\n",
              "      <td>None</td>\n",
              "      <td>None</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1503</th>\n",
              "      <td>57224</td>\n",
              "      <td>fzd5</td>\n",
              "      <td>ORPHA:98942</td>\n",
              "      <td>Orphanet: Coloboma of choroid and retina</td>\n",
              "      <td>None</td>\n",
              "      <td>Coloboma of choroid and retina is a rare, gene...</td>\n",
              "      <td>None</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "<p>1504 rows × 7 columns</p>\n",
              "</div>\n",
              "    <div class=\"colab-df-buttons\">\n",
              "\n",
              "  <div class=\"colab-df-container\">\n",
              "    <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-000f79d5-ddc9-4a5e-aa59-c239a00f46bc')\"\n",
              "            title=\"Convert this dataframe to an interactive table.\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\" viewBox=\"0 -960 960 960\">\n",
              "    <path d=\"M120-120v-720h720v720H120Zm60-500h600v-160H180v160Zm220 220h160v-160H400v160Zm0 220h160v-160H400v160ZM180-400h160v-160H180v160Zm440 0h160v-160H620v160ZM180-180h160v-160H180v160Zm440 0h160v-160H620v160Z\"/>\n",
              "  </svg>\n",
              "    </button>\n",
              "\n",
              "  <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    .colab-df-buttons div {\n",
              "      margin-bottom: 4px;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "    <script>\n",
              "      const buttonEl =\n",
              "        document.querySelector('#df-000f79d5-ddc9-4a5e-aa59-c239a00f46bc button.colab-df-convert');\n",
              "      buttonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "      async function convertToInteractive(key) {\n",
              "        const element = document.querySelector('#df-000f79d5-ddc9-4a5e-aa59-c239a00f46bc');\n",
              "        const dataTable =\n",
              "          await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                    [key], {});\n",
              "        if (!dataTable) return;\n",
              "\n",
              "        const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "          '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "          + ' to learn more about interactive tables.';\n",
              "        element.innerHTML = '';\n",
              "        dataTable['output_type'] = 'display_data';\n",
              "        await google.colab.output.renderOutput(dataTable, element);\n",
              "        const docLink = document.createElement('div');\n",
              "        docLink.innerHTML = docLinkHtml;\n",
              "        element.appendChild(docLink);\n",
              "      }\n",
              "    </script>\n",
              "  </div>\n",
              "\n",
              "\n",
              "<div id=\"df-95cb76ed-6c5c-4507-bc01-b4d834620f14\">\n",
              "  <button class=\"colab-df-quickchart\" onclick=\"quickchart('df-95cb76ed-6c5c-4507-bc01-b4d834620f14')\"\n",
              "            title=\"Suggest charts\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "<svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "     width=\"24px\">\n",
              "    <g>\n",
              "        <path d=\"M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zM9 17H7v-7h2v7zm4 0h-2V7h2v10zm4 0h-2v-4h2v4z\"/>\n",
              "    </g>\n",
              "</svg>\n",
              "  </button>\n",
              "\n",
              "<style>\n",
              "  .colab-df-quickchart {\n",
              "      --bg-color: #E8F0FE;\n",
              "      --fill-color: #1967D2;\n",
              "      --hover-bg-color: #E2EBFA;\n",
              "      --hover-fill-color: #174EA6;\n",
              "      --disabled-fill-color: #AAA;\n",
              "      --disabled-bg-color: #DDD;\n",
              "  }\n",
              "\n",
              "  [theme=dark] .colab-df-quickchart {\n",
              "      --bg-color: #3B4455;\n",
              "      --fill-color: #D2E3FC;\n",
              "      --hover-bg-color: #434B5C;\n",
              "      --hover-fill-color: #FFFFFF;\n",
              "      --disabled-bg-color: #3B4455;\n",
              "      --disabled-fill-color: #666;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart {\n",
              "    background-color: var(--bg-color);\n",
              "    border: none;\n",
              "    border-radius: 50%;\n",
              "    cursor: pointer;\n",
              "    display: none;\n",
              "    fill: var(--fill-color);\n",
              "    height: 32px;\n",
              "    padding: 0;\n",
              "    width: 32px;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart:hover {\n",
              "    background-color: var(--hover-bg-color);\n",
              "    box-shadow: 0 1px 2px rgba(60, 64, 67, 0.3), 0 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "    fill: var(--button-hover-fill-color);\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart-complete:disabled,\n",
              "  .colab-df-quickchart-complete:disabled:hover {\n",
              "    background-color: var(--disabled-bg-color);\n",
              "    fill: var(--disabled-fill-color);\n",
              "    box-shadow: none;\n",
              "  }\n",
              "\n",
              "  .colab-df-spinner {\n",
              "    border: 2px solid var(--fill-color);\n",
              "    border-color: transparent;\n",
              "    border-bottom-color: var(--fill-color);\n",
              "    animation:\n",
              "      spin 1s steps(1) infinite;\n",
              "  }\n",
              "\n",
              "  @keyframes spin {\n",
              "    0% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "      border-left-color: var(--fill-color);\n",
              "    }\n",
              "    20% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    30% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    40% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    60% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    80% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "    90% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "  }\n",
              "</style>\n",
              "\n",
              "  <script>\n",
              "    async function quickchart(key) {\n",
              "      const quickchartButtonEl =\n",
              "        document.querySelector('#' + key + ' button');\n",
              "      quickchartButtonEl.disabled = true;  // To prevent multiple clicks.\n",
              "      quickchartButtonEl.classList.add('colab-df-spinner');\n",
              "      try {\n",
              "        const charts = await google.colab.kernel.invokeFunction(\n",
              "            'suggestCharts', [key], {});\n",
              "      } catch (error) {\n",
              "        console.error('Error during call to suggestCharts:', error);\n",
              "      }\n",
              "      quickchartButtonEl.classList.remove('colab-df-spinner');\n",
              "      quickchartButtonEl.classList.add('colab-df-quickchart-complete');\n",
              "    }\n",
              "    (() => {\n",
              "      let quickchartButtonEl =\n",
              "        document.querySelector('#df-95cb76ed-6c5c-4507-bc01-b4d834620f14 button');\n",
              "      quickchartButtonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "    })();\n",
              "  </script>\n",
              "</div>\n",
              "    </div>\n",
              "  </div>\n"
            ],
            "application/vnd.google.colaboratory.intrinsic+json": {
              "type": "dataframe",
              "variable_name": "concatenated_df",
              "summary": "{\n  \"name\": \"concatenated_df\",\n  \"rows\": 1504,\n  \"fields\": [\n    {\n      \"column\": \"Unnamed: 0\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 16382,\n        \"min\": 13,\n        \"max\": 57287,\n        \"num_unique_values\": 1504,\n        \"samples\": [\n          5751,\n          22637,\n          26381\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"Gene\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 1504,\n        \"samples\": [\n          \"ndufa12\",\n          \"sgpl1\",\n          \"cacna1e\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"Disease\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 1504,\n        \"samples\": [\n          \"OMIM:618244\",\n          \"ORPHA:506334\",\n          \"OMIM:618285\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"disease_name\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 1504,\n        \"samples\": [\n          \"Mitochondrial complex I deficiency, nuclear type 23, 618244 (3)\",\n          \"Orphanet: Familial steroid-resistant nephrotic syndrome with adrenal insufficiency\",\n          \"Developmental and epileptic encephalopathy 69, 618285 (3)\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"Syn\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 1251,\n        \"samples\": [\n          \"SCID due to LAT deficiency\",\n          \"PLAAND\",\n          \"Inherited GS deficiency, Inherited glutamine synthetase deficiency\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"disease_des/def\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 1276,\n        \"samples\": [\n          \"Neurodevelopmental disorder with hyperkinetic movements, seizures, and structural brain abnormalities (NEDMSB) is a severe autosomal recessive disorder characterized by failure to thrive in infancy, global developmental delay, hypotonia, motor abnormalities with inability to walk, involuntary movements, impaired intellectual development, absent speech, seizures, and structural brain abnormalities ({1:Alkhater et al., 2018}; {2:Dafsari et al., 2022}).\",\n          \"Auriculocondylar syndrome-4 (ARCND4) is characterized by malformed ears, round face, puffy cheeks, micrognathia, microstomia, malocclusion, and abnormal mandibular condyles with temporomandibular joint abnormalities. Patients may also experience conductive hearing loss ({1:Masotti et al., 2008}).\\n\\nFor a general phenotypic description and a discussion of genetic heterogeneity of auriculocondylar syndrome, see ARCND1 ({602483}).\",\n          \"Folate-responsive megaloblastic anemia (MEGAF) is an autosomal recessive metabolic disorder characterized by megaloblastic anemia resulting from decreased folate transport into erythrocytes. Although serum levels of folate are normal, there is folate deficiency in tissues, including erythrocytes and possibly nerve cells. Serum homocysteine levels are increased and vitamin B12 levels may be decreased. Treatment with oral folate corrects the anemia and normalizes homocysteine (summary by {2:Svaton et al., 2020})\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"Clinicalfeatures/Clinical_des\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 740,\n        \"samples\": [\n          \"{1:Isidor et al. (2013)} reported 2 unrelated 5-year-old boys with a severe form of spondyloepimetaphyseal dysplasia. Both had normal birth length, early postnatal growth deficiency, severe short stature, flexion contractures in the hips, and bowing of the legs with genu varum. Skeletal x-rays showed platyspondyly and a characteristic vertebral shape with central indentation of endplates; progressive and severe metaphyseal changes; very small and irregular proximal femoral epiphyses with severe coxa vara; absence of calcifications; and mild metaphyseal irregularities in the upper limbs. Bone age was delayed in both. The authors suggested that the 2 patients represented a new form of SEMD.\",\n          \"Presentation is typically with neonatal/infantile hypotonia and feeding difficulties (affecting more than 70%). KS associates developmental delay and intellectual disability in 90% of patients. Intellectual disability is frequently mild to moderate; however, the spectrum ranges from normal intellect to severe disability. Verbal language acquisition and memory are usually better than visuospatial or processing speed. Individuals with KS usually achieve walking and language milestones. Heart malformations are observed in around 50% of patients. Kidney malformations, deafness, and seizures of different types are observed in 20%. Immune deficiency and recurrent infection are frequent. Autoimmune features increase with age affecting around 20% of the adults with KS. Scoliosis and patellar luxation require frequent screening and should be monitored carefully at puberty. Missing teeth are frequently observed. Obesity could arise from the age of 5 and requires monitoring. Growth hormone deficiency is observed in 25-30%. Premature thelarche occurs in 30% of females.\",\n          \"In 90% of cases the neuroblastoma is diagnosed before the age of five. The clinical presentation of neuroblastoma is very variable and depends on the stage and location of the tumor, which can develop at any site in the sympathetic nervous system (around 80% of cases develop in the abdomen). Localized forms are discovered fortuitously or are revealed by the presence of an abdominal or thoracic mass that can be associated with pain. At the time of diagnosis, metastatic forms represent about 50% of cases. The most frequent metastatic sites are bone marrow, bone, liver and skin. Symptoms of metastasis including bone pain, limp, paralysis, hepatomegaly (Pepper's syndrome), and exophthalmia (Hutchinson's syndrome), indicate metastatic neuroblastoma. The disease can also be associated with arterial hypertension, fever, and an altered general state (weight loss, pain, irritability, and anemia).\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    }\n  ]\n}"
            }
          },
          "metadata": {},
          "execution_count": 135
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!pip install transformers sentence-transformers datasets\n",
        "from datasets import load_dataset\n",
        "from sentence_transformers import SentenceTransformer, models\n",
        "from transformers import BertTokenizer\n",
        "from transformers import get_linear_schedule_with_warmup\n",
        "import torch\n",
        "from torch.optim import AdamW\n",
        "from torch.utils.data import DataLoader\n",
        "from tqdm import tqdm\n",
        "import time\n",
        "import datetime\n",
        "import random\n",
        "import numpy as np\n",
        "import pandas as pd\n",
        "from transformers import AutoTokenizer, AutoModel\n",
        "from transformers import BertTokenizer, RobertaTokenizer, RobertaModel\n",
        "tokenizer = AutoTokenizer.from_pretrained('microsoft/BiomedNLP-PubMedBERT-base-uncased-abstract-fulltext')\n",
        "class BertForSTS(torch.nn.Module):\n",
        "    def __init__(self):\n",
        "        super(BertForSTS, self).__init__()\n",
        "        #self.bert = models.Transformer(\"bert-base-uncased\", max_seq_length=512)\n",
        "        self.bert = models.Transformer('microsoft/BiomedNLP-PubMedBERT-base-uncased-abstract-fulltext', max_seq_length=1000)\n",
        "        #model = AutoModel.from_pretrained(model_name)\n",
        "        self.pooling_layer = models.Pooling(self.bert.get_word_embedding_dimension())\n",
        "        self.sts_bert = SentenceTransformer(modules=[self.bert, self.pooling_layer])\n",
        "    def forward(self, input_data):\n",
        "        output = self.sts_bert(input_data)['sentence_embedding']\n",
        "        return output\n",
        "# Instantiate the model and move it to GPU\n",
        "if torch.cuda.is_available():\n",
        "    device = torch.device(\"cuda\")\n",
        "    print(f'There are {torch.cuda.device_count()} GPU(s) available.')\n",
        "    print('We will use the GPU:', torch.cuda.get_device_name(0))\n",
        "else:\n",
        "    print('No GPU available, using the CPU instead.')\n",
        "    device = torch.device(\"cpu\")\n",
        "\n",
        "model = BertForSTS()\n",
        "model.to(device)\n",
        "model.eval()\n",
        "def predict_similarity(sentence_pair):\n",
        "  test_input = tokenizer(sentence_pair, padding=True, truncation=True, return_tensors=\"pt\").to(device) # max_length = 512,\n",
        "  test_input['input_ids'] = test_input['input_ids']\n",
        "  test_input['attention_mask'] = test_input['attention_mask']\n",
        "  del test_input['token_type_ids']\n",
        "  output = model(test_input)\n",
        "  sim = torch.nn.functional.cosine_similarity(output[0], output[1], dim=0).item()\n",
        "  return sim\n",
        "\n",
        "from google.colab import drive\n",
        "drive.mount('/content/drive')\n",
        "PATH = '/content/drive/My Drive/Finetunned_Bert_2.pt'\n",
        "model = BertForSTS()\n",
        "device = torch.device('cuda' if torch.cuda.is_available() else 'cpu')\n",
        "model.load_state_dict(torch.load(PATH, map_location=device))\n",
        "model.eval()\n",
        "\n",
        "print(\"##########################################################################################\")\n",
        "example_1=[ 'A cat is walking around a house.', 'A woman is peeling potato']\n",
        "print(f\"Sentence 1: {example_1[0]}\")\n",
        "print(f\"Sentence 2: {example_1[1]}\")\n",
        "print(f\"Predicted similarity score: {round(predict_similarity(example_1), 2)}\")\n",
        "\n",
        "print(\"##########################################################################################\")\n",
        "example_1=[  'arterial thoracic outlet syndrome',\n",
        " '''\n",
        " The objective of this project is to study the contribution of hemorrheology to arterial and venous thrombosis using the arterio-venous (A-V)\n",
        "   ''']\n",
        "print(f\"Sentence 1: {example_1[0]}\")\n",
        "print(f\"Sentence 2: {example_1[1]}\")\n",
        "print(f\"Predicted similarity score: {round(predict_similarity(example_1), 2)}\")\n",
        "\n",
        "\n",
        "print(\"##########################################################################################\")\n",
        "example_1=['antisynthetase syndrome',\n",
        " '''\n",
        "  A rare idiopathic inflammatory myopathy (IIM) historically characterized by symmetric proximal muscle weakness, elevated muscle enzymes (creatine kinase), myopathic findings on electromyography, and muscle biopsy showing endomyial infiltration composed mainly of macrophages and lymphocytes. The features are non-specific, thus the disease should be distinguished from similar entities with specific clinical, immunological, histological features, notably dermatomyositis, immune-mediated necrotizing myopathy, anti-synthetase syndrome, inclusion body myositis, and myositis associated with other connective tissue disorder.\n",
        "''']\n",
        "print(f\"Sentence 1: {example_1[0]}\")\n",
        "print(f\"Sentence 2: {example_1[1]}\")\n",
        "print(f\"Predicted similarity score: {round(predict_similarity(example_1), 2)}\")\n",
        "\n",
        "print(\"##########################################################################################\")\n",
        "##(one of its synonym is childhood cancer)\n",
        "example_1=['mismatch repair cancer syndrome 1 '\n",
        ",\n",
        "'''\n",
        "childhood cancer''']\n",
        "\n",
        "print(f\"Sentence 1: {example_1[0]}\")\n",
        "print(f\"Sentence 2: {example_1[1]}\")\n",
        "print(f\"Predicted similarity score: {round(predict_similarity(example_1), 2)}\")\n",
        "\n",
        "print(\"##########################################################################################\")\n",
        "example_1=[  'polymyositis',\n",
        " '''\n",
        " A rare idiopathic inflammatory myopathy (IIM) historically characterized by symmetric proximal muscle weakness, elevated muscle enzymes (creatine kinase), myopathic findings on electromyography, and muscle biopsy showing endomyial infiltration composed mainly of macrophages and lymphocytes. The features are non-specific, thus the disease should be distinguished from similar entities with specific clinical, immunological, histological features, notably dermatomyositis, immune-mediated necrotizing myopathy, anti-synthetase syndrome, inclusion body myositis, and myositis associated with other connective tissue disorder.\n",
        "   '''\n",
        "]\n",
        "print(f\"Sentence 1: {example_1[0]}\")\n",
        "print(f\"Sentence 2: {example_1[1]}\")\n",
        "print(f\"Predicted similarity score: {round(predict_similarity(example_1), 2)}\")\n",
        "\n",
        "def is_about_term(input_text, target_term):\n",
        "  sentence_pair=[input_text, target_term]\n",
        "  test_input = tokenizer(sentence_pair, padding=True, truncation=True, return_tensors=\"pt\").to(device) # max_length = 512,\n",
        "  test_input['input_ids'] = test_input['input_ids']\n",
        "  test_input['attention_mask'] = test_input['attention_mask']\n",
        "  del test_input['token_type_ids']\n",
        "  output = model(test_input)\n",
        "  sim = torch.nn.functional.cosine_similarity(output[0], output[1], dim=0).item()\n",
        "  return  round(sim,2)"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 1000,
          "referenced_widgets": [
            "e4568861fd194aeca79336631cf3e0b7",
            "5a435c6667b840059712262571ee1455",
            "e7dc37aeb264456ca3437770c196c1cc",
            "c5e156a53f384cc0adbf9276e4413e91",
            "5432ef7b6dc146beac0d889dbf5acb00",
            "1c7550edd596414291dafc0a56b58f84",
            "4f11907b4dab49dfab8783c495eb1dce",
            "3fe4167a91654d82985c693744c898b2",
            "42dc65acd9db439aa93a94c28e83d96a",
            "bb8c4d702cfa488fa394f5843de212ae",
            "11265d47ffdc44089b4e12d0d22d1c67",
            "a29233c3ed754daeacfbd69e26dd45e5",
            "8a3c303fd47745338c08050ac2a1a2bc",
            "5fb65a1046e44b7f901ddf26b5721f15",
            "dd3e556132a84472aa0792e048401a05",
            "a8f6235f61a4448dba4fbe4ce13c6fcd",
            "841cfd9e755a447a830dfa732ded6099",
            "ddfc3ab031e64c60a058bc4159d1816c",
            "40a8080b2c0f48edb2e52048a82b8671",
            "dc9a898f9e324ca0a630d03f53595190",
            "326074588d2d4e6c887dcde7c53fa0e3",
            "eee90fe279ec445090f12165aea2a25e",
            "f044c3795a02452cba3e1ab8f06ed5cc",
            "5d35f048eaba4fa795948c9fecdd4c47",
            "64257029ced24515ac59f615be545cc8",
            "193c62a3424149a6a4bbb331f0c121e0",
            "484c8f5f4b26451c81ffb6e7556fce96",
            "76a89adbe6054974bd2ef1c474621405",
            "1c1438ed93904d609929ac63f00d426f",
            "ea2fe30719c8434cb6708e351230ddec",
            "3b439c4c41ad462ca6e56a9428c2dde0",
            "52faa15345de4a17a1ad9e14f02b0997",
            "eb5499b9c147408b88e4de3acca1d2ca",
            "e5ec05d6cd25439f8daa14e0a69b67b4",
            "64147eb6c2604e0a815be7797f85ac18",
            "eb3467d485414a91b3ff855e65675f2d",
            "349ee2cb65784281a2c03fec6a4730e0",
            "259d4c7dde8c4ffaad619829e3cc1e41",
            "9799b290929d416ea33f3b383c1587a8",
            "34fba007363a400288303b059b0b75a6",
            "0ea0feb491264a689ae1c2996713031e",
            "32ee7d7f0685492fa78ab09bf72f82de",
            "84751c6490714f2fbb60224b50a2167e",
            "50c516e5e3f14590acc32ed18e98f7c0"
          ]
        },
        "id": "U8X8HTveDv_p",
        "outputId": "33b2ced3-38ce-4841-da03-96821f4927f9"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Requirement already satisfied: transformers in /usr/local/lib/python3.10/dist-packages (4.40.2)\n",
            "Collecting sentence-transformers\n",
            "  Downloading sentence_transformers-2.7.0-py3-none-any.whl (171 kB)\n",
            "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m171.5/171.5 kB\u001b[0m \u001b[31m4.5 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hCollecting datasets\n",
            "  Downloading datasets-2.19.1-py3-none-any.whl (542 kB)\n",
            "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m542.0/542.0 kB\u001b[0m \u001b[31m12.4 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hRequirement already satisfied: filelock in /usr/local/lib/python3.10/dist-packages (from transformers) (3.14.0)\n",
            "Requirement already satisfied: huggingface-hub<1.0,>=0.19.3 in /usr/local/lib/python3.10/dist-packages (from transformers) (0.20.3)\n",
            "Requirement already satisfied: numpy>=1.17 in /usr/local/lib/python3.10/dist-packages (from transformers) (1.25.2)\n",
            "Requirement already satisfied: packaging>=20.0 in /usr/local/lib/python3.10/dist-packages (from transformers) (24.0)\n",
            "Requirement already satisfied: pyyaml>=5.1 in /usr/local/lib/python3.10/dist-packages (from transformers) (6.0.1)\n",
            "Requirement already satisfied: regex!=2019.12.17 in /usr/local/lib/python3.10/dist-packages (from transformers) (2023.12.25)\n",
            "Requirement already satisfied: requests in /usr/local/lib/python3.10/dist-packages (from transformers) (2.31.0)\n",
            "Requirement already satisfied: tokenizers<0.20,>=0.19 in /usr/local/lib/python3.10/dist-packages (from transformers) (0.19.1)\n",
            "Requirement already satisfied: safetensors>=0.4.1 in /usr/local/lib/python3.10/dist-packages (from transformers) (0.4.3)\n",
            "Requirement already satisfied: tqdm>=4.27 in /usr/local/lib/python3.10/dist-packages (from transformers) (4.66.4)\n",
            "Requirement already satisfied: torch>=1.11.0 in /usr/local/lib/python3.10/dist-packages (from sentence-transformers) (2.2.1+cu121)\n",
            "Requirement already satisfied: scikit-learn in /usr/local/lib/python3.10/dist-packages (from sentence-transformers) (1.2.2)\n",
            "Requirement already satisfied: scipy in /usr/local/lib/python3.10/dist-packages (from sentence-transformers) (1.11.4)\n",
            "Requirement already satisfied: Pillow in /usr/local/lib/python3.10/dist-packages (from sentence-transformers) (9.4.0)\n",
            "Requirement already satisfied: pyarrow>=12.0.0 in /usr/local/lib/python3.10/dist-packages (from datasets) (14.0.2)\n",
            "Requirement already satisfied: pyarrow-hotfix in /usr/local/lib/python3.10/dist-packages (from datasets) (0.6)\n",
            "Collecting dill<0.3.9,>=0.3.0 (from datasets)\n",
            "  Downloading dill-0.3.8-py3-none-any.whl (116 kB)\n",
            "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m116.3/116.3 kB\u001b[0m \u001b[31m8.0 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hRequirement already satisfied: pandas in /usr/local/lib/python3.10/dist-packages (from datasets) (2.0.3)\n",
            "Collecting xxhash (from datasets)\n",
            "  Downloading xxhash-3.4.1-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (194 kB)\n",
            "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m194.1/194.1 kB\u001b[0m \u001b[31m10.2 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hCollecting multiprocess (from datasets)\n",
            "  Downloading multiprocess-0.70.16-py310-none-any.whl (134 kB)\n",
            "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m134.8/134.8 kB\u001b[0m \u001b[31m10.7 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hRequirement already satisfied: fsspec[http]<=2024.3.1,>=2023.1.0 in /usr/local/lib/python3.10/dist-packages (from datasets) (2023.6.0)\n",
            "Requirement already satisfied: aiohttp in /usr/local/lib/python3.10/dist-packages (from datasets) (3.9.5)\n",
            "Collecting huggingface-hub<1.0,>=0.19.3 (from transformers)\n",
            "  Downloading huggingface_hub-0.23.0-py3-none-any.whl (401 kB)\n",
            "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m401.2/401.2 kB\u001b[0m \u001b[31m15.7 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hRequirement already satisfied: aiosignal>=1.1.2 in /usr/local/lib/python3.10/dist-packages (from aiohttp->datasets) (1.3.1)\n",
            "Requirement already satisfied: attrs>=17.3.0 in /usr/local/lib/python3.10/dist-packages (from aiohttp->datasets) (23.2.0)\n",
            "Requirement already satisfied: frozenlist>=1.1.1 in /usr/local/lib/python3.10/dist-packages (from aiohttp->datasets) (1.4.1)\n",
            "Requirement already satisfied: multidict<7.0,>=4.5 in /usr/local/lib/python3.10/dist-packages (from aiohttp->datasets) (6.0.5)\n",
            "Requirement already satisfied: yarl<2.0,>=1.0 in /usr/local/lib/python3.10/dist-packages (from aiohttp->datasets) (1.9.4)\n",
            "Requirement already satisfied: async-timeout<5.0,>=4.0 in /usr/local/lib/python3.10/dist-packages (from aiohttp->datasets) (4.0.3)\n",
            "Requirement already satisfied: typing-extensions>=3.7.4.3 in /usr/local/lib/python3.10/dist-packages (from huggingface-hub<1.0,>=0.19.3->transformers) (4.11.0)\n",
            "Requirement already satisfied: charset-normalizer<4,>=2 in /usr/local/lib/python3.10/dist-packages (from requests->transformers) (3.3.2)\n",
            "Requirement already satisfied: idna<4,>=2.5 in /usr/local/lib/python3.10/dist-packages (from requests->transformers) (3.7)\n",
            "Requirement already satisfied: urllib3<3,>=1.21.1 in /usr/local/lib/python3.10/dist-packages (from requests->transformers) (2.0.7)\n",
            "Requirement already satisfied: certifi>=2017.4.17 in /usr/local/lib/python3.10/dist-packages (from requests->transformers) (2024.2.2)\n",
            "Requirement already satisfied: sympy in /usr/local/lib/python3.10/dist-packages (from torch>=1.11.0->sentence-transformers) (1.12)\n",
            "Requirement already satisfied: networkx in /usr/local/lib/python3.10/dist-packages (from torch>=1.11.0->sentence-transformers) (3.3)\n",
            "Requirement already satisfied: jinja2 in /usr/local/lib/python3.10/dist-packages (from torch>=1.11.0->sentence-transformers) (3.1.4)\n",
            "Collecting nvidia-cuda-nvrtc-cu12==12.1.105 (from torch>=1.11.0->sentence-transformers)\n",
            "  Using cached nvidia_cuda_nvrtc_cu12-12.1.105-py3-none-manylinux1_x86_64.whl (23.7 MB)\n",
            "Collecting nvidia-cuda-runtime-cu12==12.1.105 (from torch>=1.11.0->sentence-transformers)\n",
            "  Using cached nvidia_cuda_runtime_cu12-12.1.105-py3-none-manylinux1_x86_64.whl (823 kB)\n",
            "Collecting nvidia-cuda-cupti-cu12==12.1.105 (from torch>=1.11.0->sentence-transformers)\n",
            "  Using cached nvidia_cuda_cupti_cu12-12.1.105-py3-none-manylinux1_x86_64.whl (14.1 MB)\n",
            "Collecting nvidia-cudnn-cu12==8.9.2.26 (from torch>=1.11.0->sentence-transformers)\n",
            "  Using cached nvidia_cudnn_cu12-8.9.2.26-py3-none-manylinux1_x86_64.whl (731.7 MB)\n",
            "Collecting nvidia-cublas-cu12==12.1.3.1 (from torch>=1.11.0->sentence-transformers)\n",
            "  Using cached nvidia_cublas_cu12-12.1.3.1-py3-none-manylinux1_x86_64.whl (410.6 MB)\n",
            "Collecting nvidia-cufft-cu12==11.0.2.54 (from torch>=1.11.0->sentence-transformers)\n",
            "  Using cached nvidia_cufft_cu12-11.0.2.54-py3-none-manylinux1_x86_64.whl (121.6 MB)\n",
            "Collecting nvidia-curand-cu12==10.3.2.106 (from torch>=1.11.0->sentence-transformers)\n",
            "  Using cached nvidia_curand_cu12-10.3.2.106-py3-none-manylinux1_x86_64.whl (56.5 MB)\n",
            "Collecting nvidia-cusolver-cu12==11.4.5.107 (from torch>=1.11.0->sentence-transformers)\n",
            "  Using cached nvidia_cusolver_cu12-11.4.5.107-py3-none-manylinux1_x86_64.whl (124.2 MB)\n",
            "Collecting nvidia-cusparse-cu12==12.1.0.106 (from torch>=1.11.0->sentence-transformers)\n",
            "  Using cached nvidia_cusparse_cu12-12.1.0.106-py3-none-manylinux1_x86_64.whl (196.0 MB)\n",
            "Collecting nvidia-nccl-cu12==2.19.3 (from torch>=1.11.0->sentence-transformers)\n",
            "  Using cached nvidia_nccl_cu12-2.19.3-py3-none-manylinux1_x86_64.whl (166.0 MB)\n",
            "Collecting nvidia-nvtx-cu12==12.1.105 (from torch>=1.11.0->sentence-transformers)\n",
            "  Using cached nvidia_nvtx_cu12-12.1.105-py3-none-manylinux1_x86_64.whl (99 kB)\n",
            "Requirement already satisfied: triton==2.2.0 in /usr/local/lib/python3.10/dist-packages (from torch>=1.11.0->sentence-transformers) (2.2.0)\n",
            "Collecting nvidia-nvjitlink-cu12 (from nvidia-cusolver-cu12==11.4.5.107->torch>=1.11.0->sentence-transformers)\n",
            "  Using cached nvidia_nvjitlink_cu12-12.4.127-py3-none-manylinux2014_x86_64.whl (21.1 MB)\n",
            "Requirement already satisfied: python-dateutil>=2.8.2 in /usr/local/lib/python3.10/dist-packages (from pandas->datasets) (2.8.2)\n",
            "Requirement already satisfied: pytz>=2020.1 in /usr/local/lib/python3.10/dist-packages (from pandas->datasets) (2023.4)\n",
            "Requirement already satisfied: tzdata>=2022.1 in /usr/local/lib/python3.10/dist-packages (from pandas->datasets) (2024.1)\n",
            "Requirement already satisfied: joblib>=1.1.1 in /usr/local/lib/python3.10/dist-packages (from scikit-learn->sentence-transformers) (1.4.2)\n",
            "Requirement already satisfied: threadpoolctl>=2.0.0 in /usr/local/lib/python3.10/dist-packages (from scikit-learn->sentence-transformers) (3.5.0)\n",
            "Requirement already satisfied: six>=1.5 in /usr/local/lib/python3.10/dist-packages (from python-dateutil>=2.8.2->pandas->datasets) (1.16.0)\n",
            "Requirement already satisfied: MarkupSafe>=2.0 in /usr/local/lib/python3.10/dist-packages (from jinja2->torch>=1.11.0->sentence-transformers) (2.1.5)\n",
            "Requirement already satisfied: mpmath>=0.19 in /usr/local/lib/python3.10/dist-packages (from sympy->torch>=1.11.0->sentence-transformers) (1.3.0)\n",
            "Installing collected packages: xxhash, nvidia-nvtx-cu12, nvidia-nvjitlink-cu12, nvidia-nccl-cu12, nvidia-curand-cu12, nvidia-cufft-cu12, nvidia-cuda-runtime-cu12, nvidia-cuda-nvrtc-cu12, nvidia-cuda-cupti-cu12, nvidia-cublas-cu12, dill, nvidia-cusparse-cu12, nvidia-cudnn-cu12, multiprocess, huggingface-hub, nvidia-cusolver-cu12, datasets, sentence-transformers\n",
            "  Attempting uninstall: huggingface-hub\n",
            "    Found existing installation: huggingface-hub 0.20.3\n",
            "    Uninstalling huggingface-hub-0.20.3:\n",
            "      Successfully uninstalled huggingface-hub-0.20.3\n",
            "Successfully installed datasets-2.19.1 dill-0.3.8 huggingface-hub-0.23.0 multiprocess-0.70.16 nvidia-cublas-cu12-12.1.3.1 nvidia-cuda-cupti-cu12-12.1.105 nvidia-cuda-nvrtc-cu12-12.1.105 nvidia-cuda-runtime-cu12-12.1.105 nvidia-cudnn-cu12-8.9.2.26 nvidia-cufft-cu12-11.0.2.54 nvidia-curand-cu12-10.3.2.106 nvidia-cusolver-cu12-11.4.5.107 nvidia-cusparse-cu12-12.1.0.106 nvidia-nccl-cu12-2.19.3 nvidia-nvjitlink-cu12-12.4.127 nvidia-nvtx-cu12-12.1.105 sentence-transformers-2.7.0 xxhash-3.4.1\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "/usr/local/lib/python3.10/dist-packages/huggingface_hub/file_download.py:1132: FutureWarning: `resume_download` is deprecated and will be removed in version 1.0.0. Downloads always resume when possible. If you want to force a new download, use `force_download=True`.\n",
            "  warnings.warn(\n",
            "/usr/local/lib/python3.10/dist-packages/huggingface_hub/utils/_token.py:89: UserWarning: \n",
            "The secret `HF_TOKEN` does not exist in your Colab secrets.\n",
            "To authenticate with the Hugging Face Hub, create a token in your settings tab (https://huggingface.co/settings/tokens), set it as secret in your Google Colab and restart your session.\n",
            "You will be able to reuse this secret in all of your notebooks.\n",
            "Please note that authentication is recommended but still optional to access public models or datasets.\n",
            "  warnings.warn(\n"
          ]
        },
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "tokenizer_config.json:   0%|          | 0.00/28.0 [00:00<?, ?B/s]"
            ],
            "application/vnd.jupyter.widget-view+json": {
              "version_major": 2,
              "version_minor": 0,
              "model_id": "e4568861fd194aeca79336631cf3e0b7"
            }
          },
          "metadata": {}
        },
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "config.json:   0%|          | 0.00/385 [00:00<?, ?B/s]"
            ],
            "application/vnd.jupyter.widget-view+json": {
              "version_major": 2,
              "version_minor": 0,
              "model_id": "a29233c3ed754daeacfbd69e26dd45e5"
            }
          },
          "metadata": {}
        },
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "vocab.txt:   0%|          | 0.00/226k [00:00<?, ?B/s]"
            ],
            "application/vnd.jupyter.widget-view+json": {
              "version_major": 2,
              "version_minor": 0,
              "model_id": "f044c3795a02452cba3e1ab8f06ed5cc"
            }
          },
          "metadata": {}
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "No GPU available, using the CPU instead.\n"
          ]
        },
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "pytorch_model.bin:   0%|          | 0.00/440M [00:00<?, ?B/s]"
            ],
            "application/vnd.jupyter.widget-view+json": {
              "version_major": 2,
              "version_minor": 0,
              "model_id": "e5ec05d6cd25439f8daa14e0a69b67b4"
            }
          },
          "metadata": {}
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Mounted at /content/drive\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "Asking to truncate to max_length but no maximum length is provided and the model has no predefined maximum length. Default to no truncation.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "##########################################################################################\n",
            "Sentence 1: A cat is walking around a house.\n",
            "Sentence 2: A woman is peeling potato\n",
            "Predicted similarity score: -0.09\n",
            "##########################################################################################\n",
            "Sentence 1: arterial thoracic outlet syndrome\n",
            "Sentence 2: \n",
            " The objective of this project is to study the contribution of hemorrheology to arterial and venous thrombosis using the arterio-venous (A-V)\n",
            "   \n",
            "Predicted similarity score: 0.35\n",
            "##########################################################################################\n",
            "Sentence 1: antisynthetase syndrome\n",
            "Sentence 2: \n",
            "  A rare idiopathic inflammatory myopathy (IIM) historically characterized by symmetric proximal muscle weakness, elevated muscle enzymes (creatine kinase), myopathic findings on electromyography, and muscle biopsy showing endomyial infiltration composed mainly of macrophages and lymphocytes. The features are non-specific, thus the disease should be distinguished from similar entities with specific clinical, immunological, histological features, notably dermatomyositis, immune-mediated necrotizing myopathy, anti-synthetase syndrome, inclusion body myositis, and myositis associated with other connective tissue disorder.\n",
            "\n",
            "Predicted similarity score: 0.45\n",
            "##########################################################################################\n",
            "Sentence 1: mismatch repair cancer syndrome 1 \n",
            "Sentence 2: \n",
            "childhood cancer\n",
            "Predicted similarity score: 0.84\n",
            "##########################################################################################\n",
            "Sentence 1: polymyositis\n",
            "Sentence 2: \n",
            " A rare idiopathic inflammatory myopathy (IIM) historically characterized by symmetric proximal muscle weakness, elevated muscle enzymes (creatine kinase), myopathic findings on electromyography, and muscle biopsy showing endomyial infiltration composed mainly of macrophages and lymphocytes. The features are non-specific, thus the disease should be distinguished from similar entities with specific clinical, immunological, histological features, notably dermatomyositis, immune-mediated necrotizing myopathy, anti-synthetase syndrome, inclusion body myositis, and myositis associated with other connective tissue disorder.\n",
            "   \n",
            "Predicted similarity score: 0.84\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "Early onset primary dystonia (DYT1 dystonia)"
      ],
      "metadata": {
        "id": "ipp2uB3KE9XB"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "Dis_desc_DYT1='''\n",
        "'Dystonia' describes a neurologic condition characterized by involuntary, sustained muscle contractions affecting one or more sites of the body; 'torsion' refers to the twisting nature of body movements observed in dystonia. Dystonia has been classified as primary (dystonia as the sole or major symptom) or secondary (a symptom of another disorder), and by age of onset, muscle groups affected, and mode of inheritance (Muller and Kupke, 1990; Nemeth, 2002).\n",
        "'''\n",
        "Dis_Clincial_DYT1='''\n",
        "Primary torsion dystonia (also known as 'idiopathic' torsion dystonia; ITD) usually begins in childhood or adolescence with involuntary posturing of the trunk, neck, or limbs (Marsden et al., 1976; Nemeth, 2002). Some patients have a myostatic picture, such as was described by Wechsler and Brock (1922). Johnson et al. (1962) described an extensively affected French-Canadian family. Age of onset of affected family members ranged from 6 to 42 years and severity of the disease varied considerably, with early-onset cases being severely affected. Minor manifestations interpreted as 'formes frustes' were found in some family members. In a large North American family of non-Jewish ancestry, Brin et al. (1989) found that age of onset ranged from 4 to 43 years (mean 14.4, median 10.0). Generalization occurred within a median time of 3 years and occurred earlier in cases with onset in the leg. One 6.5-year-old was unable to walk within 3 months of onset. Batshaw et al. (1985) described a patient with severe simulated torsion dystonia as the main feature of Munchausen syndrome.\n",
        "\n",
        "Bressman et al. (1994) analyzed the haplotype of 174 Ashkenazi Jewish individuals affected with torsion dystonia. In this group, there were 90 carriers of the haplotype and 70 noncarriers. The authors found very striking differences in the phenotype between carriers and noncarriers. The age of onset in carriers was 12.5 years versus 36.5 years in the noncarriers. In 94% of carriers, symptoms began in a limb but only rarely in the neck and larynx. In contrast, the neck, larynx, or other cranial muscles were the site of onset in 79% of noncarriers. Discriminant analysis of limb onset, leg involvement, and age at onset distinguished haplotype carriers from noncarriers with 90% accuracy. In 23 of the 70 noncarriers, the disease was familial and included brachial, cervical, laryngeal, and facial dystonia.\n",
        "\n",
        "Cheng et al. (1996) studied 49 probands with cervical or cranial dystonia with age of onset greater than 12 years and with a positive family history. They found that age of onset of clinical symptoms was earlier by an average of 21.25 years in the second generation than in the first, and suggested that an unstable trinucleotide repeat may be involved in adult-onset primary cranial or cervical dystonia.\n",
        "\n",
        "Grundmann et al. (2003) stated that most cases of early-onset generalized dystonia are caused by a 3-bp deletion in the DYT1 gene (delE302/303; 605204.0001). They reported 6 patients with dystonia caused by the 3-bp deletion who exhibited wide phenotypic variability: 2 had classic early-onset primary generalized dystonia, 2 had multifocal dystonia (1 with cranial and cervical muscle involvement), and 2 had only writer's cramp with mild progression.\n",
        "\n",
        "Kostic et al. (2006) reported a large Serbian family in which 7 members carried the delE302/303 DYT1 mutation (605204.0001). However, only 2 were affected by dystonia, indicating a penetrance of 29%. One of the affected individuals had late-onset mild torticollis, and the other had generalized jerky dystonia. In addition, 3 family members with dystonia did not carry the DYT1 mutation, indicating genetic heterogeneity or possibly a psychogenic origin. Kostic et al. (2006) commented on the phenotypic variability in this family.\n",
        "\n",
        "Zirn et al. (2008) reported an 18-year-old girl with a severe form of early-onset dystonia. She had mildly delayed early motor development with stalling of further development during the second year of life. She never learned to walk independently and became wheelchair-bound at age 5 years. At age 13, she could no longer eat or drink without assistance. At age 18, she had dysphagia, severe dysarthria, facial palsy with reduced tongue mobility, dystonic movements, multiple joint contractures, increased muscle tone, hyperreflexia, and extensor plantar responses. Cognition was normal.\n",
        "'''"
      ],
      "metadata": {
        "id": "hTaOkh1LEv6r"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "def cal_sem_def(x):\n",
        "  if x!=None:\n",
        "     try:\n",
        "      return is_about_term(x, Dis_desc_DYT1)\n",
        "     except:\n",
        "      return is_about_term(x[:1000], Dis_desc_DYT1)\n",
        "  else:\n",
        "    return None\n",
        "def cal_sem_clinical(x):\n",
        "   if x!= None:\n",
        "      try:\n",
        "         return  is_about_term(x, Dis_Clincial_DYT1)\n",
        "      except:\n",
        "         return  is_about_term(x[:1000], Dis_Clincial_DYT1)\n",
        "   else:\n",
        "     return None\n",
        "\n",
        "concatenated_df['sem_def']=concatenated_df.apply(lambda x:  cal_sem_def(x[ 'disease_des/def'])  ,axis=1)\n",
        "concatenated_df['sem_clinical']=concatenated_df.apply( lambda x:  cal_sem_def(x['Clinicalfeatures/Clinical_des'])  ,axis=1)"
      ],
      "metadata": {
        "id": "YKqCzpu-GFtU"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "concatenated_df.to_csv('concatenated_df_new.csv')"
      ],
      "metadata": {
        "id": "JOdcfaJedimo"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "concatenated_df        =  pd.read_csv('/content/concatenated_df_new.csv')\n",
        "concatenated_df['Syn'] =  concatenated_df.apply(lambda x: x['Syn'].split(',')\tif type(x['Syn'])==str else []  , axis=1)\n",
        "concatenated_df['disease_name'] =  concatenated_df.apply(lambda x: x['disease_name'].replace('?','').replace('{','').replace('}','').replace('[','').replace(']','') , axis=1)\n",
        "concatenated_df['disease_name'] =  concatenated_df.apply(lambda x: ','.join(x['disease_name'].split(',')[:-1]) if len(x['disease_name'].split(','))>1 else\tx['disease_name'] , axis=1)\n",
        "concatenated_df['disease_name+syn']= concatenated_df.apply(lambda x: [x['disease_name']]+ x['Syn'] , axis=1)\n",
        "\n",
        "concatenated_df"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 912
        },
        "id": "NLr0d7HAcgnT",
        "outputId": "75722da6-0538-44d6-b113-bfd5b6d9ef62"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "      Unnamed: 0.1  Unnamed: 0     Gene       Disease  \\\n",
              "0                0         197    kif4a   OMIM:313490   \n",
              "1                1         359   epha10   OMIM:620283   \n",
              "2                2         657  slc14a1   OMIM:111000   \n",
              "3                3         674   gabrb1   OMIM:617153   \n",
              "4                4         707      mos   OMIM:620383   \n",
              "...            ...         ...      ...           ...   \n",
              "1499          1499       56605  cyp51a1  ORPHA:521432   \n",
              "1500          1500       56616     g6pd  ORPHA:466026   \n",
              "1501          1501       56682     dhfr  ORPHA:319651   \n",
              "1502          1502       56970      gp6   ORPHA:98885   \n",
              "1503          1503       57224     fzd5   ORPHA:98942   \n",
              "\n",
              "                                           disease_name  \\\n",
              "0       Taurodontism, microdontia, and dens invaginatus   \n",
              "1                       Deafness, autosomal dominant 88   \n",
              "2                                     Blood group, Kidd   \n",
              "3         Developmental and epileptic encephalopathy 45   \n",
              "4             Oocyte/zygote/embryo maturation arrest 20   \n",
              "...                                                 ...   \n",
              "1499  Orphanet: Congenital cataract-severe neonatal ...   \n",
              "1500  Orphanet: Class I glucose-6-phosphate dehydrog...   \n",
              "1501  Orphanet: Constitutional megaloblastic anemia ...   \n",
              "1502  Orphanet: Bleeding diathesis due to glycoprote...   \n",
              "1503           Orphanet: Coloboma of choroid and retina   \n",
              "\n",
              "                                                    Syn  \\\n",
              "0                       [KIF4A,  KIF4,  XLID100,  TMDI]   \n",
              "1                                     [EPHA10,  DFNA88]   \n",
              "2                            [SLC14A1,  JK,  UTE,  UT1]   \n",
              "3                                      [GABRB1,  DEE45]   \n",
              "4                                 [MOS,  MSV,  OZEMA20]   \n",
              "...                                                 ...   \n",
              "1499                                                 []   \n",
              "1500  [Class I G6PD deficiency,  Severe hemolytic an...   \n",
              "1501  [DHFR deficiency,  Dihydrofolate reductase def...   \n",
              "1502                                                 []   \n",
              "1503                                                 []   \n",
              "\n",
              "                                        disease_des/def  \\\n",
              "0     Taurodontism, microdontia, and dens invaginatu...   \n",
              "1     Autosomal dominant deafness-88 (DFNA88) is cha...   \n",
              "2     The Kidd blood group locus encodes a urea tran...   \n",
              "3     Developmental and epileptic encephalopathy-45 ...   \n",
              "4     Oocyte/zygote/embryo maturation arrest-20 (OZE...   \n",
              "...                                                 ...   \n",
              "1499  A rare genetic disease characterized by congen...   \n",
              "1500  A rare constitutional hemolytic anemia due to ...   \n",
              "1501                                                NaN   \n",
              "1502                                                NaN   \n",
              "1503  Coloboma of choroid and retina is a rare, gene...   \n",
              "\n",
              "                          Clinicalfeatures/Clinical_des  sem_def  \\\n",
              "0     {1:Casamassimo et al. (1978)} described a fami...     0.58   \n",
              "1     {2:Jiang et al. (2011)} reported a large 4-gen...     0.76   \n",
              "2     Although Jk-null red blood cells have reduced ...     0.54   \n",
              "3     The {2:Epi4K Consortium and Epilepsy Phenome/G...     0.89   \n",
              "4     {3:Zhang et al. (2021)} reported 3 unrelated i...     0.76   \n",
              "...                                                 ...      ...   \n",
              "1499                                                NaN     0.18   \n",
              "1500                                                NaN     0.35   \n",
              "1501                                                NaN      NaN   \n",
              "1502                                                NaN      NaN   \n",
              "1503                                                NaN     0.57   \n",
              "\n",
              "      sem_clinical                                   disease_name+syn  \n",
              "0             0.66  [Taurodontism, microdontia, and dens invaginat...  \n",
              "1             0.55  [Deafness, autosomal dominant 88, EPHA10,  DFN...  \n",
              "2             0.52      [Blood group, Kidd, SLC14A1,  JK,  UTE,  UT1]  \n",
              "3             0.90  [Developmental and epileptic encephalopathy 45...  \n",
              "4             0.49  [Oocyte/zygote/embryo maturation arrest 20, MO...  \n",
              "...            ...                                                ...  \n",
              "1499           NaN  [Orphanet: Congenital cataract-severe neonatal...  \n",
              "1500           NaN  [Orphanet: Class I glucose-6-phosphate dehydro...  \n",
              "1501           NaN  [Orphanet: Constitutional megaloblastic anemia...  \n",
              "1502           NaN  [Orphanet: Bleeding diathesis due to glycoprot...  \n",
              "1503           NaN         [Orphanet: Coloboma of choroid and retina]  \n",
              "\n",
              "[1504 rows x 11 columns]"
            ],
            "text/html": [
              "\n",
              "  <div id=\"df-5e23a45f-5c48-487e-9dc3-981349352154\" class=\"colab-df-container\">\n",
              "    <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>Unnamed: 0.1</th>\n",
              "      <th>Unnamed: 0</th>\n",
              "      <th>Gene</th>\n",
              "      <th>Disease</th>\n",
              "      <th>disease_name</th>\n",
              "      <th>Syn</th>\n",
              "      <th>disease_des/def</th>\n",
              "      <th>Clinicalfeatures/Clinical_des</th>\n",
              "      <th>sem_def</th>\n",
              "      <th>sem_clinical</th>\n",
              "      <th>disease_name+syn</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>0</td>\n",
              "      <td>197</td>\n",
              "      <td>kif4a</td>\n",
              "      <td>OMIM:313490</td>\n",
              "      <td>Taurodontism, microdontia, and dens invaginatus</td>\n",
              "      <td>[KIF4A,  KIF4,  XLID100,  TMDI]</td>\n",
              "      <td>Taurodontism, microdontia, and dens invaginatu...</td>\n",
              "      <td>{1:Casamassimo et al. (1978)} described a fami...</td>\n",
              "      <td>0.58</td>\n",
              "      <td>0.66</td>\n",
              "      <td>[Taurodontism, microdontia, and dens invaginat...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>1</td>\n",
              "      <td>359</td>\n",
              "      <td>epha10</td>\n",
              "      <td>OMIM:620283</td>\n",
              "      <td>Deafness, autosomal dominant 88</td>\n",
              "      <td>[EPHA10,  DFNA88]</td>\n",
              "      <td>Autosomal dominant deafness-88 (DFNA88) is cha...</td>\n",
              "      <td>{2:Jiang et al. (2011)} reported a large 4-gen...</td>\n",
              "      <td>0.76</td>\n",
              "      <td>0.55</td>\n",
              "      <td>[Deafness, autosomal dominant 88, EPHA10,  DFN...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>2</td>\n",
              "      <td>657</td>\n",
              "      <td>slc14a1</td>\n",
              "      <td>OMIM:111000</td>\n",
              "      <td>Blood group, Kidd</td>\n",
              "      <td>[SLC14A1,  JK,  UTE,  UT1]</td>\n",
              "      <td>The Kidd blood group locus encodes a urea tran...</td>\n",
              "      <td>Although Jk-null red blood cells have reduced ...</td>\n",
              "      <td>0.54</td>\n",
              "      <td>0.52</td>\n",
              "      <td>[Blood group, Kidd, SLC14A1,  JK,  UTE,  UT1]</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>3</td>\n",
              "      <td>674</td>\n",
              "      <td>gabrb1</td>\n",
              "      <td>OMIM:617153</td>\n",
              "      <td>Developmental and epileptic encephalopathy 45</td>\n",
              "      <td>[GABRB1,  DEE45]</td>\n",
              "      <td>Developmental and epileptic encephalopathy-45 ...</td>\n",
              "      <td>The {2:Epi4K Consortium and Epilepsy Phenome/G...</td>\n",
              "      <td>0.89</td>\n",
              "      <td>0.90</td>\n",
              "      <td>[Developmental and epileptic encephalopathy 45...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>4</td>\n",
              "      <td>707</td>\n",
              "      <td>mos</td>\n",
              "      <td>OMIM:620383</td>\n",
              "      <td>Oocyte/zygote/embryo maturation arrest 20</td>\n",
              "      <td>[MOS,  MSV,  OZEMA20]</td>\n",
              "      <td>Oocyte/zygote/embryo maturation arrest-20 (OZE...</td>\n",
              "      <td>{3:Zhang et al. (2021)} reported 3 unrelated i...</td>\n",
              "      <td>0.76</td>\n",
              "      <td>0.49</td>\n",
              "      <td>[Oocyte/zygote/embryo maturation arrest 20, MO...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>...</th>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1499</th>\n",
              "      <td>1499</td>\n",
              "      <td>56605</td>\n",
              "      <td>cyp51a1</td>\n",
              "      <td>ORPHA:521432</td>\n",
              "      <td>Orphanet: Congenital cataract-severe neonatal ...</td>\n",
              "      <td>[]</td>\n",
              "      <td>A rare genetic disease characterized by congen...</td>\n",
              "      <td>NaN</td>\n",
              "      <td>0.18</td>\n",
              "      <td>NaN</td>\n",
              "      <td>[Orphanet: Congenital cataract-severe neonatal...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1500</th>\n",
              "      <td>1500</td>\n",
              "      <td>56616</td>\n",
              "      <td>g6pd</td>\n",
              "      <td>ORPHA:466026</td>\n",
              "      <td>Orphanet: Class I glucose-6-phosphate dehydrog...</td>\n",
              "      <td>[Class I G6PD deficiency,  Severe hemolytic an...</td>\n",
              "      <td>A rare constitutional hemolytic anemia due to ...</td>\n",
              "      <td>NaN</td>\n",
              "      <td>0.35</td>\n",
              "      <td>NaN</td>\n",
              "      <td>[Orphanet: Class I glucose-6-phosphate dehydro...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1501</th>\n",
              "      <td>1501</td>\n",
              "      <td>56682</td>\n",
              "      <td>dhfr</td>\n",
              "      <td>ORPHA:319651</td>\n",
              "      <td>Orphanet: Constitutional megaloblastic anemia ...</td>\n",
              "      <td>[DHFR deficiency,  Dihydrofolate reductase def...</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>[Orphanet: Constitutional megaloblastic anemia...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1502</th>\n",
              "      <td>1502</td>\n",
              "      <td>56970</td>\n",
              "      <td>gp6</td>\n",
              "      <td>ORPHA:98885</td>\n",
              "      <td>Orphanet: Bleeding diathesis due to glycoprote...</td>\n",
              "      <td>[]</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>[Orphanet: Bleeding diathesis due to glycoprot...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1503</th>\n",
              "      <td>1503</td>\n",
              "      <td>57224</td>\n",
              "      <td>fzd5</td>\n",
              "      <td>ORPHA:98942</td>\n",
              "      <td>Orphanet: Coloboma of choroid and retina</td>\n",
              "      <td>[]</td>\n",
              "      <td>Coloboma of choroid and retina is a rare, gene...</td>\n",
              "      <td>NaN</td>\n",
              "      <td>0.57</td>\n",
              "      <td>NaN</td>\n",
              "      <td>[Orphanet: Coloboma of choroid and retina]</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "<p>1504 rows × 11 columns</p>\n",
              "</div>\n",
              "    <div class=\"colab-df-buttons\">\n",
              "\n",
              "  <div class=\"colab-df-container\">\n",
              "    <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-5e23a45f-5c48-487e-9dc3-981349352154')\"\n",
              "            title=\"Convert this dataframe to an interactive table.\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\" viewBox=\"0 -960 960 960\">\n",
              "    <path d=\"M120-120v-720h720v720H120Zm60-500h600v-160H180v160Zm220 220h160v-160H400v160Zm0 220h160v-160H400v160ZM180-400h160v-160H180v160Zm440 0h160v-160H620v160ZM180-180h160v-160H180v160Zm440 0h160v-160H620v160Z\"/>\n",
              "  </svg>\n",
              "    </button>\n",
              "\n",
              "  <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    .colab-df-buttons div {\n",
              "      margin-bottom: 4px;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "    <script>\n",
              "      const buttonEl =\n",
              "        document.querySelector('#df-5e23a45f-5c48-487e-9dc3-981349352154 button.colab-df-convert');\n",
              "      buttonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "      async function convertToInteractive(key) {\n",
              "        const element = document.querySelector('#df-5e23a45f-5c48-487e-9dc3-981349352154');\n",
              "        const dataTable =\n",
              "          await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                    [key], {});\n",
              "        if (!dataTable) return;\n",
              "\n",
              "        const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "          '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "          + ' to learn more about interactive tables.';\n",
              "        element.innerHTML = '';\n",
              "        dataTable['output_type'] = 'display_data';\n",
              "        await google.colab.output.renderOutput(dataTable, element);\n",
              "        const docLink = document.createElement('div');\n",
              "        docLink.innerHTML = docLinkHtml;\n",
              "        element.appendChild(docLink);\n",
              "      }\n",
              "    </script>\n",
              "  </div>\n",
              "\n",
              "\n",
              "<div id=\"df-cd283840-bdec-48bc-9cc1-b285e626ffd7\">\n",
              "  <button class=\"colab-df-quickchart\" onclick=\"quickchart('df-cd283840-bdec-48bc-9cc1-b285e626ffd7')\"\n",
              "            title=\"Suggest charts\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "<svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "     width=\"24px\">\n",
              "    <g>\n",
              "        <path d=\"M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zM9 17H7v-7h2v7zm4 0h-2V7h2v10zm4 0h-2v-4h2v4z\"/>\n",
              "    </g>\n",
              "</svg>\n",
              "  </button>\n",
              "\n",
              "<style>\n",
              "  .colab-df-quickchart {\n",
              "      --bg-color: #E8F0FE;\n",
              "      --fill-color: #1967D2;\n",
              "      --hover-bg-color: #E2EBFA;\n",
              "      --hover-fill-color: #174EA6;\n",
              "      --disabled-fill-color: #AAA;\n",
              "      --disabled-bg-color: #DDD;\n",
              "  }\n",
              "\n",
              "  [theme=dark] .colab-df-quickchart {\n",
              "      --bg-color: #3B4455;\n",
              "      --fill-color: #D2E3FC;\n",
              "      --hover-bg-color: #434B5C;\n",
              "      --hover-fill-color: #FFFFFF;\n",
              "      --disabled-bg-color: #3B4455;\n",
              "      --disabled-fill-color: #666;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart {\n",
              "    background-color: var(--bg-color);\n",
              "    border: none;\n",
              "    border-radius: 50%;\n",
              "    cursor: pointer;\n",
              "    display: none;\n",
              "    fill: var(--fill-color);\n",
              "    height: 32px;\n",
              "    padding: 0;\n",
              "    width: 32px;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart:hover {\n",
              "    background-color: var(--hover-bg-color);\n",
              "    box-shadow: 0 1px 2px rgba(60, 64, 67, 0.3), 0 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "    fill: var(--button-hover-fill-color);\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart-complete:disabled,\n",
              "  .colab-df-quickchart-complete:disabled:hover {\n",
              "    background-color: var(--disabled-bg-color);\n",
              "    fill: var(--disabled-fill-color);\n",
              "    box-shadow: none;\n",
              "  }\n",
              "\n",
              "  .colab-df-spinner {\n",
              "    border: 2px solid var(--fill-color);\n",
              "    border-color: transparent;\n",
              "    border-bottom-color: var(--fill-color);\n",
              "    animation:\n",
              "      spin 1s steps(1) infinite;\n",
              "  }\n",
              "\n",
              "  @keyframes spin {\n",
              "    0% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "      border-left-color: var(--fill-color);\n",
              "    }\n",
              "    20% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    30% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    40% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    60% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    80% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "    90% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "  }\n",
              "</style>\n",
              "\n",
              "  <script>\n",
              "    async function quickchart(key) {\n",
              "      const quickchartButtonEl =\n",
              "        document.querySelector('#' + key + ' button');\n",
              "      quickchartButtonEl.disabled = true;  // To prevent multiple clicks.\n",
              "      quickchartButtonEl.classList.add('colab-df-spinner');\n",
              "      try {\n",
              "        const charts = await google.colab.kernel.invokeFunction(\n",
              "            'suggestCharts', [key], {});\n",
              "      } catch (error) {\n",
              "        console.error('Error during call to suggestCharts:', error);\n",
              "      }\n",
              "      quickchartButtonEl.classList.remove('colab-df-spinner');\n",
              "      quickchartButtonEl.classList.add('colab-df-quickchart-complete');\n",
              "    }\n",
              "    (() => {\n",
              "      let quickchartButtonEl =\n",
              "        document.querySelector('#df-cd283840-bdec-48bc-9cc1-b285e626ffd7 button');\n",
              "      quickchartButtonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "    })();\n",
              "  </script>\n",
              "</div>\n",
              "    </div>\n",
              "  </div>\n"
            ],
            "application/vnd.google.colaboratory.intrinsic+json": {
              "type": "dataframe",
              "variable_name": "concatenated_df",
              "summary": "{\n  \"name\": \"concatenated_df\",\n  \"rows\": 1504,\n  \"fields\": [\n    {\n      \"column\": \"Unnamed: 0.1\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 434,\n        \"min\": 0,\n        \"max\": 1503,\n        \"num_unique_values\": 1504,\n        \"samples\": [\n          51,\n          976,\n          184\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"Unnamed: 0\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 16382,\n        \"min\": 13,\n        \"max\": 57287,\n        \"num_unique_values\": 1504,\n        \"samples\": [\n          5751,\n          22637,\n          26381\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"Gene\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 1504,\n        \"samples\": [\n          \"ndufa12\",\n          \"sgpl1\",\n          \"cacna1e\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"Disease\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 1504,\n        \"samples\": [\n          \"OMIM:618244\",\n          \"ORPHA:506334\",\n          \"OMIM:618285\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"disease_name\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 1493,\n        \"samples\": [\n          \"Orphanet: Pseudohypoaldosteronism type 2C\",\n          \"Orphanet: NON RARE IN EUROPE: Non rare obesity\",\n          \"Orphanet: Autosomal recessive cutis laxa type 2\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"Syn\",\n      \"properties\": {\n        \"dtype\": \"object\",\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"disease_des/def\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 1276,\n        \"samples\": [\n          \"Neurodevelopmental disorder with hyperkinetic movements, seizures, and structural brain abnormalities (NEDMSB) is a severe autosomal recessive disorder characterized by failure to thrive in infancy, global developmental delay, hypotonia, motor abnormalities with inability to walk, involuntary movements, impaired intellectual development, absent speech, seizures, and structural brain abnormalities ({1:Alkhater et al., 2018}; {2:Dafsari et al., 2022}).\",\n          \"Auriculocondylar syndrome-4 (ARCND4) is characterized by malformed ears, round face, puffy cheeks, micrognathia, microstomia, malocclusion, and abnormal mandibular condyles with temporomandibular joint abnormalities. Patients may also experience conductive hearing loss ({1:Masotti et al., 2008}).\\n\\nFor a general phenotypic description and a discussion of genetic heterogeneity of auriculocondylar syndrome, see ARCND1 ({602483}).\",\n          \"Folate-responsive megaloblastic anemia (MEGAF) is an autosomal recessive metabolic disorder characterized by megaloblastic anemia resulting from decreased folate transport into erythrocytes. Although serum levels of folate are normal, there is folate deficiency in tissues, including erythrocytes and possibly nerve cells. Serum homocysteine levels are increased and vitamin B12 levels may be decreased. Treatment with oral folate corrects the anemia and normalizes homocysteine (summary by {2:Svaton et al., 2020})\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"Clinicalfeatures/Clinical_des\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 740,\n        \"samples\": [\n          \"{1:Isidor et al. (2013)} reported 2 unrelated 5-year-old boys with a severe form of spondyloepimetaphyseal dysplasia. Both had normal birth length, early postnatal growth deficiency, severe short stature, flexion contractures in the hips, and bowing of the legs with genu varum. Skeletal x-rays showed platyspondyly and a characteristic vertebral shape with central indentation of endplates; progressive and severe metaphyseal changes; very small and irregular proximal femoral epiphyses with severe coxa vara; absence of calcifications; and mild metaphyseal irregularities in the upper limbs. Bone age was delayed in both. The authors suggested that the 2 patients represented a new form of SEMD.\",\n          \"Presentation is typically with neonatal/infantile hypotonia and feeding difficulties (affecting more than 70%). KS associates developmental delay and intellectual disability in 90% of patients. Intellectual disability is frequently mild to moderate; however, the spectrum ranges from normal intellect to severe disability. Verbal language acquisition and memory are usually better than visuospatial or processing speed. Individuals with KS usually achieve walking and language milestones. Heart malformations are observed in around 50% of patients. Kidney malformations, deafness, and seizures of different types are observed in 20%. Immune deficiency and recurrent infection are frequent. Autoimmune features increase with age affecting around 20% of the adults with KS. Scoliosis and patellar luxation require frequent screening and should be monitored carefully at puberty. Missing teeth are frequently observed. Obesity could arise from the age of 5 and requires monitoring. Growth hormone deficiency is observed in 25-30%. Premature thelarche occurs in 30% of females.\",\n          \"In 90% of cases the neuroblastoma is diagnosed before the age of five. The clinical presentation of neuroblastoma is very variable and depends on the stage and location of the tumor, which can develop at any site in the sympathetic nervous system (around 80% of cases develop in the abdomen). Localized forms are discovered fortuitously or are revealed by the presence of an abdominal or thoracic mass that can be associated with pain. At the time of diagnosis, metastatic forms represent about 50% of cases. The most frequent metastatic sites are bone marrow, bone, liver and skin. Symptoms of metastasis including bone pain, limp, paralysis, hepatomegaly (Pepper's syndrome), and exophthalmia (Hutchinson's syndrome), indicate metastatic neuroblastoma. The disease can also be associated with arterial hypertension, fever, and an altered general state (weight loss, pain, irritability, and anemia).\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"sem_def\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 0.2522937467978647,\n        \"min\": 0.04,\n        \"max\": 0.98,\n        \"num_unique_values\": 95,\n        \"samples\": [\n          0.21,\n          0.64,\n          0.08\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"sem_clinical\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 0.2247025463130534,\n        \"min\": 0.04,\n        \"max\": 1.0,\n        \"num_unique_values\": 90,\n        \"samples\": [\n          0.98,\n          0.45,\n          0.46\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"disease_name+syn\",\n      \"properties\": {\n        \"dtype\": \"object\",\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    }\n  ]\n}"
            }
          },
          "metadata": {},
          "execution_count": 81
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "Find_dis_dict={}\n",
        "for i in concatenated_df.index:\n",
        "    for j in concatenated_df['disease_name+syn'][i]:\n",
        "      Find_dis_dict[ j.strip().lower()   ]  =   concatenated_df['Disease'][i]\n",
        "Find_dis_dict"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "1XDGCNAbg7jO",
        "outputId": "57857571-59f5-411c-e183-02be6a3a11f0"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "{'taurodontism, microdontia, and dens invaginatus': 'OMIM:313490',\n",
              " 'kif4a': 'OMIM:313490',\n",
              " 'kif4': 'OMIM:313490',\n",
              " 'xlid100': 'OMIM:313490',\n",
              " 'tmdi': 'OMIM:313490',\n",
              " 'deafness, autosomal dominant 88': 'OMIM:620283',\n",
              " 'epha10': 'OMIM:620283',\n",
              " 'dfna88': 'OMIM:620283',\n",
              " 'blood group, kidd': 'OMIM:111000',\n",
              " 'slc14a1': 'OMIM:111000',\n",
              " 'jk': 'OMIM:111000',\n",
              " 'ute': 'OMIM:111000',\n",
              " 'ut1': 'OMIM:111000',\n",
              " 'developmental and epileptic encephalopathy 45': 'OMIM:617153',\n",
              " 'gabrb1': 'OMIM:617153',\n",
              " 'dee45': 'OMIM:617153',\n",
              " 'oocyte/zygote/embryo maturation arrest 20': 'OMIM:620383',\n",
              " 'mos': 'OMIM:620383',\n",
              " 'msv': 'OMIM:620383',\n",
              " 'ozema20': 'OMIM:620383',\n",
              " 'intellectual developmental disorder, x-linked syndromic, bain type': 'OMIM:300986',\n",
              " 'hnrnph2': 'OMIM:300986',\n",
              " 'nrph2': 'OMIM:300986',\n",
              " 'mrxsb': 'OMIM:300986',\n",
              " 'cerebral infarction, susceptibility to': 'OMIM:601367',\n",
              " 'prkch': 'OMIM:601367',\n",
              " 'pkcl': 'OMIM:601367',\n",
              " 'prkcl': 'OMIM:601367',\n",
              " 'dyskeratosis congenita, autosomal recessive 8': 'OMIM:620133',\n",
              " 'dclre1b': 'OMIM:620133',\n",
              " 'snm1b': 'OMIM:620133',\n",
              " 'apollo': 'OMIM:620133',\n",
              " 'dkcb8': 'OMIM:620133',\n",
              " 'immunodeficiency 97 with autoinflammation': 'OMIM:619802',\n",
              " 'pik3cg': 'OMIM:619802',\n",
              " 'imd97': 'OMIM:619802',\n",
              " 'delayed puberty, self-limited': 'OMIM:619613',\n",
              " 'lgr4': 'OMIM:619613',\n",
              " 'gpr48': 'OMIM:619613',\n",
              " 'bnmd17': 'OMIM:619613',\n",
              " 'dpsl': 'OMIM:619613',\n",
              " 'myopia 28, autosomal recessive': 'OMIM:619781',\n",
              " 'loxl3': 'OMIM:619781',\n",
              " 'myp28': 'OMIM:619781',\n",
              " 'diabetes mellitus, insulin-dependent': 'OMIM:222100',\n",
              " 'hnf1a': 'OMIM:222100',\n",
              " 'tcf1': 'OMIM:222100',\n",
              " 'mody3': 'OMIM:222100',\n",
              " 'iddm20': 'OMIM:222100',\n",
              " 'deafness, autosomal recessive 26': 'OMIM:605428',\n",
              " 'gab1': 'OMIM:605428',\n",
              " 'dfnb26': 'OMIM:605428',\n",
              " 'intellectual developmental disorder with hypotonia and behavioral abnormalities': 'OMIM:618748',\n",
              " 'cdk8': 'OMIM:618748',\n",
              " 'iddhba': 'OMIM:618748',\n",
              " 'proteasome-associated autoinflammatory syndrome 4': 'OMIM:619183',\n",
              " 'psmg2': 'OMIM:619183',\n",
              " 'tnfsf5ip1': 'OMIM:619183',\n",
              " 'hcca3': 'OMIM:619183',\n",
              " 'clast3': 'OMIM:619183',\n",
              " 'pac2': 'OMIM:619183',\n",
              " 'praas4': 'OMIM:619183',\n",
              " 'ichthyosis, congenital, autosomal recessive 12': 'OMIM:617320',\n",
              " 'casp14': 'OMIM:617320',\n",
              " 'arci12': 'OMIM:617320',\n",
              " 'thyroid hormone metabolism, abnormal, 2': 'OMIM:619855',\n",
              " 'dio1': 'OMIM:619855',\n",
              " 'txdi1': 'OMIM:619855',\n",
              " 'thma2': 'OMIM:619855',\n",
              " 'short sleep, familial natural, 2': 'OMIM:618591',\n",
              " 'adrb1': 'OMIM:618591',\n",
              " 'adrb1r': 'OMIM:618591',\n",
              " 'rhr': 'OMIM:618591',\n",
              " 'fnss2': 'OMIM:618591',\n",
              " 'obesity, mild, early-onset': 'OMIM:601665',\n",
              " 'nr0b2': 'OMIM:601665',\n",
              " 'shp': 'OMIM:601665',\n",
              " 'attention deficit-hyperactivity disorder, susceptibility to, 7': 'OMIM:613003',\n",
              " 'tph2': 'OMIM:613003',\n",
              " 'ntph': 'OMIM:613003',\n",
              " 'adhd7': 'OMIM:613003',\n",
              " 'congenital disorder of glycosylation, type iibb': 'OMIM:620546',\n",
              " 'cog3': 'OMIM:620546',\n",
              " 'sec34': 'OMIM:620546',\n",
              " 'cdg2bb': 'OMIM:620546',\n",
              " 'intellectual developmental disorder with dysmorphic facies and ptosis': 'OMIM:617333',\n",
              " 'brpf1': 'OMIM:617333',\n",
              " 'br140': 'OMIM:617333',\n",
              " 'idddfp': 'OMIM:617333',\n",
              " 'snijders blok-fisher syndrome': 'OMIM:618604',\n",
              " 'pou3f3': 'OMIM:618604',\n",
              " 'snibfis': 'OMIM:618604',\n",
              " 'coronary heart disease, susceptibility to, 6': 'OMIM:614466',\n",
              " 'mmp3': 'OMIM:614466',\n",
              " 'stmy1': 'OMIM:614466',\n",
              " 'chds6': 'OMIM:614466',\n",
              " 'diabetes mellitus, noninsulin-dependent': 'OMIM:125853',\n",
              " 'abcc8': 'OMIM:125853',\n",
              " 'sur': 'OMIM:125853',\n",
              " 'phhi': 'OMIM:125853',\n",
              " 'sur1': 'OMIM:125853',\n",
              " 'hhf1': 'OMIM:125853',\n",
              " 'tndm2': 'OMIM:125853',\n",
              " 'pndm3': 'OMIM:125853',\n",
              " 'intellectual developmental disorder, autosomal dominant 57': 'OMIM:618050',\n",
              " 'tlk2': 'OMIM:618050',\n",
              " 'mrd57': 'OMIM:618050',\n",
              " 'colon cancer, advanced, somatic': 'OMIM:114500',\n",
              " 'src': 'OMIM:114500',\n",
              " 'asv': 'OMIM:114500',\n",
              " 'src1': 'OMIM:114500',\n",
              " 'thc6': 'OMIM:114500',\n",
              " 'myocardial infarction, decreased susceptibility to': 'OMIM:608446',\n",
              " 'f7': 'OMIM:608446',\n",
              " 'mycobacterium tuberculosis, protection against': 'OMIM:607948',\n",
              " 'irgm': 'OMIM:607948',\n",
              " 'lrg47': 'OMIM:607948',\n",
              " 'ifi1': 'OMIM:607948',\n",
              " 'ibd19': 'OMIM:607948',\n",
              " 'leukoencephalopathy, acute reversible, with increased urinary alpha-ketoglutarate': 'OMIM:618384',\n",
              " 'slc13a3': 'OMIM:618384',\n",
              " 'nadc3': 'OMIM:618384',\n",
              " 'arliak': 'OMIM:618384',\n",
              " 'oocyte/zygote/embryo maturation arrest 16': 'OMIM:617234',\n",
              " 'padi6': 'OMIM:617234',\n",
              " 'ozema16': 'OMIM:617234',\n",
              " 'pancreatic cancer': 'OMIM:260350',\n",
              " 'tp53': 'OMIM:260350',\n",
              " 'p53': 'OMIM:260350',\n",
              " 'lfs1': 'OMIM:260350',\n",
              " 'bcc7': 'OMIM:260350',\n",
              " 'bmfs5': 'OMIM:260350',\n",
              " 'acetyl-coa carboxylase deficiency': 'OMIM:613933',\n",
              " 'acaca': 'OMIM:613933',\n",
              " 'acac': 'OMIM:613933',\n",
              " 'acc1': 'OMIM:613933',\n",
              " 'acacad': 'OMIM:613933',\n",
              " 'acetylation, slow': 'OMIM:243400',\n",
              " 'nat2': 'OMIM:243400',\n",
              " 'aac2': 'OMIM:243400',\n",
              " 'encephalopathy, acute, infection-induced (herpes-specific), susceptibility to, 7': 'OMIM:616532',\n",
              " 'irf3': 'OMIM:616532',\n",
              " 'iiae7': 'OMIM:616532',\n",
              " 'fatty liver disease, protection from': 'OMIM:620116',\n",
              " 'hsd17b13': 'OMIM:620116',\n",
              " 'scdr9': 'OMIM:620116',\n",
              " 'fldp': 'OMIM:620116',\n",
              " 'acat2 deficiency': 'OMIM:614055',\n",
              " 'acat2': 'OMIM:614055',\n",
              " 'diarrhea 13': 'OMIM:620357',\n",
              " 'acsl5': 'OMIM:620357',\n",
              " 'facl5': 'OMIM:620357',\n",
              " 'acs5': 'OMIM:620357',\n",
              " 'diar13': 'OMIM:620357',\n",
              " 'leukodystrophy, hypomyelinating, 17': 'OMIM:618006',\n",
              " 'aimp2': 'OMIM:618006',\n",
              " 'jtv1': 'OMIM:618006',\n",
              " 'hld17': 'OMIM:618006',\n",
              " 'amyotrophic lateral sclerosis 22 with or without frontotemporal dementia': 'OMIM:616208',\n",
              " 'tuba4a': 'OMIM:616208',\n",
              " 'tuba1': 'OMIM:616208',\n",
              " 'als22': 'OMIM:616208',\n",
              " 'mega-corpus-callosum syndrome with cerebellar hypoplasia and cortical malformations': 'OMIM:618273',\n",
              " 'mast1': 'OMIM:618273',\n",
              " 'sast': 'OMIM:618273',\n",
              " 'kiaa0973': 'OMIM:618273',\n",
              " 'mccchcm': 'OMIM:618273',\n",
              " 'hypogonadotropic hypogonadism 7 without anosmia': 'OMIM:146110',\n",
              " 'gnrhr': 'OMIM:146110',\n",
              " 'lhrhr': 'OMIM:146110',\n",
              " 'hh7': 'OMIM:146110',\n",
              " 'primordial dwarfism-immunodeficiency-lipodystrophy syndrome': 'OMIM:620005',\n",
              " 'prim1': 'OMIM:620005',\n",
              " 'pdil': 'OMIM:620005',\n",
              " 'oocyte/zygote/embryo maturation arrest 21': 'OMIM:620610',\n",
              " 'chek1': 'OMIM:620610',\n",
              " 'chk1': 'OMIM:620610',\n",
              " 'ozema21': 'OMIM:620610',\n",
              " 'mitochondrial complex i deficiency, nuclear type 22': 'OMIM:618243',\n",
              " 'ndufa10': 'OMIM:618243',\n",
              " 'mc1dn22': 'OMIM:618243',\n",
              " 'mitochondrial complex iv deficiency, nuclear type 21': 'OMIM:619065',\n",
              " 'coxfa4': 'OMIM:619065',\n",
              " 'ndufa4': 'OMIM:619065',\n",
              " 'mc4dn21': 'OMIM:619065',\n",
              " 'mitochondrial complex i deficiency, nuclear type 39': 'OMIM:620135',\n",
              " 'ndufb7': 'OMIM:620135',\n",
              " 'mc1dn39': 'OMIM:620135',\n",
              " 'mitochondrial complex i deficiency, nuclear type 37': 'OMIM:619272',\n",
              " 'ndufa8': 'OMIM:619272',\n",
              " 'mc1dn37': 'OMIM:619272',\n",
              " 'mitochondrial complex i deficiency, nuclear type 32': 'OMIM:618252',\n",
              " 'ndufb8': 'OMIM:618252',\n",
              " 'mc1dn32': 'OMIM:618252',\n",
              " 'mitochondrial complex i deficiency, nuclear type 36': 'OMIM:619170',\n",
              " 'ndufc2': 'OMIM:619170',\n",
              " 'mc1dn36': 'OMIM:619170',\n",
              " 'mitochondrial complex i deficiency, nuclear type 26': 'OMIM:618247',\n",
              " 'ndufa9': 'OMIM:618247',\n",
              " 'mc1dn26': 'OMIM:618247',\n",
              " 'mitochondrial complex i deficiency, nuclear type 23': 'OMIM:618244',\n",
              " 'ndufa12': 'OMIM:618244',\n",
              " 'mc1dn23': 'OMIM:618244',\n",
              " 'intellectual developmental disorder with macrocephaly, seizures, and speech delay': 'OMIM:618158',\n",
              " 'pak1': 'OMIM:618158',\n",
              " 'iddmssd': 'OMIM:618158',\n",
              " 'thiopurines, poor metabolism of, 2': 'OMIM:616903',\n",
              " 'nudt15': 'OMIM:616903',\n",
              " 'mth2': 'OMIM:616903',\n",
              " 'nudt15d': 'OMIM:616903',\n",
              " 'neurodevelopmental disorder with microcephaly, hypotonia, and absent language': 'OMIM:620038',\n",
              " 'psmb1': 'OMIM:620038',\n",
              " 'nedmhal': 'OMIM:620038',\n",
              " 'obesity, susceptibility to, bmiq19': 'OMIM:617885',\n",
              " 'adcy3': 'OMIM:617885',\n",
              " 'bmiq19': 'OMIM:617885',\n",
              " 'lung cancer susceptibility 2': 'OMIM:612052',\n",
              " 'chrna3': 'OMIM:612052',\n",
              " 'lncr2': 'OMIM:612052',\n",
              " 'paod2': 'OMIM:612052',\n",
              " 'baiprck': 'OMIM:612052',\n",
              " 'skeletal muscle glycogen content and metabolism qtl': 'OMIM:619030',\n",
              " 'prkag3': 'OMIM:619030',\n",
              " 'smgmqtl': 'OMIM:619030',\n",
              " 'pontocerebellar hypoplasia, type 14': 'OMIM:619301',\n",
              " 'ppil1': 'OMIM:619301',\n",
              " 'cypl1': 'OMIM:619301',\n",
              " 'pch14': 'OMIM:619301',\n",
              " 'thiopurines, poor metabolism of, 1': 'OMIM:610460',\n",
              " 'tpmt': 'OMIM:610460',\n",
              " 'tpmtd': 'OMIM:610460',\n",
              " 'hypotonia, hypoventilation, impaired intellectual development, dysautonomia, epilepsy, and eye abnormalities': 'OMIM:618493',\n",
              " 'p4htm': 'OMIM:618493',\n",
              " 'ph4': 'OMIM:618493',\n",
              " 'hidea': 'OMIM:618493',\n",
              " 'progesterone resistance': 'OMIM:264080',\n",
              " 'pgr': 'OMIM:264080',\n",
              " 'neurodegeneration, childhood-onset, with progressive microcephaly': 'OMIM:619847',\n",
              " 'dtymk': 'OMIM:619847',\n",
              " 'tymk': 'OMIM:619847',\n",
              " 'cdc8': 'OMIM:619847',\n",
              " 'conpm': 'OMIM:619847',\n",
              " 'neurodegeneration, childhood-onset, stress-induced, with variable ataxia and seizures': 'OMIM:618170',\n",
              " 'adprhl2': 'OMIM:618170',\n",
              " 'arh3': 'OMIM:618170',\n",
              " 'condsias': 'OMIM:618170',\n",
              " 'uric acid concentration, serum, qtl1': 'OMIM:138900',\n",
              " 'abcg2': 'OMIM:138900',\n",
              " 'bcrp': 'OMIM:138900',\n",
              " 'abcp': 'OMIM:138900',\n",
              " 'uaqtl1': 'OMIM:138900',\n",
              " 'gout1': 'OMIM:138900',\n",
              " 'auriculocondylar syndrome 4': 'OMIM:620457',\n",
              " 'hdac9': 'OMIM:620457',\n",
              " 'mitr': 'OMIM:620457',\n",
              " 'hdac7b': 'OMIM:620457',\n",
              " 'kiaa0744': 'OMIM:620457',\n",
              " 'arcnd4': 'OMIM:620457',\n",
              " 'asthma susceptibility 5': 'OMIM:611064',\n",
              " 'irak3': 'OMIM:611064',\n",
              " 'irakm': 'OMIM:611064',\n",
              " 'asrt5': 'OMIM:611064',\n",
              " 'blood group, ok': 'OMIM:111380',\n",
              " 'bsg': 'OMIM:111380',\n",
              " 'immunodeficiency 60 and autoimmunity': 'OMIM:618394',\n",
              " 'bach2': 'OMIM:618394',\n",
              " 'imd60': 'OMIM:618394',\n",
              " 'macrocephaly, neurodevelopmental delay, lymphoid hyperplasia, and persistent fetal hemoglobin': 'OMIM:619769',\n",
              " 'zbtb7a': 'OMIM:619769',\n",
              " 'fbi1': 'OMIM:619769',\n",
              " 'lrf': 'OMIM:619769',\n",
              " 'mndlfh': 'OMIM:619769',\n",
              " 'dystonia 31': 'OMIM:619565',\n",
              " 'aopep': 'OMIM:619565',\n",
              " 'c9orf3': 'OMIM:619565',\n",
              " 'dyt31': 'OMIM:619565',\n",
              " 'parkinson disease 25, autosomal recessive early-onset, with impaired intellectual development': 'OMIM:620482',\n",
              " 'ptpa': 'OMIM:620482',\n",
              " 'ppp2r4': 'OMIM:620482',\n",
              " 'park25': 'OMIM:620482',\n",
              " 'microcephaly, cataracts, impaired intellectual development, and dystonia with abnormal striatum': 'OMIM:618284',\n",
              " 'kcna4': 'OMIM:618284',\n",
              " 'kcna8': 'OMIM:618284',\n",
              " 'mcidds': 'OMIM:618284',\n",
              " 'bone marrow failure syndrome 6': 'OMIM:618849',\n",
              " 'mdm4': 'OMIM:618849',\n",
              " 'bmfs6': 'OMIM:618849',\n",
              " 'migraine, with or without aura, susceptibility to, 13': 'OMIM:613656',\n",
              " 'kcnk18': 'OMIM:613656',\n",
              " 'tresk': 'OMIM:613656',\n",
              " 'trik': 'OMIM:613656',\n",
              " 'mgr13': 'OMIM:613656',\n",
              " 'pulmonary fibrosis and/or bone marrow failure syndrome, telomere-related, 6': 'OMIM:619767',\n",
              " 'rpa1': 'OMIM:619767',\n",
              " 'pfbmft6': 'OMIM:619767',\n",
              " 'ichthyotic keratoderma, spasticity, hypomyelination, and dysmorphic facies': 'OMIM:618527',\n",
              " 'elovl1': 'OMIM:618527',\n",
              " 'ssc1': 'OMIM:618527',\n",
              " 'ikshd': 'OMIM:618527',\n",
              " 'asthma, susceptibility to, 1': 'OMIM:607277',\n",
              " 'ptgdr': 'OMIM:607277',\n",
              " 'as1': 'OMIM:607277',\n",
              " 'asrt1': 'OMIM:607277',\n",
              " 'white-kernohan syndrome': 'OMIM:619426',\n",
              " 'ddb1': 'OMIM:619426',\n",
              " 'whikers': 'OMIM:619426',\n",
              " 'lipodystrophy, familial partial, type 8': 'OMIM:620679',\n",
              " 'adra2a': 'OMIM:620679',\n",
              " 'adra2r': 'OMIM:620679',\n",
              " 'fpld8': 'OMIM:620679',\n",
              " 'intellectual developmental disorder, autosomal recessive 80, with variant lissencephaly': 'OMIM:620653',\n",
              " 'casp2': 'OMIM:620653',\n",
              " 'nedd2': 'OMIM:620653',\n",
              " 'ich1': 'OMIM:620653',\n",
              " 'mrt80': 'OMIM:620653',\n",
              " 'neurodevelopmental disorder with impaired language, behavioral abnormalities, and dysmorphic facies': 'OMIM:620494',\n",
              " 'ubap2l': 'OMIM:620494',\n",
              " 'nice4': 'OMIM:620494',\n",
              " 'nedlbf': 'OMIM:620494',\n",
              " 'intellectual developmental disorder, autosomal dominant 74': 'OMIM:620688',\n",
              " 'hnrnpc': 'OMIM:620688',\n",
              " 'hnrpc': 'OMIM:620688',\n",
              " 'mrd74': 'OMIM:620688',\n",
              " 'retinitis pigmentosa 92': 'OMIM:619614',\n",
              " 'hkdc1': 'OMIM:619614',\n",
              " 'rp92': 'OMIM:619614',\n",
              " 'deafness, autosomal dominant 79': 'OMIM:619086',\n",
              " 'scd5': 'OMIM:619086',\n",
              " 'acod4': 'OMIM:619086',\n",
              " 'dfna79': 'OMIM:619086',\n",
              " 'aids, delayed/rapid progression to': 'OMIM:609423',\n",
              " 'kir3dl1': 'OMIM:609423',\n",
              " 'nkat3': 'OMIM:609423',\n",
              " 'nkb1': 'OMIM:609423',\n",
              " 'amb11': 'OMIM:609423',\n",
              " 'kir3ds1': 'OMIM:609423',\n",
              " 'neurodevelopmental disorder with language impairment and behavioral abnormalities': 'OMIM:618917',\n",
              " 'gria2': 'OMIM:618917',\n",
              " 'glur2': 'OMIM:618917',\n",
              " 'nedlib': 'OMIM:618917',\n",
              " 'fasting plasma glucose level qtl 5': 'OMIM:613463',\n",
              " 'gckr': 'OMIM:613463',\n",
              " 'gkrp': 'OMIM:613463',\n",
              " 'fgqtl5': 'OMIM:613463',\n",
              " 'schizophrenia': 'OMIM:181500',\n",
              " 'disc2': 'OMIM:181500',\n",
              " 'blood group, abo system': 'OMIM:616093',\n",
              " 'abo': 'OMIM:616093',\n",
              " 'calvarial doughnut lesions with bone fragility with or without spondylometaphyseal dysplasia': 'OMIM:126550',\n",
              " 'sgms2': 'OMIM:126550',\n",
              " 'sms2': 'OMIM:126550',\n",
              " 'cdl': 'OMIM:126550',\n",
              " 'attention deficit-hyperactivity disorder, susceptibility to': 'OMIM:143465',\n",
              " 'drd5': 'OMIM:143465',\n",
              " 'drd1b': 'OMIM:143465',\n",
              " 'drd1l2': 'OMIM:143465',\n",
              " 'retinitis pigmentosa 89': 'OMIM:618955',\n",
              " 'kif3b': 'OMIM:618955',\n",
              " 'kiaa0359': 'OMIM:618955',\n",
              " 'rp89': 'OMIM:618955',\n",
              " 'inflammatory poikiloderma with hair abnormalities and acral keratoses': 'OMIM:620199',\n",
              " 'ltv1': 'OMIM:620199',\n",
              " 'iphak': 'OMIM:620199',\n",
              " 'megaloblastic anemia, folate-responsive': 'OMIM:601775',\n",
              " 'slc19a1': 'OMIM:601775',\n",
              " 'folt': 'OMIM:601775',\n",
              " 'megaf': 'OMIM:601775',\n",
              " 'imd114': 'OMIM:601775',\n",
              " 'phenylthiocarbamide tasting': 'OMIM:171200',\n",
              " 'tas2r38': 'OMIM:171200',\n",
              " 't2r61': 'OMIM:171200',\n",
              " 'ptc': 'OMIM:171200',\n",
              " 'thiot': 'OMIM:171200',\n",
              " 'rheumatoid arthritis, progression of': 'OMIM:180300',\n",
              " 'il10': 'OMIM:180300',\n",
              " 'csif': 'OMIM:180300',\n",
              " 'gvhds': 'OMIM:180300',\n",
              " 'intellectual developmental disorder, x-linked syndromic, pilorge type': 'OMIM:301076',\n",
              " 'glra2': 'OMIM:301076',\n",
              " 'mrxsp': 'OMIM:301076',\n",
              " 'lipodystrophy, familial partial, type 9': 'OMIM:620683',\n",
              " 'plaat3': 'OMIM:620683',\n",
              " 'pla2g16': 'OMIM:620683',\n",
              " 'hrasls3': 'OMIM:620683',\n",
              " 'hrsl3': 'OMIM:620683',\n",
              " 'hrev107': 'OMIM:620683',\n",
              " 'fpld9': 'OMIM:620683',\n",
              " 'eosinophil peroxidase deficiency': 'OMIM:261500',\n",
              " 'epx': 'OMIM:261500',\n",
              " 'epxd': 'OMIM:261500',\n",
              " 'immune dysregulation, autoimmunity, and autoinflammation': 'OMIM:620514',\n",
              " 'plcg1': 'OMIM:620514',\n",
              " 'plc1': 'OMIM:620514',\n",
              " 'idaa': 'OMIM:620514',\n",
              " 'cholestasis, progressive familial intrahepatic, 6': 'OMIM:619484',\n",
              " 'slc51a': 'OMIM:619484',\n",
              " 'osta': 'OMIM:619484',\n",
              " 'pfic6': 'OMIM:619484',\n",
              " 'leprosy, protection against': 'OMIM:613223',\n",
              " 'tlr1': 'OMIM:613223',\n",
              " 'til. lprs5': 'OMIM:613223',\n",
              " 'intellectual developmental disorder, autosomal dominant 59': 'OMIM:618522',\n",
              " 'camk2g': 'OMIM:618522',\n",
              " 'camkg': 'OMIM:618522',\n",
              " 'mrd59': 'OMIM:618522',\n",
              " 'diabetes mellitus, insulin-dependent, 22': 'OMIM:612522',\n",
              " 'ccr5': 'OMIM:612522',\n",
              " 'cmkbr5': 'OMIM:612522',\n",
              " 'ccckr5': 'OMIM:612522',\n",
              " 'iddm22': 'OMIM:612522',\n",
              " 'aerodigestive tract cancer, squamous cell, alcohol-related, protection against': 'OMIM:103780',\n",
              " 'adh1b': 'OMIM:103780',\n",
              " 'adh2': 'OMIM:103780',\n",
              " 'autoinflammatory-pancytopenia syndrome': 'OMIM:619858',\n",
              " 'dnase2': 'OMIM:619858',\n",
              " 'dnl': 'OMIM:619858',\n",
              " 'aipcs': 'OMIM:619858',\n",
              " 'neurodevelopmental disorder with dysmorphic facies and ischiopubic hypoplasia': 'OMIM:620210',\n",
              " 'nae1': 'OMIM:620210',\n",
              " 'appbp1': 'OMIM:620210',\n",
              " 'nedfih': 'OMIM:620210',\n",
              " 'acromesomelic dysplasia 4': 'OMIM:619636',\n",
              " 'prkg2': 'OMIM:619636',\n",
              " 'prkgr2': 'OMIM:619636',\n",
              " 'amd4': 'OMIM:619636',\n",
              " 'smdp': 'OMIM:619636',\n",
              " 'low density lipoprotein cholesterol level qtl6': 'OMIM:613589',\n",
              " 'sort1': 'OMIM:613589',\n",
              " 'nt3': 'OMIM:613589',\n",
              " 'ldlcq6': 'OMIM:613589',\n",
              " 'epilepsy, idiopathic generalized, susceptibility to, 15': 'OMIM:618357',\n",
              " 'rorb': 'OMIM:618357',\n",
              " 'rzrb': 'OMIM:618357',\n",
              " 'eig15': 'OMIM:618357',\n",
              " 'obesity, susceptibility to': 'OMIM:607514',\n",
              " 'ffar4': 'OMIM:607514',\n",
              " 'o3far1': 'OMIM:607514',\n",
              " 'gpr120': 'OMIM:607514',\n",
              " 'pgr4': 'OMIM:607514',\n",
              " 'bmiq10': 'OMIM:607514',\n",
              " 'spondyloepiphyseal dysplasia, kondo-fu type': 'OMIM:618392',\n",
              " 'mbtps1': 'OMIM:618392',\n",
              " 's1p': 'OMIM:618392',\n",
              " 'sedkf': 'OMIM:618392',\n",
              " 'neurodevelopmental disorder with microcephaly and structural brain anomalies': 'OMIM:618492',\n",
              " 'dync1i2': 'OMIM:618492',\n",
              " 'dnci2': 'OMIM:618492',\n",
              " 'ic2': 'OMIM:618492',\n",
              " 'nedmiba': 'OMIM:618492',\n",
              " 'systemic lupus erythematosus, resistance to': 'OMIM:601744',\n",
              " 'tlr5': 'OMIM:601744',\n",
              " 'til3': 'OMIM:601744',\n",
              " 'sleb1': 'OMIM:601744',\n",
              " 'melios': 'OMIM:601744',\n",
              " 'immunodeficiency 62': 'OMIM:618459',\n",
              " 'arhgef1': 'OMIM:618459',\n",
              " 'lbcl2': 'OMIM:618459',\n",
              " 'imd62': 'OMIM:618459',\n",
              " 'intellectual developmental disorder, autosomal dominant 51': 'OMIM:617788',\n",
              " 'kmt5b': 'OMIM:617788',\n",
              " 'suv420h1': 'OMIM:617788',\n",
              " 'cgi85': 'OMIM:617788',\n",
              " 'mrd51': 'OMIM:617788',\n",
              " 'tan-almurshedi syndrome': 'OMIM:620641',\n",
              " 'drg1': 'OMIM:620641',\n",
              " 'nedd3': 'OMIM:620641',\n",
              " 'tanals': 'OMIM:620641',\n",
              " 'alcohol sensitivity, acute': 'OMIM:610251',\n",
              " 'aldh2': 'OMIM:610251',\n",
              " 'autoinflammation with pulmonary and cutaneous vasculitis': 'OMIM:620296',\n",
              " 'hck': 'OMIM:620296',\n",
              " 'aipcv': 'OMIM:620296',\n",
              " 'asthma, diminished response to antileukotriene treatment in': 'OMIM:600807',\n",
              " 'alox5': 'OMIM:600807',\n",
              " 'neurodevelopmental disorder with cardiomyopathy, spasticity, and brain abnormalities': 'OMIM:619121',\n",
              " 'shmt2': 'OMIM:619121',\n",
              " 'glya': 'OMIM:619121',\n",
              " 'nedcasb': 'OMIM:619121',\n",
              " 'developmental and epileptic encephalopathy 108': 'OMIM:620115',\n",
              " 'mast3': 'OMIM:620115',\n",
              " 'kiaa0561': 'OMIM:620115',\n",
              " 'dee108': 'OMIM:620115',\n",
              " 'lethal congenital contracture syndrome 9': 'OMIM:616503',\n",
              " 'adgrg6': 'OMIM:616503',\n",
              " 'pr126': 'OMIM:616503',\n",
              " 'vigr': 'OMIM:616503',\n",
              " 'lccs9': 'OMIM:616503',\n",
              " 'bone mineral density variation qtl, osteoporosis': 'OMIM:166710',\n",
              " 'col1a1': 'OMIM:166710',\n",
              " 'oi1': 'OMIM:166710',\n",
              " 'oi2': 'OMIM:166710',\n",
              " 'oi3': 'OMIM:166710',\n",
              " 'oi4': 'OMIM:166710',\n",
              " 'edsarth1': 'OMIM:166710',\n",
              " 'cafyd': 'OMIM:166710',\n",
              " 'platelet-activating factor acetylhydrolase deficiency': 'OMIM:614278',\n",
              " 'pla2g7': 'OMIM:614278',\n",
              " 'pafah': 'OMIM:614278',\n",
              " 'pafad': 'OMIM:614278',\n",
              " 'neurodevelopmental disorder with hypotonia and autistic features with or without hyperkinetic movements': 'OMIM:618760',\n",
              " 'vamp2': 'OMIM:618760',\n",
              " 'syb2': 'OMIM:618760',\n",
              " 'nedhahm': 'OMIM:618760',\n",
              " 'malaria, cerebral, reduced risk of': 'OMIM:611162',\n",
              " 'cd36': 'OMIM:611162',\n",
              " 'chds7': 'OMIM:611162',\n",
              " 'bdplt10': 'OMIM:611162',\n",
              " 'neural tube defects, folate-sensitive, susceptibility to': 'OMIM:601634',\n",
              " 'mthfd1': 'OMIM:601634',\n",
              " 'mthfc': 'OMIM:601634',\n",
              " 'cimah': 'OMIM:601634',\n",
              " 'muscular dystrophy, congenital': 'OMIM:254100',\n",
              " 'bet1': 'OMIM:254100',\n",
              " 'mdrp': 'OMIM:254100',\n",
              " 'lui-jee-baron syndrome': 'OMIM:301114',\n",
              " 'spin4': 'OMIM:301114',\n",
              " 'ljbs': 'OMIM:301114',\n",
              " 'neurodevelopmental disorder with hyperkinetic movements, seizures and structural brain abnormalities': 'OMIM:620732',\n",
              " 'pi4k2a': 'OMIM:620732',\n",
              " 'nedmsb': 'OMIM:620732',\n",
              " 'deafness, autosomal dominant 87': 'OMIM:620281',\n",
              " 'pi4kb': 'OMIM:620281',\n",
              " 'pik4cb': 'OMIM:620281',\n",
              " 'dfna87': 'OMIM:620281',\n",
              " 'intellectual developmental disorder, x-linked 106': 'OMIM:300997',\n",
              " 'ogt': 'OMIM:300997',\n",
              " 'xlid106': 'OMIM:300997',\n",
              " 'congenital disorder of glycosylation, type iit': 'OMIM:618885',\n",
              " 'galnt2': 'OMIM:618885',\n",
              " 'cdg2t': 'OMIM:618885',\n",
              " 'proteasome-associated autoinflammatory syndrome 3 and digenic forms': 'OMIM:617591',\n",
              " 'psmb4': 'OMIM:617591',\n",
              " 'praas3': 'OMIM:617591',\n",
              " 'proteasome-associated autoinflammatory syndrome 5': 'OMIM:619175',\n",
              " 'psmb10': 'OMIM:619175',\n",
              " 'mecl1': 'OMIM:619175',\n",
              " 'lmp10': 'OMIM:619175',\n",
              " 'praas5': 'OMIM:619175',\n",
              " 'birk-aharoni syndrome': 'OMIM:620071',\n",
              " 'psmc1': 'OMIM:620071',\n",
              " 's4': 'OMIM:620071',\n",
              " 'nedgth': 'OMIM:620071',\n",
              " 'deafness, cataract, impaired intellectual development, and polyneuropathy': 'OMIM:619354',\n",
              " 'psmc3': 'OMIM:619354',\n",
              " 'tbp1': 'OMIM:619354',\n",
              " 'dcidp': 'OMIM:619354',\n",
              " 'microcornea, rod-cone dystrophy, cataract, and posterior staphyloma 1': 'OMIM:619082',\n",
              " 'arl2': 'OMIM:619082',\n",
              " 'mrcs1': 'OMIM:619082',\n",
              " 'methylmalonic aciduria and homocystinuria, cblc type': 'OMIM:277400',\n",
              " 'mmachc': 'OMIM:277400',\n",
              " 'spondyloepimetaphyseal dysplasia, krakow type': 'OMIM:618162',\n",
              " 'sik3': 'OMIM:618162',\n",
              " 'kiaa0999': 'OMIM:618162',\n",
              " 'semdk': 'OMIM:618162',\n",
              " 'corneal dystrophy, punctiform and polychromatic pre-descemet': 'OMIM:619871',\n",
              " 'prdx3': 'OMIM:619871',\n",
              " 'aop1': 'OMIM:619871',\n",
              " 'scar32': 'OMIM:619871',\n",
              " 'pppcd': 'OMIM:619871',\n",
              " 'hypervalinemia or hyperleucine-isoleucinemia': 'OMIM:618850',\n",
              " 'bcat2': 'OMIM:618850',\n",
              " 'bct2': 'OMIM:618850',\n",
              " 'hvli': 'OMIM:618850',\n",
              " 'immunodeficiency 133 with autoimmunity and autoinflammation': 'OMIM:620565',\n",
              " 'arpc5': 'OMIM:620565',\n",
              " 'arc16': 'OMIM:620565',\n",
              " 'imd113': 'OMIM:620565',\n",
              " 'brachycephaly, trichomegaly, and developmental delay': 'OMIM:617412',\n",
              " 'rps23': 'OMIM:617412',\n",
              " 'btdd': 'OMIM:617412',\n",
              " 'mcins': 'OMIM:617412',\n",
              " 'spondyloepimetaphyseal dysplasia, isidor-toutain type': 'OMIM:618728',\n",
              " 'rpl13': 'OMIM:618728',\n",
              " 'bbc1': 'OMIM:618728',\n",
              " 'd16s44e': 'OMIM:618728',\n",
              " 'semdist': 'OMIM:618728',\n",
              " 'blood group, yt system': 'OMIM:112100',\n",
              " 'ache': 'OMIM:112100',\n",
              " 'yt': 'OMIM:112100',\n",
              " 'drug metabolism, altered, ces1-related': 'OMIM:618057',\n",
              " 'ces1': 'OMIM:618057',\n",
              " 'ses1': 'OMIM:618057',\n",
              " 'vitamin d-dependent rickets, type 3': 'OMIM:619073',\n",
              " 'cyp3a4': 'OMIM:619073',\n",
              " 'vddr3': 'OMIM:619073',\n",
              " 'hypertension, essential': 'OMIM:145500',\n",
              " 'ptgis': 'OMIM:145500',\n",
              " 'cyp8a1': 'OMIM:145500',\n",
              " 'pgis': 'OMIM:145500',\n",
              " 'cyp8': 'OMIM:145500',\n",
              " 'coumarin resistance': 'OMIM:122700',\n",
              " 'cyp2a6': 'OMIM:122700',\n",
              " 'cyp2a3': 'OMIM:122700',\n",
              " 'cyp2a': 'OMIM:122700',\n",
              " 'p450c2a': 'OMIM:122700',\n",
              " 'clopidogrel, impaired responsiveness to': 'OMIM:609535',\n",
              " 'cyp2c': 'OMIM:609535',\n",
              " 'cyp2c19': 'OMIM:609535',\n",
              " 'drug metabolism, altered, cyp2c8-related': 'OMIM:618018',\n",
              " 'cyp2c8': 'OMIM:618018',\n",
              " 'cyp2c8dm': 'OMIM:618018',\n",
              " 'codeine sensitivity': 'OMIM:608902',\n",
              " 'cyp2d6': 'OMIM:608902',\n",
              " 'cpd6': 'OMIM:608902',\n",
              " 'p450db1': 'OMIM:608902',\n",
              " 'efavirenz, poor metabolism of': 'OMIM:614546',\n",
              " 'cyp2b6': 'OMIM:614546',\n",
              " 'cyp2b': 'OMIM:614546',\n",
              " 'efvm': 'OMIM:614546',\n",
              " 'aquaporin-1 deficiency': 'OMIM:110450',\n",
              " 'aqp1': 'OMIM:110450',\n",
              " 'chip28': 'OMIM:110450',\n",
              " 'co': 'OMIM:110450',\n",
              " 'parkinson disease, late-onset, susceptibility to': 'OMIM:168600',\n",
              " 'atxn2': 'OMIM:168600',\n",
              " 'atx2': 'OMIM:168600',\n",
              " 'sca2': 'OMIM:168600',\n",
              " 'asl13': 'OMIM:168600',\n",
              " 'cerebellar atrophy with seizures and variable developmental delay': 'OMIM:618501',\n",
              " 'cacna2d2': 'OMIM:618501',\n",
              " 'kiaa0558': 'OMIM:618501',\n",
              " 'casvdd': 'OMIM:618501',\n",
              " 'boudin-mortier syndrome': 'OMIM:619543',\n",
              " 'npr3': 'OMIM:619543',\n",
              " 'anprc': 'OMIM:619543',\n",
              " 'bomos': 'OMIM:619543',\n",
              " 'immunodeficiency 59 and hypoglycemia': 'OMIM:233600',\n",
              " 'hyou1': 'OMIM:233600',\n",
              " 'imd59': 'OMIM:233600',\n",
              " 'developmental and epileptic encephalopathy 112': 'OMIM:620537',\n",
              " 'kcnh5': 'OMIM:620537',\n",
              " 'dee112': 'OMIM:620537',\n",
              " 'developmental and epileptic encephalopathy 103': 'OMIM:619913',\n",
              " 'kcnc2': 'OMIM:619913',\n",
              " 'dee103': 'OMIM:619913',\n",
              " 'gastric cancer, somatic': 'OMIM:613659',\n",
              " 'apc': 'OMIM:114550',\n",
              " 'gs': 'OMIM:114550',\n",
              " 'fpc': 'OMIM:114550',\n",
              " 'btps2': 'OMIM:114550',\n",
              " 'desmd': 'OMIM:114550',\n",
              " 'myopathy, sarcoplasmic body': 'OMIM:620286',\n",
              " 'mb': 'OMIM:620286',\n",
              " 'myosb': 'OMIM:620286',\n",
              " 'anxiety-related personality traits': 'OMIM:607834',\n",
              " 'slc6a4': 'OMIM:607834',\n",
              " 'htt': 'OMIM:607834',\n",
              " 'ocd1': 'OMIM:607834',\n",
              " 'nabais sa-de vries syndrome, type 2': 'OMIM:618829',\n",
              " 'spop': 'OMIM:618829',\n",
              " 'nsdvs1': 'OMIM:618829',\n",
              " 'nsdvs2': 'OMIM:618829',\n",
              " 'hypertension, diastolic, resistance to': 'OMIM:608622',\n",
              " 'kcnmb1': 'OMIM:608622',\n",
              " 'al kaissi syndrome': 'OMIM:617694',\n",
              " 'cdk10': 'OMIM:617694',\n",
              " 'pisslre': 'OMIM:617694',\n",
              " 'alsas': 'OMIM:617694',\n",
              " 'intellectual developmental disorder with or without peripheral neuropathy': 'OMIM:619844',\n",
              " 'nudt2': 'OMIM:619844',\n",
              " 'apah1': 'OMIM:619844',\n",
              " 'iddpn': 'OMIM:619844',\n",
              " 'major affective disorder-7, susceptibility to': 'OMIM:612371',\n",
              " 'xbp1': 'OMIM:612371',\n",
              " 'xbp2': 'OMIM:612371',\n",
              " 'nasopharyngeal carcinoma, susceptibility to, 3': 'OMIM:617075',\n",
              " 'mst1r': 'OMIM:617075',\n",
              " 'ron': 'OMIM:617075',\n",
              " 'npca3': 'OMIM:617075',\n",
              " 'sulfide:quinone oxidoreductase deficiency': 'OMIM:619221',\n",
              " 'sqor': 'OMIM:619221',\n",
              " 'sqrdl': 'OMIM:619221',\n",
              " 'deafness, autosomal recessive 121': 'OMIM:620551',\n",
              " 'gpr156': 'OMIM:620551',\n",
              " 'gababl': 'OMIM:620551',\n",
              " 'dfnb121': 'OMIM:620551',\n",
              " 'intellectual developmental disorder with dysmorphic facies and behavioral abnormalities': 'OMIM:618089',\n",
              " 'fbxo11': 'OMIM:618089',\n",
              " 'fbx11': 'OMIM:618089',\n",
              " 'vit1': 'OMIM:618089',\n",
              " 'prmt9': 'OMIM:618089',\n",
              " 'iddfba': 'OMIM:618089',\n",
              " 'leukoencephalopathy, motor delay, spasticity, and dysarthria syndrome': 'OMIM:618878',\n",
              " 'eif2ak1': 'OMIM:618878',\n",
              " 'hri': 'OMIM:618878',\n",
              " 'lemspad': 'OMIM:618878',\n",
              " 'muscle hypertrophy': 'OMIM:614160',\n",
              " 'mstn': 'OMIM:614160',\n",
              " 'gdf8': 'OMIM:614160',\n",
              " 'mslhp': 'OMIM:614160',\n",
              " 'asthma, susceptibility to, 2': 'OMIM:608584',\n",
              " 'npsr1': 'OMIM:608584',\n",
              " 'gpr154': 'OMIM:608584',\n",
              " 'gpra': 'OMIM:608584',\n",
              " 'vrr1': 'OMIM:608584',\n",
              " 'pgr14': 'OMIM:608584',\n",
              " 'asrt2': 'OMIM:608584',\n",
              " 'immunodeficiency 82 with systemic inflammation': 'OMIM:619381',\n",
              " 'syk': 'OMIM:619381',\n",
              " 'imd82': 'OMIM:619381',\n",
              " 'leprosy, susceptibility to, 4': 'OMIM:610988',\n",
              " 'lta': 'OMIM:610988',\n",
              " 'tnfb': 'OMIM:610988',\n",
              " 'immunodeficiency 106, susceptibility to viral infections': 'OMIM:619935',\n",
              " 'ifnar1': 'OMIM:619935',\n",
              " 'imd106': 'OMIM:619935',\n",
              " 'hypercholesterolemia, familial, 1': 'OMIM:143890',\n",
              " 'ldlr': 'OMIM:143890',\n",
              " 'fhcl1': 'OMIM:143890',\n",
              " 'fhc': 'OMIM:143890',\n",
              " 'fh': 'OMIM:143890',\n",
              " 'ldlcq2': 'OMIM:143890',\n",
              " 'combined oxidative phosphorylation deficiency 45': 'OMIM:618951',\n",
              " 'mrpl12': 'OMIM:618951',\n",
              " 'rpml12': 'OMIM:618951',\n",
              " 'neurodevelopmental disorder with or without variable movement or behavioral abnormalities': 'OMIM:619725',\n",
              " 'kcnn2': 'OMIM:619725',\n",
              " 'sk2': 'OMIM:619725',\n",
              " 'dyt34': 'OMIM:619725',\n",
              " 'nedmab': 'OMIM:619725',\n",
              " 'developmental and epileptic encephalopathy 69': 'OMIM:618285',\n",
              " 'cacna1e': 'OMIM:618285',\n",
              " 'cacnl1a6': 'OMIM:618285',\n",
              " 'dee69': 'OMIM:618285',\n",
              " 'spermatogenic failure, y-linked, 2': 'OMIM:415000',\n",
              " 'usp9y': 'OMIM:415000',\n",
              " 'dffry': 'OMIM:415000',\n",
              " 'spgfy2': 'OMIM:415000',\n",
              " 'asthma and nasal polyps': 'OMIM:208550',\n",
              " 'tbx21': 'OMIM:208550',\n",
              " 'tbet': 'OMIM:208550',\n",
              " 'imd88': 'OMIM:208550',\n",
              " 'blood group, lewis': 'OMIM:618983',\n",
              " 'fut3': 'OMIM:618983',\n",
              " 'le': 'OMIM:618983',\n",
              " 'bartter syndrome, type 1': 'OMIM:601678',\n",
              " 'slc12a1': 'OMIM:601678',\n",
              " 'nkcc2': 'OMIM:601678',\n",
              " 'autoinflammatory disease, systemic, with vasculitis': 'OMIM:620376',\n",
              " 'lyn': 'OMIM:620376',\n",
              " 'saidv': 'OMIM:620376',\n",
              " 'cortical dysplasia, complex, with other brain malformations 5': 'OMIM:615763',\n",
              " 'tubb2a': 'OMIM:615763',\n",
              " 'cdcbm5': 'OMIM:615763',\n",
              " 'facial palsy, congenital, with ptosis and velopharyngeal dysfunction': 'OMIM:617732',\n",
              " 'tubb6': 'OMIM:617732',\n",
              " 'fpvepd': 'OMIM:617732',\n",
              " 'major depressive disorder 1': 'OMIM:608516',\n",
              " 'mdd1': 'OMIM:608516',\n",
              " 'congenital myopathy 15': 'OMIM:620161',\n",
              " 'tnnc2': 'OMIM:620161',\n",
              " 'cmyp15': 'OMIM:620161',\n",
              " 'chronic recurrent multifocal osteomyelitis 3': 'OMIM:259680',\n",
              " 'il1r1': 'OMIM:259680',\n",
              " 'il1ra': 'OMIM:259680',\n",
              " 'crmo3': 'OMIM:259680',\n",
              " 'hyperemesis gravidarum, susceptibility to': 'OMIM:620730',\n",
              " 'gdf15': 'OMIM:620730',\n",
              " 'plab': 'OMIM:620730',\n",
              " 'pdf': 'OMIM:620730',\n",
              " 'mic1': 'OMIM:620730',\n",
              " 'hg': 'OMIM:620730',\n",
              " 'suleiman-el-hattab syndrome': 'OMIM:618950',\n",
              " 'tasp1': 'OMIM:618950',\n",
              " 'c20orf13': 'OMIM:618950',\n",
              " 'sulehs': 'OMIM:618950',\n",
              " 'neurocardiofaciodigital syndrome': 'OMIM:619869',\n",
              " 'mapkapk5': 'OMIM:619869',\n",
              " 'prak': 'OMIM:619869',\n",
              " 'ncfd': 'OMIM:619869',\n",
              " 'immunodeficiency 96': 'OMIM:619774',\n",
              " 'lig1': 'OMIM:619774',\n",
              " 'imd96': 'OMIM:619774',\n",
              " 'hiatt-neu-cooper neurodevelopmental syndrome': 'OMIM:619311',\n",
              " 'rala': 'OMIM:619311',\n",
              " 'hincons': 'OMIM:619311',\n",
              " 'neurodevelopmental disorder with dysmorphic facies, sleep disturbance, and brain abnormalities': 'OMIM:619103',\n",
              " 'kat5': 'OMIM:619103',\n",
              " 'htatip': 'OMIM:619103',\n",
              " 'tip60': 'OMIM:619103',\n",
              " 'esa1': 'OMIM:619103',\n",
              " 'plip': 'OMIM:619103',\n",
              " 'nedfasb': 'OMIM:619103',\n",
              " 'moyamoya disease 7': 'OMIM:620687',\n",
              " 'ano1': 'OMIM:620687',\n",
              " 'tmem16a': 'OMIM:620687',\n",
              " 'flj10261': 'OMIM:620687',\n",
              " 'indms': 'OMIM:620687',\n",
              " 'mymy7': 'OMIM:620687',\n",
              " 'intellectual developmental disorder, autosomal recessive 67': 'OMIM:618295',\n",
              " 'eif3f': 'OMIM:618295',\n",
              " 'eif3s5': 'OMIM:618295',\n",
              " 'mrt67': 'OMIM:618295',\n",
              " 'febrile seizures, familial, 2': 'OMIM:602477',\n",
              " 'hcn2': 'OMIM:602477',\n",
              " 'bcng2': 'OMIM:602477',\n",
              " 'gefsp11': 'OMIM:602477',\n",
              " 'feb2': 'OMIM:602477',\n",
              " 'eig17': 'OMIM:602477',\n",
              " 'immunodeficiency 71 with inflammatory disease and congenital thrombocytopenia': 'OMIM:617718',\n",
              " 'arpc1b': 'OMIM:617718',\n",
              " 'arc41': 'OMIM:617718',\n",
              " 'imd71': 'OMIM:617718',\n",
              " 'progressive external ophthalmoplegia with mitochondrial dna deletions, autosomal recessive 6': 'OMIM:620647',\n",
              " 'rrm1': 'OMIM:620647',\n",
              " 'peob6': 'OMIM:620647',\n",
              " 'leukotriene c4 synthase deficiency': 'OMIM:614037',\n",
              " 'ltc4s': 'OMIM:614037',\n",
              " 'lissencephaly 7 with cerebellar hypoplasia': 'OMIM:616342',\n",
              " 'cdk5': 'OMIM:616342',\n",
              " 'lis7': 'OMIM:616342',\n",
              " 'hypermetabolism due to uncoupled mitochondrial oxidative phosphorylation 2': 'OMIM:620085',\n",
              " 'atp5b': 'OMIM:620085',\n",
              " 'atpsb': 'OMIM:620085',\n",
              " 'humop2': 'OMIM:620085',\n",
              " 'spermatogenic failure 21': 'OMIM:617644',\n",
              " 'brdt': 'OMIM:617644',\n",
              " 'spgf21': 'OMIM:617644',\n",
              " 'immunodeficiency 109 with lymphoproliferation': 'OMIM:620282',\n",
              " 'tnfrsf9': 'OMIM:620282',\n",
              " 'ila': 'OMIM:620282',\n",
              " 'cd137': 'OMIM:620282',\n",
              " 'imd109': 'OMIM:620282',\n",
              " 'hypotaurinemic retinal degeneration and cardiomyopathy': 'OMIM:145350',\n",
              " 'slc6a6': 'OMIM:145350',\n",
              " 'taut': 'OMIM:145350',\n",
              " 'htrdc': 'OMIM:145350',\n",
              " 'chorea, childhood-onset, with psychomotor retardation': 'OMIM:616939',\n",
              " 'gpr88': 'OMIM:616939',\n",
              " 'strg': 'OMIM:616939',\n",
              " 'cocpmr': 'OMIM:616939',\n",
              " 'neurodevelopmental disorder with microcephaly, cataracts, and renal abnormalities': 'OMIM:617913',\n",
              " 'gemin4': 'OMIM:617913',\n",
              " 'nedmcr': 'OMIM:617913',\n",
              " 'diaphragmatic hernia 4, with cardiovascular defects': 'OMIM:620025',\n",
              " 'aldh1a2': 'OMIM:620025',\n",
              " 'raldh2': 'OMIM:620025',\n",
              " 'dih4': 'OMIM:620025',\n",
              " 'chilton-okur-chung neurodevelopmental syndrome': 'OMIM:619841',\n",
              " 'cdc42bpb': 'OMIM:619841',\n",
              " 'mrckb': 'OMIM:619841',\n",
              " 'chocns': 'OMIM:619841',\n",
              " 'vitamin b12 plasma level qtl1': 'OMIM:612542',\n",
              " 'fut2': 'OMIM:612542',\n",
              " 'se': 'OMIM:612542',\n",
              " 'b12qtl1': 'OMIM:612542',\n",
              " 'oculomotor-abducens synkinesis': 'OMIM:619215',\n",
              " 'ackr3': 'OMIM:619215',\n",
              " 'cxcr7': 'OMIM:619215',\n",
              " 'cmkor1': 'OMIM:619215',\n",
              " 'gpr159': 'OMIM:619215',\n",
              " 'rdc1': 'OMIM:619215',\n",
              " 'blood group, indian system': 'OMIM:609027',\n",
              " 'cd44': 'OMIM:609027',\n",
              " 'mdu2': 'OMIM:609027',\n",
              " 'mdu3': 'OMIM:609027',\n",
              " 'mic4': 'OMIM:609027',\n",
              " 'in': 'OMIM:609027',\n",
              " 'ichthyosis with erythrokeratoderma': 'OMIM:620507',\n",
              " 'klk11': 'OMIM:620507',\n",
              " 'prss20': 'OMIM:620507',\n",
              " 'tlsp': 'OMIM:620507',\n",
              " 'iekd': 'OMIM:620507',\n",
              " 'combined oxidative phosphorylation deficiency 55': 'OMIM:619743',\n",
              " 'polrmt': 'OMIM:619743',\n",
              " 'apolmt': 'OMIM:619743',\n",
              " 'coxpd55': 'OMIM:619743',\n",
              " 'developmental and epileptic encephalopathy 82': 'OMIM:618721',\n",
              " 'got2': 'OMIM:618721',\n",
              " 'dee82': 'OMIM:618721',\n",
              " 'breast cancer, lobular, somatic': 'OMIM:114480',\n",
              " 'cdh1': 'OMIM:114480',\n",
              " 'uvo': 'OMIM:114480',\n",
              " 'lcam': 'OMIM:114480',\n",
              " 'ecad': 'OMIM:114480',\n",
              " 'bcds1': 'OMIM:114480',\n",
              " 'pancreatic lipase deficiency': 'OMIM:614338',\n",
              " 'pnlip': 'OMIM:614338',\n",
              " 'pnlipd': 'OMIM:614338',\n",
              " 'encephalopathy, acute, infection-induced, susceptibility to, 12': 'OMIM:620461',\n",
              " 'rnh1': 'OMIM:620461',\n",
              " 'rnh': 'OMIM:620461',\n",
              " 'iiae12': 'OMIM:620461',\n",
              " 'mitochondrial dna depletion syndrome 18': 'OMIM:618811',\n",
              " 'slc25a21': 'OMIM:618811',\n",
              " 'odc': 'OMIM:618811',\n",
              " 'mtdps18': 'OMIM:618811',\n",
              " 'neuronopathy, distal hereditary motor, autosomal recessive 8': 'OMIM:618912',\n",
              " 'sord': 'OMIM:618912',\n",
              " 'sord1': 'OMIM:618912',\n",
              " 'hmnr8': 'OMIM:618912',\n",
              " 'squalene synthase deficiency': 'OMIM:618156',\n",
              " 'fdft1': 'OMIM:618156',\n",
              " 'dgpt': 'OMIM:618156',\n",
              " 'sqsd': 'OMIM:618156',\n",
              " 'cerebellar dysfunction, impaired intellectual development, and hypogonadotropic hypogonadism': 'OMIM:619761',\n",
              " 'prdm13': 'OMIM:619761',\n",
              " 'cdidhh': 'OMIM:619761',\n",
              " 'pch17': 'OMIM:619761',\n",
              " 'axillary odor, variation in': 'OMIM:117800',\n",
              " 'abcc11': 'OMIM:117800',\n",
              " 'mrp8': 'OMIM:117800',\n",
              " 'ewwd': 'OMIM:117800',\n",
              " 'ww': 'OMIM:117800',\n",
              " 'microvascular complications of diabetes 1': 'OMIM:603933',\n",
              " 'vegf': 'OMIM:603933',\n",
              " 'mvcd1': 'OMIM:603933',\n",
              " 'ciliary dyskinesia, primary, 51': 'OMIM:620438',\n",
              " 'brwd1': 'OMIM:620438',\n",
              " 'wrd9': 'OMIM:620438',\n",
              " 'c21orf107': 'OMIM:620438',\n",
              " 'cild51': 'OMIM:620438',\n",
              " 'developmental and epileptic encephalopathy 57': 'OMIM:617771',\n",
              " 'kcnt2': 'OMIM:617771',\n",
              " 'slick': 'OMIM:617771',\n",
              " 'dee57': 'OMIM:617771',\n",
              " 'immunodeficiency 98 with autoinflammation, x-linked': 'OMIM:301078',\n",
              " 'tlr8': 'OMIM:301078',\n",
              " 'imd98': 'OMIM:301078',\n",
              " 'leukodystrophy, hypomyelinating, 18': 'OMIM:618404',\n",
              " 'degs1': 'OMIM:618404',\n",
              " 'des1': 'OMIM:618404',\n",
              " 'mld': 'ORPHA:309256',\n",
              " 'hld18': 'OMIM:618404',\n",
              " 'intellectual developmental disorder, autosomal dominant 34': 'OMIM:616351',\n",
              " 'col4a3bp': 'OMIM:616351',\n",
              " 'gpbp': 'OMIM:616351',\n",
              " 'cert': 'OMIM:616351',\n",
              " 'mrd34': 'OMIM:616351',\n",
              " 'chronic infections, due to mbl deficiency': 'OMIM:614372',\n",
              " 'mbl2': 'OMIM:614372',\n",
              " 'mbl': 'OMIM:614372',\n",
              " 'mbp1': 'OMIM:614372',\n",
              " 'mbl2d': 'OMIM:614372',\n",
              " 'mbpd': 'OMIM:614372',\n",
              " 'mitochondrial myopathy with lactic acidosis': 'OMIM:251950',\n",
              " 'pnpla8': 'OMIM:251950',\n",
              " 'mmla': 'OMIM:251950',\n",
              " 'congenital disorder of glycosylation with defective fucosylation 1': 'OMIM:618005',\n",
              " 'fut8': 'OMIM:618005',\n",
              " 'cdgf1': 'OMIM:618005',\n",
              " 'delayed sleep phase disorder, susceptibility to': 'OMIM:614163',\n",
              " 'cry1': 'OMIM:614163',\n",
              " 'phll1': 'OMIM:614163',\n",
              " 'dspd': 'OMIM:614163',\n",
              " 'intellectual developmental disorder, autosomal recessive 69': 'OMIM:618383',\n",
              " 'zbtb11': 'OMIM:618383',\n",
              " 'mrt69': 'OMIM:618383',\n",
              " 'okt4 epitope deficiency': 'OMIM:613949',\n",
              " 'cd4': 'OMIM:613949',\n",
              " 'okt4d': 'OMIM:613949',\n",
              " 'imd79': 'OMIM:613949',\n",
              " 'combined oxidative phosphorylation deficiency 46': 'OMIM:618952',\n",
              " 'mrps23': 'OMIM:618952',\n",
              " 'coxpd46': 'OMIM:618952',\n",
              " 'osteogenesis imperfecta, type xxiii': 'OMIM:620639',\n",
              " 'phldb1': 'OMIM:620639',\n",
              " 'll5a': 'OMIM:620639',\n",
              " 'kiaa0638': 'OMIM:620639',\n",
              " 'oi23': 'OMIM:620639',\n",
              " 'immunodeficiency 93 and hypertrophic cardiomyopathy': 'OMIM:619705',\n",
              " 'fnip1': 'OMIM:619705',\n",
              " 'kiaa1961': 'OMIM:619705',\n",
              " 'imd93': 'OMIM:619705',\n",
              " 'neuropathy, hereditary motor and sensory, type vic, with optic atrophy': 'OMIM:618511',\n",
              " 'pdxk': 'OMIM:618511',\n",
              " 'pnk': 'OMIM:618511',\n",
              " 'pkh': 'OMIM:618511',\n",
              " 'hmsn6c': 'OMIM:618511',\n",
              " 'immunodeficiency 47': 'OMIM:300972',\n",
              " 'atp6ap1': 'OMIM:300972',\n",
              " 'atp6ip1': 'OMIM:300972',\n",
              " 'atp6s1': 'OMIM:300972',\n",
              " 'vatps1': 'OMIM:300972',\n",
              " 'leukodystrophy, hypomyelinating, 15': 'OMIM:617951',\n",
              " 'eprs': 'OMIM:617951',\n",
              " 'pars': 'OMIM:617951',\n",
              " 'hld15': 'OMIM:617951',\n",
              " 'cardiac valvular dysplasia 1': 'OMIM:212093',\n",
              " 'pld1': 'OMIM:212093',\n",
              " 'cvdp1': 'OMIM:212093',\n",
              " 'phosphoribosylaminoimidazole carboxylase deficiency': 'OMIM:619859',\n",
              " 'paics': 'OMIM:619859',\n",
              " 'airc': 'OMIM:619859',\n",
              " 'paicsd': 'OMIM:619859',\n",
              " 'ventricular tachycardia, idiopathic': 'OMIM:192605',\n",
              " 'gnai2': 'OMIM:192605',\n",
              " 'gnai2b': 'OMIM:192605',\n",
              " 'gip': 'OMIM:192605',\n",
              " 'combined oxidative phosphorylation deficiency 20': 'OMIM:615917',\n",
              " 'vars2': 'OMIM:615917',\n",
              " 'kiaa1885': 'OMIM:615917',\n",
              " 'coxpd20': 'ORPHA:420728',\n",
              " 'cataract 36': 'OMIM:613887',\n",
              " 'tdrd7': 'OMIM:613887',\n",
              " 'kiaa1529': 'OMIM:613887',\n",
              " 'trap': 'OMIM:613887',\n",
              " 'catc4': 'OMIM:613887',\n",
              " 'ctrct36': 'OMIM:613887',\n",
              " ...}"
            ]
          },
          "metadata": {},
          "execution_count": 82
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "# Literature Mining"
      ],
      "metadata": {
        "id": "_urtkGw-axDl"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "import pandas as pd\n",
        "Lit_mining=pd.read_csv('/content/dyt1.csv')\n",
        "\n",
        "import ast\n",
        "\n",
        "def extract_names_from_string(txt_str):\n",
        "    # Convert the string representation of the list to an actual list\n",
        "    txt = ast.literal_eval(txt_str)\n",
        "    unique_names = set()\n",
        "    for sublist in txt:\n",
        "        for name in sublist:\n",
        "          if name.lower() not in ['dyt1','tor1a','primary dystonias'  ]:\n",
        "            unique_names.add(name.strip())\n",
        "    return list(unique_names)\n",
        "\n",
        "\n",
        "Lit_mining['related_gene']=Lit_mining.apply(lambda x:  extract_names_from_string(x['texts']),axis=1)\n",
        "display(Lit_mining['related_gene'])"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 234
        },
        "id": "1LeG1MdPa1pU",
        "outputId": "50886dde-b79c-4d16-a1cd-bfc27066da20"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "0      [movement anomalies, dystonias, dyskinesias, s...\n",
              "1      [dystonias, DYT24, DYT23, DYT5a, GNAL, dystoni...\n",
              "2      [R149Q, jaw muscles, c.1322A > G, G > A, GNAL,...\n",
              "3      [DYT24, DYT23, PNKD, GNAL, dystonic tremor, dy...\n",
              "4      [HVA, GAG, human, primary dystonia, dystonia, ...\n",
              "                             ...                        \n",
              "380                                   [torsion dystonia]\n",
              "381                        [idiopathic torsion dystonia]\n",
              "382                         [primary torsional dystonia]\n",
              "383                                   [primary dystonia]\n",
              "384                                                   []\n",
              "Name: related_gene, Length: 385, dtype: object"
            ]
          },
          "metadata": {}
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "Lit_mining['title+abstract']=Lit_mining['a.title']+Lit_mining['a.abstractText']\n",
        "Dis_dic={}\n",
        "all_disease= sum(concatenated_df['disease_name+syn'],[])\n",
        "all_disease = [disease.strip().lower() for disease in all_disease if disease != '' and len(disease.strip()) > 2]\n",
        "for  i in Lit_mining.index:\n",
        "  tetxt= Lit_mining['title+abstract'][i].lower()\n",
        "  for j in all_disease:\n",
        "     if j in tetxt:\n",
        "        if j in tetxt.split():\n",
        "           dis = Find_dis_dict[ j ]\n",
        "           if dis in Dis_dic: Dis_dic[dis].append( (Lit_mining['pubmed_id'][i],j))\n",
        "           else:              Dis_dic[dis] = [(Lit_mining['pubmed_id'][i],j)  ]\n",
        "Dis_dic.keys()"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "VuX9ik5ZrdHq",
        "outputId": "6f71f996-d7ea-4aab-b863-f02e6c67f181"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "dict_keys(['ORPHA:206436', 'ORPHA:550', 'ORPHA:974', 'ORPHA:256', 'ORPHA:420429', 'ORPHA:329308', 'ORPHA:98805', 'OMIM:616093', 'ORPHA:440427', 'ORPHA:647', 'ORPHA:803', 'OMIM:112100', 'ORPHA:216866', 'ORPHA:35069', 'OMIM:617951', 'ORPHA:856', 'ORPHA:53351'])"
            ]
          },
          "metadata": {},
          "execution_count": 93
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "for i in list(Dis_dic.keys()):\n",
        "  print(concatenated_df[ concatenated_df ['Disease']==i]['disease_name'].iloc[0])"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "Bk0dp46RtJSB",
        "outputId": "159c7ee4-e266-4c69-fa70-2f2ba12994d2"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Orphanet: Infantile Krabbe disease\n",
            "Orphanet: MELAS\n",
            "Orphanet: Adams-Oliver syndrome\n",
            "Orphanet: Early-onset generalized limb-onset dystonia\n",
            "Orphanet: Glycogen storage disease due to acid maltase deficiency\n",
            "Orphanet: Fatty acid hydroxylase-associated neurodegeneration\n",
            "Orphanet: Primary dystonia\n",
            "Blood group, ABO system\n",
            "Orphanet: Severe early-onset pulmonary alveolar proteinosis due to MARS deficiency\n",
            "Orphanet: Nijmegen breakage syndrome\n",
            "Orphanet: Amyotrophic lateral sclerosis\n",
            "Blood group, Yt system\n",
            "Orphanet: Classic pantothenate kinase-associated neurodegeneration\n",
            "Orphanet: Infantile neuroaxonal dystrophy\n",
            "Leukodystrophy, hypomyelinating, 15\n",
            "Orphanet: NON RARE IN EUROPE: Tourette syndrome\n",
            "Orphanet: X-linked dystonia-parkinsonism\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "Dis_dic"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "zhzBqumy4rzH",
        "outputId": "bd00e5c9-da2f-40eb-cfd2-26fd4ed6db2c"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "{'ORPHA:206436': [('\"18299128\"', 'early-onset'),\n",
              "  ('\"31618684\"', 'early-onset'),\n",
              "  ('\"22166420\"', 'early-onset'),\n",
              "  ('\"36757831\"', 'early-onset'),\n",
              "  ('\"20223934\"', 'early-onset'),\n",
              "  ('\"21995941\"', 'early-onset'),\n",
              "  ('\"19685389\"', 'early-onset'),\n",
              "  ('\"21800139\"', 'early-onset'),\n",
              "  ('\"11528394\"', 'early-onset'),\n",
              "  ('\"37134150\"', 'early-onset'),\n",
              "  ('\"25379658\"', 'early-onset'),\n",
              "  ('\"22770546\"', 'early-onset'),\n",
              "  ('\"32243914\"', 'early-onset'),\n",
              "  ('\"12194383\"', 'early-onset'),\n",
              "  ('\"10814722\"', 'early-onset'),\n",
              "  ('\"35303767\"', 'early-onset'),\n",
              "  ('\"23196519\"', 'early-onset'),\n",
              "  ('\"17956903\"', 'early-onset'),\n",
              "  ('\"36329866\"', 'early-onset'),\n",
              "  ('\"19039992\"', 'early-onset'),\n",
              "  ('\"18827015\"', 'early-onset'),\n",
              "  ('\"22722545\"', 'early-onset'),\n",
              "  ('\"20482602\"', 'early-onset'),\n",
              "  ('\"26004199\"', 'early-onset'),\n",
              "  ('\"15930383\"', 'early-onset'),\n",
              "  ('\"12671990\"', 'early-onset'),\n",
              "  ('\"24938147\"', 'early-onset'),\n",
              "  ('\"25347348\"', 'early-onset'),\n",
              "  ('\"28432771\"', 'early-onset'),\n",
              "  ('\"27143436\"', 'early-onset'),\n",
              "  ('\"23724359\"', 'early-onset'),\n",
              "  ('\"36546658\"', 'early-onset'),\n",
              "  ('\"21136869\"', 'early-onset'),\n",
              "  ('\"15593317\"', 'early-onset'),\n",
              "  ('\"20301665\"', 'early-onset'),\n",
              "  ('\"21078339\"', 'early-onset'),\n",
              "  ('\"20584926\"', 'early-onset'),\n",
              "  ('\"36417958\"', 'early-onset'),\n",
              "  ('\"22345219\"', 'early-onset'),\n",
              "  ('\"24976531\"', 'early-onset'),\n",
              "  ('\"24931141\"', 'early-onset'),\n",
              "  ('\"33445430\"', 'early-onset'),\n",
              "  ('\"34189496\"', 'early-onset'),\n",
              "  ('\"19284587\"', 'early-onset'),\n",
              "  ('\"10477437\"', 'early-onset'),\n",
              "  ('\"23967309\"', 'early-onset'),\n",
              "  ('\"19345148\"', 'early-onset'),\n",
              "  ('\"22080833\"', 'early-onset'),\n",
              "  ('\"12007841\"', 'early-onset'),\n",
              "  ('\"10720299\"', 'early-onset'),\n",
              "  ('\"9585364\"', 'early-onset'),\n",
              "  ('\"14872019\"', 'early-onset'),\n",
              "  ('\"28756192\"', 'early-onset'),\n",
              "  ('\"15548549\"', 'early-onset'),\n",
              "  ('\"18224570\"', 'early-onset'),\n",
              "  ('\"20627944\"', 'early-onset'),\n",
              "  ('\"10627938\"', 'early-onset'),\n",
              "  ('\"10714658\"', 'early-onset'),\n",
              "  ('\"15390042\"', 'early-onset'),\n",
              "  ('\"25203860\"', 'early-onset'),\n",
              "  ('\"8813214\"', 'early-onset'),\n",
              "  ('\"9618171\"', 'early-onset'),\n",
              "  ('\"9288096\"', 'early-onset'),\n",
              "  ('\"18299520\"', 'early-onset'),\n",
              "  ('\"31805123\"', 'early-onset'),\n",
              "  ('\"16537570\"', 'early-onset'),\n",
              "  ('\"10225357\"', 'early-onset'),\n",
              "  ('\"15555920\"', 'early-onset'),\n",
              "  ('\"18267263\"', 'early-onset'),\n",
              "  ('\"18571468\"', 'early-onset'),\n",
              "  ('\"9667600\"', 'early-onset'),\n",
              "  ('\"33832800\"', 'early-onset'),\n",
              "  ('\"28843013\"', 'early-onset'),\n",
              "  ('\"27666935\"', 'early-onset'),\n",
              "  ('\"23840813\"', 'early-onset'),\n",
              "  ('\"18685824\"', 'early-onset'),\n",
              "  ('\"21839475\"', 'early-onset'),\n",
              "  ('\"20687193\"', 'early-onset'),\n",
              "  ('\"9749609\"', 'early-onset'),\n",
              "  ('\"17713116\"', 'early-onset'),\n",
              "  ('\"18477710\"', 'early-onset'),\n",
              "  ('\"29739751\"', 'early-onset'),\n",
              "  ('\"8004804\"', 'early-onset'),\n",
              "  ('\"11921134\"', 'early-onset'),\n",
              "  ('\"16773641\"', 'early-onset'),\n",
              "  ('\"16415980\"', 'early-onset'),\n",
              "  ('\"19345147\"', 'early-onset'),\n",
              "  ('\"25192508\"', 'early-onset'),\n",
              "  ('\"20590811\"', 'early-onset'),\n",
              "  ('\"10435508\"', 'early-onset'),\n",
              "  ('\"9837831\"', 'early-onset'),\n",
              "  ('\"7845403\"', 'early-onset'),\n",
              "  ('\"10448655\"', 'early-onset'),\n",
              "  ('\"20211909\"', 'early-onset'),\n",
              "  ('\"12975293\"', 'early-onset'),\n",
              "  ('\"22622408\"', 'early-onset'),\n",
              "  ('\"36263189\"', 'early-onset'),\n",
              "  ('\"18552369\"', 'early-onset'),\n",
              "  ('\"15326234\"', 'early-onset'),\n",
              "  ('\"11104212\"', 'early-onset'),\n",
              "  ('\"14989804\"', 'early-onset'),\n",
              "  ('\"12821514\"', 'early-onset'),\n",
              "  ('\"17290457\"', 'early-onset'),\n",
              "  ('\"17027035\"', 'early-onset'),\n",
              "  ('\"17503336\"', 'early-onset'),\n",
              "  ('\"26592310\"', 'early-onset'),\n",
              "  ('\"21496603\"', 'early-onset'),\n",
              "  ('\"8079990\"', 'early-onset'),\n",
              "  ('\"9149944\"', 'early-onset'),\n",
              "  ('\"18322712\"', 'early-onset'),\n",
              "  ('\"9750906\"', 'early-onset'),\n",
              "  ('\"7979224\"', 'early-onset'),\n",
              "  ('\"33476878\"', 'early-onset'),\n",
              "  ('\"23107556\"', 'early-onset')],\n",
              " 'ORPHA:550': [('\"36445406\"', 'encephalopathy'),\n",
              "  ('\"31580306\"', 'encephalopathy'),\n",
              "  ('\"22722545\"', 'encephalopathy'),\n",
              "  ('\"24938147\"', 'encephalopathy'),\n",
              "  ('\"17894340\"', 'encephalopathy'),\n",
              "  ('\"25779878\"', 'encephalopathy')],\n",
              " 'ORPHA:974': [('\"22166420\"', 'limb'),\n",
              "  ('\"36757831\"', 'limb'),\n",
              "  ('\"32662044\"', 'limb'),\n",
              "  ('\"22770546\"', 'limb'),\n",
              "  ('\"11984483\"', 'limb'),\n",
              "  ('\"20482602\"', 'limb'),\n",
              "  ('\"19765051\"', 'limb'),\n",
              "  ('\"23724359\"', 'limb'),\n",
              "  ('\"25209853\"', 'limb'),\n",
              "  ('\"24976531\"', 'limb'),\n",
              "  ('\"26087139\"', 'limb'),\n",
              "  ('\"35022352\"', 'limb'),\n",
              "  ('\"18224570\"', 'limb'),\n",
              "  ('\"11261511\"', 'limb'),\n",
              "  ('\"10050893\"', 'limb'),\n",
              "  ('\"17713116\"', 'limb'),\n",
              "  ('\"9191768\"', 'limb'),\n",
              "  ('\"34092466\"', 'limb'),\n",
              "  ('\"16773641\"', 'limb'),\n",
              "  ('\"19345147\"', 'limb'),\n",
              "  ('\"7845403\"', 'limb'),\n",
              "  ('\"10449567\"', 'limb'),\n",
              "  ('\"8309575\"', 'limb'),\n",
              "  ('\"7979224\"', 'limb'),\n",
              "  ('\"33476878\"', 'limb')],\n",
              " 'ORPHA:256': [('\"20223934\"', 'eotd'), ('\"17129379\"', 'eotd')],\n",
              " 'ORPHA:420429': [('\"32243914\"', 'late-onset'),\n",
              "  ('\"32243914\"', 'late-onset'),\n",
              "  ('\"32243914\"', 'late-onset'),\n",
              "  ('\"32243914\"', 'late-onset'),\n",
              "  ('\"32243914\"', 'late-onset'),\n",
              "  ('\"32243914\"', 'late-onset'),\n",
              "  ('\"32243914\"', 'late-onset'),\n",
              "  ('\"32243914\"', 'late-onset'),\n",
              "  ('\"32243914\"', 'late-onset'),\n",
              "  ('\"23724359\"', 'late-onset'),\n",
              "  ('\"23724359\"', 'late-onset'),\n",
              "  ('\"23724359\"', 'late-onset'),\n",
              "  ('\"23724359\"', 'late-onset'),\n",
              "  ('\"23724359\"', 'late-onset'),\n",
              "  ('\"23724359\"', 'late-onset'),\n",
              "  ('\"23724359\"', 'late-onset'),\n",
              "  ('\"23724359\"', 'late-onset'),\n",
              "  ('\"23724359\"', 'late-onset'),\n",
              "  ('\"19284587\"', 'late-onset'),\n",
              "  ('\"19284587\"', 'late-onset'),\n",
              "  ('\"19284587\"', 'late-onset'),\n",
              "  ('\"19284587\"', 'late-onset'),\n",
              "  ('\"19284587\"', 'late-onset'),\n",
              "  ('\"19284587\"', 'late-onset'),\n",
              "  ('\"19284587\"', 'late-onset'),\n",
              "  ('\"19284587\"', 'late-onset'),\n",
              "  ('\"19284587\"', 'late-onset'),\n",
              "  ('\"10627938\"', 'late-onset'),\n",
              "  ('\"10627938\"', 'late-onset'),\n",
              "  ('\"10627938\"', 'late-onset'),\n",
              "  ('\"10627938\"', 'late-onset'),\n",
              "  ('\"10627938\"', 'late-onset'),\n",
              "  ('\"10627938\"', 'late-onset'),\n",
              "  ('\"10627938\"', 'late-onset'),\n",
              "  ('\"10627938\"', 'late-onset'),\n",
              "  ('\"10627938\"', 'late-onset'),\n",
              "  ('\"20590811\"', 'late-onset'),\n",
              "  ('\"20590811\"', 'late-onset'),\n",
              "  ('\"20590811\"', 'late-onset'),\n",
              "  ('\"20590811\"', 'late-onset'),\n",
              "  ('\"20590811\"', 'late-onset'),\n",
              "  ('\"20590811\"', 'late-onset'),\n",
              "  ('\"20590811\"', 'late-onset'),\n",
              "  ('\"20590811\"', 'late-onset'),\n",
              "  ('\"20590811\"', 'late-onset'),\n",
              "  ('\"19955557\"', 'late-onset'),\n",
              "  ('\"19955557\"', 'late-onset'),\n",
              "  ('\"19955557\"', 'late-onset'),\n",
              "  ('\"19955557\"', 'late-onset'),\n",
              "  ('\"19955557\"', 'late-onset'),\n",
              "  ('\"19955557\"', 'late-onset'),\n",
              "  ('\"19955557\"', 'late-onset'),\n",
              "  ('\"19955557\"', 'late-onset'),\n",
              "  ('\"19955557\"', 'late-onset'),\n",
              "  ('\"21301909\"', 'late-onset'),\n",
              "  ('\"21301909\"', 'late-onset'),\n",
              "  ('\"21301909\"', 'late-onset'),\n",
              "  ('\"21301909\"', 'late-onset'),\n",
              "  ('\"21301909\"', 'late-onset'),\n",
              "  ('\"21301909\"', 'late-onset'),\n",
              "  ('\"21301909\"', 'late-onset'),\n",
              "  ('\"21301909\"', 'late-onset'),\n",
              "  ('\"21301909\"', 'late-onset'),\n",
              "  ('\"17027035\"', 'late-onset'),\n",
              "  ('\"17027035\"', 'late-onset'),\n",
              "  ('\"17027035\"', 'late-onset'),\n",
              "  ('\"17027035\"', 'late-onset'),\n",
              "  ('\"17027035\"', 'late-onset'),\n",
              "  ('\"17027035\"', 'late-onset'),\n",
              "  ('\"17027035\"', 'late-onset'),\n",
              "  ('\"17027035\"', 'late-onset'),\n",
              "  ('\"17027035\"', 'late-onset')],\n",
              " 'ORPHA:329308': [('\"23196519\"', 'fahn'), ('\"12821512\"', 'fahn')],\n",
              " 'ORPHA:98805': [('\"21956287\"', 'dyt4')],\n",
              " 'OMIM:616093': [('\"1427899\"', 'abo'), ('\"1639411\"', 'abo')],\n",
              " 'ORPHA:440427': [('\"32516804\"', 'pap')],\n",
              " 'ORPHA:647': [('\"22345219\"', 'ataxia-telangiectasia')],\n",
              " 'ORPHA:803': [('\"24311730\"', 'als')],\n",
              " 'OMIM:112100': [('\"34173686\"', 'ache')],\n",
              " 'ORPHA:216866': [('\"26087139\"', 'nbia1'), ('\"17061185\"', 'pkan')],\n",
              " 'ORPHA:35069': [('\"30028274\"', 'plan')],\n",
              " 'OMIM:617951': [('\"9585364\"', 'pars'),\n",
              "  ('\"32079672\"', 'pars'),\n",
              "  ('\"15455404\"', 'pars'),\n",
              "  ('\"12074905\"', 'pars'),\n",
              "  ('\"31604331\"', 'pars')],\n",
              " 'ORPHA:856': [('\"17702041\"', 'gts')],\n",
              " 'ORPHA:53351': [('\"18299520\"', 'dyt3')]}"
            ]
          },
          "metadata": {},
          "execution_count": 96
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 162
        },
        "id": "nhPrhGyIBT63",
        "outputId": "d902f8d8-96b0-4700-8a43-529032e62d32"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "'\"The monogenic primary dystonias.\"\"Presently, 17 distinct monogenic primary dystonias referred to as dystonias 1- 4, 5a,b, 6-8, 10-13 and 15-18 (loci DYT 1-4, 5a,b, 6-8, 10-13, 15-18) have been recognized. Twelve forms are inherited as autosomal dominant, four as autosomal recessive and one as an X-linked recessive trait. Three additional autosomal dominant forms (DYT9, DYT19 and DYT20) might exist based on linkage mapping to regions apparently different from, yet in close proximity to or overlapping with the known loci DYT18, DYT10 and DYT8. Clinically, this group of movement disorders includes pure dystonias and dystonia plus syndromes. In addition, dyskinesias (paroxysmal dystonias), although phenotypically distinct from classical dystonias, are discussed within this group. In pure dystonias, dystonia is occasionally accompanied by tremor. In dystonia plus syndromes, dystonia as the prominent sign concurs with other movement abnormalities such as myoclonus and parkinsonism. In the dyskinesias, dystonia occurs as a paroxysmal sign in association with other movement anomalies and sometimes seizures. While gross neuropathological changes are absent in most primary dystonias, including the paroxysmal forms, striking morphological alterations are found in some, such as in the X-linked dystonia-parkinsonism syndrome (DYT3). Neuropathological findings at the microscopic level have also been reported in several cases of dystonia 1 and 5, both of which were previously thought to be morphologically normal. One locus, DYT14 had been erroneously assigned, by linkage mapping, in a family with dystonia 5. There are two forms of dystonia 5, one autosomal dominant and one autosomal recessive. These forms are designated here as dystonia 5a and dystonia 5b (DYT5a, DYT5b), respectively. The disease gene has been identified in 10 primary dystonias, seven autosomal dominant (TOR1A/DYT1, GCH1/DYT5a, THAP1/DYT6, PNKD1/MR-1/DYT8, SGCE/DYT11, ATP1A3/DYT12 and SLC2A1/DYT18), two autosomal recessive (TH/DYT5b and PRKRA/DYT16) and one X-chromosomal recessive (TAF1/DYT3). This article summarizes all known aspects on each of the monogenic primary dystonias, including phenotype, neuropathology, imaging, inheritance, mapping, molecular genetics, molecular pathology, animal models and treatment. Suggestions for the diagnostic procedure in primary dystonias are given. Although much is now known about the molecular basis of primary dystonias, treatment of patients is still mainly symptomatic. The only exceptions are dystonias 5a and 5b with their excellent long-term response to L-dopa substitution.\"'"
            ],
            "application/vnd.google.colaboratory.intrinsic+json": {
              "type": "string"
            }
          },
          "metadata": {},
          "execution_count": 100
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "concatenated_df=concatenated_df[['Gene', 'Disease', 'disease_name', 'Syn',\n",
        "       'disease_des/def', 'Clinicalfeatures/Clinical_des', 'sem_def',\n",
        "       'sem_clinical', 'disease_name+syn', 'Liter_chcek']]"
      ],
      "metadata": {
        "id": "_1uJMyKEKHT_"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "dis_potential_match= list(Dis_dic.keys() )\n",
        "for i in concatenated_df.index:\n",
        "   if concatenated_df['Disease'][i] in dis_potential_match:\n",
        "       litt=Dis_dic[concatenated_df['Disease'][i]]\n",
        "       concatenated_df.at[i,'Liter_chcek']= str(litt)\n",
        "       n=0\n",
        "       for j in litt:\n",
        "                defi=concatenated_df['disease_des/def'][i] if type(concatenated_df['disease_des/def'][i]) ==str else '' + concatenated_df[ 'Clinicalfeatures/Clinical_des'][i]  if type(concatenated_df['Clinicalfeatures/Clinical_des'][i]) ==str else ''\n",
        "                abstr=Lit_mining[Lit_mining['pubmed_id']==j[0]\t]['title+abstract'].iloc[0]\n",
        "                num=None\n",
        "                try:\n",
        "                     num=is_about_term(abstr, defi)\n",
        "                except:\n",
        "                     num=is_about_term(abstr[:1000], defi)\n",
        "\n",
        "                concatenated_df.at[i,'Liter_chcek_sem_'+str(n)]=num\n",
        "                n+=1"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 1000
        },
        "id": "4aPpZLRq_fT5",
        "outputId": "32fd86d5-2075-4ef4-ded5-9fe8790e2382"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "<ipython-input-117-aa4176138aee>:16: SettingWithCopyWarning: \n",
            "A value is trying to be set on a copy of a slice from a DataFrame.\n",
            "Try using .loc[row_indexer,col_indexer] = value instead\n",
            "\n",
            "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
            "  concatenated_df.at[i,'Liter_chcek_sem_'+str(n)]=num\n",
            "<ipython-input-117-aa4176138aee>:16: SettingWithCopyWarning: \n",
            "A value is trying to be set on a copy of a slice from a DataFrame.\n",
            "Try using .loc[row_indexer,col_indexer] = value instead\n",
            "\n",
            "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
            "  concatenated_df.at[i,'Liter_chcek_sem_'+str(n)]=num\n",
            "<ipython-input-117-aa4176138aee>:16: SettingWithCopyWarning: \n",
            "A value is trying to be set on a copy of a slice from a DataFrame.\n",
            "Try using .loc[row_indexer,col_indexer] = value instead\n",
            "\n",
            "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
            "  concatenated_df.at[i,'Liter_chcek_sem_'+str(n)]=num\n",
            "<ipython-input-117-aa4176138aee>:16: SettingWithCopyWarning: \n",
            "A value is trying to be set on a copy of a slice from a DataFrame.\n",
            "Try using .loc[row_indexer,col_indexer] = value instead\n",
            "\n",
            "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
            "  concatenated_df.at[i,'Liter_chcek_sem_'+str(n)]=num\n",
            "<ipython-input-117-aa4176138aee>:16: SettingWithCopyWarning: \n",
            "A value is trying to be set on a copy of a slice from a DataFrame.\n",
            "Try using .loc[row_indexer,col_indexer] = value instead\n",
            "\n",
            "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
            "  concatenated_df.at[i,'Liter_chcek_sem_'+str(n)]=num\n"
          ]
        },
        {
          "output_type": "error",
          "ename": "KeyboardInterrupt",
          "evalue": "",
          "traceback": [
            "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
            "\u001b[0;31mRuntimeError\u001b[0m                              Traceback (most recent call last)",
            "\u001b[0;32m<ipython-input-117-aa4176138aee>\u001b[0m in \u001b[0;36m<cell line: 2>\u001b[0;34m()\u001b[0m\n\u001b[1;32m     11\u001b[0m                 \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 12\u001b[0;31m                      \u001b[0mnum\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mis_about_term\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mabstr\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mdefi\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     13\u001b[0m                 \u001b[0;32mexcept\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
            "\u001b[0;32m<ipython-input-97-fdd6e889a858>\u001b[0m in \u001b[0;36mis_about_term\u001b[0;34m(input_text, target_term)\u001b[0m\n\u001b[1;32m    109\u001b[0m   \u001b[0;32mdel\u001b[0m \u001b[0mtest_input\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'token_type_ids'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 110\u001b[0;31m   \u001b[0moutput\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mmodel\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mtest_input\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    111\u001b[0m   \u001b[0msim\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mtorch\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mnn\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mfunctional\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mcosine_similarity\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0moutput\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0moutput\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m1\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mdim\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mitem\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
            "\u001b[0;32m/usr/local/lib/python3.10/dist-packages/torch/nn/modules/module.py\u001b[0m in \u001b[0;36m_wrapped_call_impl\u001b[0;34m(self, *args, **kwargs)\u001b[0m\n\u001b[1;32m   1510\u001b[0m         \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1511\u001b[0;31m             \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_call_impl\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0margs\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1512\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
            "\u001b[0;32m/usr/local/lib/python3.10/dist-packages/torch/nn/modules/module.py\u001b[0m in \u001b[0;36m_call_impl\u001b[0;34m(self, *args, **kwargs)\u001b[0m\n\u001b[1;32m   1519\u001b[0m                 or _global_forward_hooks or _global_forward_pre_hooks):\n\u001b[0;32m-> 1520\u001b[0;31m             \u001b[0;32mreturn\u001b[0m \u001b[0mforward_call\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0margs\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1521\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
            "\u001b[0;32m<ipython-input-97-fdd6e889a858>\u001b[0m in \u001b[0;36mforward\u001b[0;34m(self, input_data)\u001b[0m\n\u001b[1;32m     26\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0mforward\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0minput_data\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 27\u001b[0;31m         \u001b[0moutput\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msts_bert\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0minput_data\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'sentence_embedding'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     28\u001b[0m         \u001b[0;32mreturn\u001b[0m \u001b[0moutput\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
            "\u001b[0;32m/usr/local/lib/python3.10/dist-packages/torch/nn/modules/module.py\u001b[0m in \u001b[0;36m_wrapped_call_impl\u001b[0;34m(self, *args, **kwargs)\u001b[0m\n\u001b[1;32m   1510\u001b[0m         \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1511\u001b[0;31m             \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_call_impl\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0margs\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1512\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
            "\u001b[0;32m/usr/local/lib/python3.10/dist-packages/torch/nn/modules/module.py\u001b[0m in \u001b[0;36m_call_impl\u001b[0;34m(self, *args, **kwargs)\u001b[0m\n\u001b[1;32m   1519\u001b[0m                 or _global_forward_hooks or _global_forward_pre_hooks):\n\u001b[0;32m-> 1520\u001b[0;31m             \u001b[0;32mreturn\u001b[0m \u001b[0mforward_call\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0margs\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1521\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
            "\u001b[0;32m/usr/local/lib/python3.10/dist-packages/torch/nn/modules/container.py\u001b[0m in \u001b[0;36mforward\u001b[0;34m(self, input)\u001b[0m\n\u001b[1;32m    216\u001b[0m         \u001b[0;32mfor\u001b[0m \u001b[0mmodule\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 217\u001b[0;31m             \u001b[0minput\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mmodule\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0minput\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    218\u001b[0m         \u001b[0;32mreturn\u001b[0m \u001b[0minput\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
            "\u001b[0;32m/usr/local/lib/python3.10/dist-packages/torch/nn/modules/module.py\u001b[0m in \u001b[0;36m_wrapped_call_impl\u001b[0;34m(self, *args, **kwargs)\u001b[0m\n\u001b[1;32m   1510\u001b[0m         \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1511\u001b[0;31m             \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_call_impl\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0margs\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1512\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
            "\u001b[0;32m/usr/local/lib/python3.10/dist-packages/torch/nn/modules/module.py\u001b[0m in \u001b[0;36m_call_impl\u001b[0;34m(self, *args, **kwargs)\u001b[0m\n\u001b[1;32m   1519\u001b[0m                 or _global_forward_hooks or _global_forward_pre_hooks):\n\u001b[0;32m-> 1520\u001b[0;31m             \u001b[0;32mreturn\u001b[0m \u001b[0mforward_call\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0margs\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1521\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
            "\u001b[0;32m/usr/local/lib/python3.10/dist-packages/sentence_transformers/models/Transformer.py\u001b[0m in \u001b[0;36mforward\u001b[0;34m(self, features)\u001b[0m\n\u001b[1;32m     97\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 98\u001b[0;31m         \u001b[0moutput_states\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mauto_model\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m**\u001b[0m\u001b[0mtrans_features\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mreturn_dict\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mFalse\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     99\u001b[0m         \u001b[0moutput_tokens\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0moutput_states\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
            "\u001b[0;32m/usr/local/lib/python3.10/dist-packages/torch/nn/modules/module.py\u001b[0m in \u001b[0;36m_wrapped_call_impl\u001b[0;34m(self, *args, **kwargs)\u001b[0m\n\u001b[1;32m   1510\u001b[0m         \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1511\u001b[0;31m             \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_call_impl\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0margs\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1512\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
            "\u001b[0;32m/usr/local/lib/python3.10/dist-packages/torch/nn/modules/module.py\u001b[0m in \u001b[0;36m_call_impl\u001b[0;34m(self, *args, **kwargs)\u001b[0m\n\u001b[1;32m   1519\u001b[0m                 or _global_forward_hooks or _global_forward_pre_hooks):\n\u001b[0;32m-> 1520\u001b[0;31m             \u001b[0;32mreturn\u001b[0m \u001b[0mforward_call\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0margs\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1521\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
            "\u001b[0;32m/usr/local/lib/python3.10/dist-packages/transformers/models/bert/modeling_bert.py\u001b[0m in \u001b[0;36mforward\u001b[0;34m(self, input_ids, attention_mask, token_type_ids, position_ids, head_mask, inputs_embeds, encoder_hidden_states, encoder_attention_mask, past_key_values, use_cache, output_attentions, output_hidden_states, return_dict)\u001b[0m\n\u001b[1;32m    953\u001b[0m                 \u001b[0mbuffered_token_type_ids\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0membeddings\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mtoken_type_ids\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m:\u001b[0m\u001b[0mseq_length\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 954\u001b[0;31m                 \u001b[0mbuffered_token_type_ids_expanded\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mbuffered_token_type_ids\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mexpand\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mbatch_size\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mseq_length\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    955\u001b[0m                 \u001b[0mtoken_type_ids\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mbuffered_token_type_ids_expanded\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
            "\u001b[0;31mRuntimeError\u001b[0m: The expanded size of the tensor (520) must match the existing size (512) at non-singleton dimension 1.  Target sizes: [2, 520].  Tensor sizes: [1, 512]",
            "\nDuring handling of the above exception, another exception occurred:\n",
            "\u001b[0;31mKeyboardInterrupt\u001b[0m                         Traceback (most recent call last)",
            "\u001b[0;32m<ipython-input-117-aa4176138aee>\u001b[0m in \u001b[0;36m<cell line: 2>\u001b[0;34m()\u001b[0m\n\u001b[1;32m     12\u001b[0m                      \u001b[0mnum\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mis_about_term\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mabstr\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mdefi\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     13\u001b[0m                 \u001b[0;32mexcept\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 14\u001b[0;31m                      \u001b[0mnum\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mis_about_term\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mabstr\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;36m1000\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mdefi\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     15\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     16\u001b[0m                 \u001b[0mconcatenated_df\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mat\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mi\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m'Liter_chcek_sem_'\u001b[0m\u001b[0;34m+\u001b[0m\u001b[0mstr\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mn\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mnum\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
            "\u001b[0;32m<ipython-input-97-fdd6e889a858>\u001b[0m in \u001b[0;36mis_about_term\u001b[0;34m(input_text, target_term)\u001b[0m\n\u001b[1;32m    108\u001b[0m   \u001b[0mtest_input\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'attention_mask'\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mtest_input\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'attention_mask'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    109\u001b[0m   \u001b[0;32mdel\u001b[0m \u001b[0mtest_input\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'token_type_ids'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 110\u001b[0;31m   \u001b[0moutput\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mmodel\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mtest_input\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    111\u001b[0m   \u001b[0msim\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mtorch\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mnn\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mfunctional\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mcosine_similarity\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0moutput\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0moutput\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m1\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mdim\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mitem\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    112\u001b[0m   \u001b[0;32mreturn\u001b[0m  \u001b[0mround\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0msim\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;36m2\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
            "\u001b[0;32m/usr/local/lib/python3.10/dist-packages/torch/nn/modules/module.py\u001b[0m in \u001b[0;36m_wrapped_call_impl\u001b[0;34m(self, *args, **kwargs)\u001b[0m\n\u001b[1;32m   1509\u001b[0m             \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_compiled_call_impl\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0margs\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m  \u001b[0;31m# type: ignore[misc]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1510\u001b[0m         \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1511\u001b[0;31m             \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_call_impl\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0margs\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1512\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1513\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m_call_impl\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m*\u001b[0m\u001b[0margs\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
            "\u001b[0;32m/usr/local/lib/python3.10/dist-packages/torch/nn/modules/module.py\u001b[0m in \u001b[0;36m_call_impl\u001b[0;34m(self, *args, **kwargs)\u001b[0m\n\u001b[1;32m   1518\u001b[0m                 \u001b[0;32mor\u001b[0m \u001b[0m_global_backward_pre_hooks\u001b[0m \u001b[0;32mor\u001b[0m \u001b[0m_global_backward_hooks\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1519\u001b[0m                 or _global_forward_hooks or _global_forward_pre_hooks):\n\u001b[0;32m-> 1520\u001b[0;31m             \u001b[0;32mreturn\u001b[0m \u001b[0mforward_call\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0margs\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1521\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1522\u001b[0m         \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
            "\u001b[0;32m<ipython-input-97-fdd6e889a858>\u001b[0m in \u001b[0;36mforward\u001b[0;34m(self, input_data)\u001b[0m\n\u001b[1;32m     25\u001b[0m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msts_bert\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mSentenceTransformer\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mmodules\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mbert\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mpooling_layer\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     26\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0mforward\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0minput_data\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 27\u001b[0;31m         \u001b[0moutput\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msts_bert\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0minput_data\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'sentence_embedding'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     28\u001b[0m         \u001b[0;32mreturn\u001b[0m \u001b[0moutput\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     29\u001b[0m \u001b[0;31m# Instantiate the model and move it to GPU\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
            "\u001b[0;32m/usr/local/lib/python3.10/dist-packages/torch/nn/modules/module.py\u001b[0m in \u001b[0;36m_wrapped_call_impl\u001b[0;34m(self, *args, **kwargs)\u001b[0m\n\u001b[1;32m   1509\u001b[0m             \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_compiled_call_impl\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0margs\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m  \u001b[0;31m# type: ignore[misc]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1510\u001b[0m         \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1511\u001b[0;31m             \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_call_impl\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0margs\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1512\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1513\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m_call_impl\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m*\u001b[0m\u001b[0margs\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
            "\u001b[0;32m/usr/local/lib/python3.10/dist-packages/torch/nn/modules/module.py\u001b[0m in \u001b[0;36m_call_impl\u001b[0;34m(self, *args, **kwargs)\u001b[0m\n\u001b[1;32m   1518\u001b[0m                 \u001b[0;32mor\u001b[0m \u001b[0m_global_backward_pre_hooks\u001b[0m \u001b[0;32mor\u001b[0m \u001b[0m_global_backward_hooks\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1519\u001b[0m                 or _global_forward_hooks or _global_forward_pre_hooks):\n\u001b[0;32m-> 1520\u001b[0;31m             \u001b[0;32mreturn\u001b[0m \u001b[0mforward_call\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0margs\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1521\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1522\u001b[0m         \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
            "\u001b[0;32m/usr/local/lib/python3.10/dist-packages/torch/nn/modules/container.py\u001b[0m in \u001b[0;36mforward\u001b[0;34m(self, input)\u001b[0m\n\u001b[1;32m    215\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0mforward\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0minput\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    216\u001b[0m         \u001b[0;32mfor\u001b[0m \u001b[0mmodule\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 217\u001b[0;31m             \u001b[0minput\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mmodule\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0minput\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    218\u001b[0m         \u001b[0;32mreturn\u001b[0m \u001b[0minput\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    219\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
            "\u001b[0;32m/usr/local/lib/python3.10/dist-packages/torch/nn/modules/module.py\u001b[0m in \u001b[0;36m_wrapped_call_impl\u001b[0;34m(self, *args, **kwargs)\u001b[0m\n\u001b[1;32m   1509\u001b[0m             \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_compiled_call_impl\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0margs\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m  \u001b[0;31m# type: ignore[misc]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1510\u001b[0m         \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1511\u001b[0;31m             \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_call_impl\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0margs\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1512\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1513\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m_call_impl\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m*\u001b[0m\u001b[0margs\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
            "\u001b[0;32m/usr/local/lib/python3.10/dist-packages/torch/nn/modules/module.py\u001b[0m in \u001b[0;36m_call_impl\u001b[0;34m(self, *args, **kwargs)\u001b[0m\n\u001b[1;32m   1518\u001b[0m                 \u001b[0;32mor\u001b[0m \u001b[0m_global_backward_pre_hooks\u001b[0m \u001b[0;32mor\u001b[0m \u001b[0m_global_backward_hooks\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1519\u001b[0m                 or _global_forward_hooks or _global_forward_pre_hooks):\n\u001b[0;32m-> 1520\u001b[0;31m             \u001b[0;32mreturn\u001b[0m \u001b[0mforward_call\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0margs\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1521\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1522\u001b[0m         \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
            "\u001b[0;32m/usr/local/lib/python3.10/dist-packages/sentence_transformers/models/Transformer.py\u001b[0m in \u001b[0;36mforward\u001b[0;34m(self, features)\u001b[0m\n\u001b[1;32m     96\u001b[0m             \u001b[0mtrans_features\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m\"token_type_ids\"\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mfeatures\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m\"token_type_ids\"\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     97\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 98\u001b[0;31m         \u001b[0moutput_states\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mauto_model\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m**\u001b[0m\u001b[0mtrans_features\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mreturn_dict\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mFalse\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     99\u001b[0m         \u001b[0moutput_tokens\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0moutput_states\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    100\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
            "\u001b[0;32m/usr/local/lib/python3.10/dist-packages/torch/nn/modules/module.py\u001b[0m in \u001b[0;36m_wrapped_call_impl\u001b[0;34m(self, *args, **kwargs)\u001b[0m\n\u001b[1;32m   1509\u001b[0m             \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_compiled_call_impl\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0margs\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m  \u001b[0;31m# type: ignore[misc]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1510\u001b[0m         \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1511\u001b[0;31m             \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_call_impl\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0margs\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1512\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1513\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m_call_impl\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m*\u001b[0m\u001b[0margs\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
            "\u001b[0;32m/usr/local/lib/python3.10/dist-packages/torch/nn/modules/module.py\u001b[0m in \u001b[0;36m_call_impl\u001b[0;34m(self, *args, **kwargs)\u001b[0m\n\u001b[1;32m   1518\u001b[0m                 \u001b[0;32mor\u001b[0m \u001b[0m_global_backward_pre_hooks\u001b[0m \u001b[0;32mor\u001b[0m \u001b[0m_global_backward_hooks\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1519\u001b[0m                 or _global_forward_hooks or _global_forward_pre_hooks):\n\u001b[0;32m-> 1520\u001b[0;31m             \u001b[0;32mreturn\u001b[0m \u001b[0mforward_call\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0margs\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1521\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1522\u001b[0m         \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
            "\u001b[0;32m/usr/local/lib/python3.10/dist-packages/transformers/models/bert/modeling_bert.py\u001b[0m in \u001b[0;36mforward\u001b[0;34m(self, input_ids, attention_mask, token_type_ids, position_ids, head_mask, inputs_embeds, encoder_hidden_states, encoder_attention_mask, past_key_values, use_cache, output_attentions, output_hidden_states, return_dict)\u001b[0m\n\u001b[1;32m    986\u001b[0m             \u001b[0mpast_key_values_length\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mpast_key_values_length\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    987\u001b[0m         )\n\u001b[0;32m--> 988\u001b[0;31m         encoder_outputs = self.encoder(\n\u001b[0m\u001b[1;32m    989\u001b[0m             \u001b[0membedding_output\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    990\u001b[0m             \u001b[0mattention_mask\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mextended_attention_mask\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
            "\u001b[0;32m/usr/local/lib/python3.10/dist-packages/torch/nn/modules/module.py\u001b[0m in \u001b[0;36m_wrapped_call_impl\u001b[0;34m(self, *args, **kwargs)\u001b[0m\n\u001b[1;32m   1509\u001b[0m             \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_compiled_call_impl\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0margs\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m  \u001b[0;31m# type: ignore[misc]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1510\u001b[0m         \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1511\u001b[0;31m             \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_call_impl\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0margs\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1512\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1513\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m_call_impl\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m*\u001b[0m\u001b[0margs\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
            "\u001b[0;32m/usr/local/lib/python3.10/dist-packages/torch/nn/modules/module.py\u001b[0m in \u001b[0;36m_call_impl\u001b[0;34m(self, *args, **kwargs)\u001b[0m\n\u001b[1;32m   1518\u001b[0m                 \u001b[0;32mor\u001b[0m \u001b[0m_global_backward_pre_hooks\u001b[0m \u001b[0;32mor\u001b[0m \u001b[0m_global_backward_hooks\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1519\u001b[0m                 or _global_forward_hooks or _global_forward_pre_hooks):\n\u001b[0;32m-> 1520\u001b[0;31m             \u001b[0;32mreturn\u001b[0m \u001b[0mforward_call\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0margs\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1521\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1522\u001b[0m         \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
            "\u001b[0;32m/usr/local/lib/python3.10/dist-packages/transformers/models/bert/modeling_bert.py\u001b[0m in \u001b[0;36mforward\u001b[0;34m(self, hidden_states, attention_mask, head_mask, encoder_hidden_states, encoder_attention_mask, past_key_values, use_cache, output_attentions, output_hidden_states, return_dict)\u001b[0m\n\u001b[1;32m    580\u001b[0m                 )\n\u001b[1;32m    581\u001b[0m             \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 582\u001b[0;31m                 layer_outputs = layer_module(\n\u001b[0m\u001b[1;32m    583\u001b[0m                     \u001b[0mhidden_states\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    584\u001b[0m                     \u001b[0mattention_mask\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
            "\u001b[0;32m/usr/local/lib/python3.10/dist-packages/torch/nn/modules/module.py\u001b[0m in \u001b[0;36m_wrapped_call_impl\u001b[0;34m(self, *args, **kwargs)\u001b[0m\n\u001b[1;32m   1509\u001b[0m             \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_compiled_call_impl\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0margs\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m  \u001b[0;31m# type: ignore[misc]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1510\u001b[0m         \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1511\u001b[0;31m             \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_call_impl\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0margs\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1512\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1513\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m_call_impl\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m*\u001b[0m\u001b[0margs\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
            "\u001b[0;32m/usr/local/lib/python3.10/dist-packages/torch/nn/modules/module.py\u001b[0m in \u001b[0;36m_call_impl\u001b[0;34m(self, *args, **kwargs)\u001b[0m\n\u001b[1;32m   1518\u001b[0m                 \u001b[0;32mor\u001b[0m \u001b[0m_global_backward_pre_hooks\u001b[0m \u001b[0;32mor\u001b[0m \u001b[0m_global_backward_hooks\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1519\u001b[0m                 or _global_forward_hooks or _global_forward_pre_hooks):\n\u001b[0;32m-> 1520\u001b[0;31m             \u001b[0;32mreturn\u001b[0m \u001b[0mforward_call\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0margs\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1521\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1522\u001b[0m         \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
            "\u001b[0;32m/usr/local/lib/python3.10/dist-packages/transformers/models/bert/modeling_bert.py\u001b[0m in \u001b[0;36mforward\u001b[0;34m(self, hidden_states, attention_mask, head_mask, encoder_hidden_states, encoder_attention_mask, past_key_value, output_attentions)\u001b[0m\n\u001b[1;32m    470\u001b[0m         \u001b[0;31m# decoder uni-directional self-attention cached key/values tuple is at positions 1,2\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    471\u001b[0m         \u001b[0mself_attn_past_key_value\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mpast_key_value\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;36m2\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;32mif\u001b[0m \u001b[0mpast_key_value\u001b[0m \u001b[0;32mis\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0;32mNone\u001b[0m \u001b[0;32melse\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 472\u001b[0;31m         self_attention_outputs = self.attention(\n\u001b[0m\u001b[1;32m    473\u001b[0m             \u001b[0mhidden_states\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    474\u001b[0m             \u001b[0mattention_mask\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
            "\u001b[0;32m/usr/local/lib/python3.10/dist-packages/torch/nn/modules/module.py\u001b[0m in \u001b[0;36m_wrapped_call_impl\u001b[0;34m(self, *args, **kwargs)\u001b[0m\n\u001b[1;32m   1509\u001b[0m             \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_compiled_call_impl\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0margs\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m  \u001b[0;31m# type: ignore[misc]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1510\u001b[0m         \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1511\u001b[0;31m             \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_call_impl\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0margs\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1512\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1513\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m_call_impl\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m*\u001b[0m\u001b[0margs\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
            "\u001b[0;32m/usr/local/lib/python3.10/dist-packages/torch/nn/modules/module.py\u001b[0m in \u001b[0;36m_call_impl\u001b[0;34m(self, *args, **kwargs)\u001b[0m\n\u001b[1;32m   1518\u001b[0m                 \u001b[0;32mor\u001b[0m \u001b[0m_global_backward_pre_hooks\u001b[0m \u001b[0;32mor\u001b[0m \u001b[0m_global_backward_hooks\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1519\u001b[0m                 or _global_forward_hooks or _global_forward_pre_hooks):\n\u001b[0;32m-> 1520\u001b[0;31m             \u001b[0;32mreturn\u001b[0m \u001b[0mforward_call\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0margs\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1521\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1522\u001b[0m         \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
            "\u001b[0;32m/usr/local/lib/python3.10/dist-packages/transformers/models/bert/modeling_bert.py\u001b[0m in \u001b[0;36mforward\u001b[0;34m(self, hidden_states, attention_mask, head_mask, encoder_hidden_states, encoder_attention_mask, past_key_value, output_attentions)\u001b[0m\n\u001b[1;32m    409\u001b[0m             \u001b[0moutput_attentions\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    410\u001b[0m         )\n\u001b[0;32m--> 411\u001b[0;31m         \u001b[0mattention_output\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0moutput\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself_outputs\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mhidden_states\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    412\u001b[0m         \u001b[0moutputs\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m(\u001b[0m\u001b[0mattention_output\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;34m+\u001b[0m \u001b[0mself_outputs\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m1\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m]\u001b[0m  \u001b[0;31m# add attentions if we output them\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    413\u001b[0m         \u001b[0;32mreturn\u001b[0m \u001b[0moutputs\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
            "\u001b[0;32m/usr/local/lib/python3.10/dist-packages/torch/nn/modules/module.py\u001b[0m in \u001b[0;36m_wrapped_call_impl\u001b[0;34m(self, *args, **kwargs)\u001b[0m\n\u001b[1;32m   1509\u001b[0m             \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_compiled_call_impl\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0margs\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m  \u001b[0;31m# type: ignore[misc]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1510\u001b[0m         \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1511\u001b[0;31m             \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_call_impl\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0margs\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1512\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1513\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m_call_impl\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m*\u001b[0m\u001b[0margs\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
            "\u001b[0;32m/usr/local/lib/python3.10/dist-packages/torch/nn/modules/module.py\u001b[0m in \u001b[0;36m_call_impl\u001b[0;34m(self, *args, **kwargs)\u001b[0m\n\u001b[1;32m   1518\u001b[0m                 \u001b[0;32mor\u001b[0m \u001b[0m_global_backward_pre_hooks\u001b[0m \u001b[0;32mor\u001b[0m \u001b[0m_global_backward_hooks\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1519\u001b[0m                 or _global_forward_hooks or _global_forward_pre_hooks):\n\u001b[0;32m-> 1520\u001b[0;31m             \u001b[0;32mreturn\u001b[0m \u001b[0mforward_call\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0margs\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1521\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1522\u001b[0m         \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
            "\u001b[0;32m/usr/local/lib/python3.10/dist-packages/transformers/models/bert/modeling_bert.py\u001b[0m in \u001b[0;36mforward\u001b[0;34m(self, hidden_states, input_tensor)\u001b[0m\n\u001b[1;32m    359\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    360\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0mforward\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mhidden_states\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0mtorch\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mTensor\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0minput_tensor\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0mtorch\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mTensor\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;34m->\u001b[0m \u001b[0mtorch\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mTensor\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 361\u001b[0;31m         \u001b[0mhidden_states\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mdense\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mhidden_states\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    362\u001b[0m         \u001b[0mhidden_states\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mdropout\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mhidden_states\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    363\u001b[0m         \u001b[0mhidden_states\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mLayerNorm\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mhidden_states\u001b[0m \u001b[0;34m+\u001b[0m \u001b[0minput_tensor\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
            "\u001b[0;32m/usr/local/lib/python3.10/dist-packages/torch/nn/modules/module.py\u001b[0m in \u001b[0;36m_wrapped_call_impl\u001b[0;34m(self, *args, **kwargs)\u001b[0m\n\u001b[1;32m   1509\u001b[0m             \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_compiled_call_impl\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0margs\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m  \u001b[0;31m# type: ignore[misc]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1510\u001b[0m         \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1511\u001b[0;31m             \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_call_impl\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0margs\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1512\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1513\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m_call_impl\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m*\u001b[0m\u001b[0margs\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
            "\u001b[0;32m/usr/local/lib/python3.10/dist-packages/torch/nn/modules/module.py\u001b[0m in \u001b[0;36m_call_impl\u001b[0;34m(self, *args, **kwargs)\u001b[0m\n\u001b[1;32m   1518\u001b[0m                 \u001b[0;32mor\u001b[0m \u001b[0m_global_backward_pre_hooks\u001b[0m \u001b[0;32mor\u001b[0m \u001b[0m_global_backward_hooks\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1519\u001b[0m                 or _global_forward_hooks or _global_forward_pre_hooks):\n\u001b[0;32m-> 1520\u001b[0;31m             \u001b[0;32mreturn\u001b[0m \u001b[0mforward_call\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0margs\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1521\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1522\u001b[0m         \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
            "\u001b[0;32m/usr/local/lib/python3.10/dist-packages/torch/nn/modules/linear.py\u001b[0m in \u001b[0;36mforward\u001b[0;34m(self, input)\u001b[0m\n\u001b[1;32m    114\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    115\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0mforward\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0minput\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0mTensor\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;34m->\u001b[0m \u001b[0mTensor\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 116\u001b[0;31m         \u001b[0;32mreturn\u001b[0m \u001b[0mF\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mlinear\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0minput\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mweight\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mbias\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    117\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    118\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0mextra_repr\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;34m->\u001b[0m \u001b[0mstr\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
            "\u001b[0;31mKeyboardInterrupt\u001b[0m: "
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "concatenated_df.to_csv('concatenated_df_2.csv')"
      ],
      "metadata": {
        "id": "iG0QRG4AGhEr"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "import nltk\n",
        "nltk.download('punkt')\n",
        "def get_stem(text):\n",
        "    # Initialize the Porter Stemmer\n",
        "    stemmer = PorterStemmer()\n",
        "    # Tokenize the text into words\n",
        "    words = nltk.word_tokenize(text)\n",
        "    # Get the stem of each word and join them back into a sentence\n",
        "    stemmed_words = [stemmer.stem(word) for word in words]\n",
        "    stemmed_text = ' '.join(stemmed_words)\n",
        "    return stemmed_text"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "we9B0jsSrm0p",
        "outputId": "fbee8efc-1097-4a38-c77f-38185274a218"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "[nltk_data] Downloading package punkt to /root/nltk_data...\n",
            "[nltk_data]   Unzipping tokenizers/punkt.zip.\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "import re\n",
        "\n",
        "nltk.download('punkt')\n",
        "def get_stem(text):\n",
        "    # Initialize the Porter Stemmer\n",
        "    stemmer = PorterStemmer()\n",
        "    # Tokenize the text into words\n",
        "    words = nltk.word_tokenize(text)\n",
        "    # Get the stem of each word and join them back into a sentence\n",
        "    stemmed_words = [stemmer.stem(word) for word in words]\n",
        "    stemmed_text = ' '.join(stemmed_words)\n",
        "    return stemmed_text\n",
        "list_tor1a = ['dyt1', 'tor1a', 'primary dystonias', 'DYT1 dystonia', 'TOR1A dystonia',\n",
        "                  'Early-onset torsion dystonia', \"Oppenheim's dystonia\", 'Primary torsion dystonia',\n",
        "                  'Idiopathic torsion dystonia']\n",
        "\n",
        "def calculate_count(shared_gene_phenotype_table, Lit_mining):\n",
        "    # Initialize count column to zeros\n",
        "    #shared_gene_phenotype_table['count'] = 0\n",
        "    # Compile a regex pattern to split text into sentences\n",
        "    sentence_splitter = re.compile(r'(?<!\\w\\.\\w.)(?<![A-Z][a-z]\\.)(?<=\\.|\\?)\\s')\n",
        "    # Iterate over rows in shared_gene_phenotype_table\n",
        "\n",
        "        # Iterate over rows in Lit_mining\n",
        "    for _, lit_row in Lit_mining.iterrows():\n",
        "        abstract_text = lit_row['a.abstractText'].lower()\n",
        "        sentences =[lit_row['a.title'].lower()]+ sentence_splitter.split(abstract_text)\n",
        "        for index, row in shared_gene_phenotype_table.iterrows():\n",
        "            phenotype_values = row['phenotype']\n",
        "            # Check for concurrency with phenotype_values and list_tor1a\n",
        "            if any(any(phenotype.lower() in sentence and any(tor1a.lower() in sentence for tor1a in list_tor1a) for phenotype in phenotype_values) for sentence in sentences):\n",
        "                shared_gene_phenotype_table.at[index, 'count_phenotype'] += 1\n",
        "                #break  # Move to the next row in shared_gene_phenotype_table once count is updated\n",
        "\n",
        "    return shared_gene_phenotype_table\n",
        "#Lit_mining1=Lit_mining[Lit_mining['a.abstractText']!='\"\"']\n",
        "shared_gene_phenotype_table1=shared_gene_phenotype_table# [shared_gene_phenotype_table['phenotype']!= []]\n",
        "shared_gene_phenotype_table1= calculate_count(shared_gene_phenotype_table1, Lit_mining)\n",
        "shared_gene_phenotype_table1"
      ],
      "metadata": {
        "id": "UIJrnqwKbMNz"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "import re\n",
        "list_tor1a = ['dyt1', 'tor1a', 'primary dystonias', 'DYT1 dystonia', 'TOR1A dystonia',\n",
        "                  'Early-onset torsion dystonia', \"Oppenheim's dystonia\", 'Primary torsion dystonia',\n",
        "                  'Idiopathic torsion dystonia']\n",
        "\n",
        "def calculate_count(shared_gene_phenotype_table, Lit_mining):\n",
        "    # Initialize count column to zeros\n",
        "    # Compile a regex pattern to split text into sentences\n",
        "    sentence_splitter = re.compile(r'(?<!\\w\\.\\w.)(?<![A-Z][a-z]\\.)(?<=\\.|\\?)\\s')\n",
        "    # Iterate over rows in shared_gene_phenotype_table\n",
        "\n",
        "        # Iterate over rows in Lit_mining\n",
        "    for _, lit_row in Lit_mining.iterrows():\n",
        "        abstract_text = lit_row['a.abstractText'].lower()\n",
        "        sentences =[lit_row['a.title'].lower()]+ sentence_splitter.split(abstract_text)\n",
        "        for index, row in shared_gene_phenotype_table.iterrows():\n",
        "            phenotype_values = row['shared_gene']\n",
        "            # Check for concurrency with phenotype_values and list_tor1a\n",
        "            if any(any(phenotype.lower() == word and any(tor1a.lower() == word for tor1a in list_tor1a) for word in sentence.split()) for phenotype in phenotype_values for sentence in sentences):\n",
        "                shared_gene_phenotype_table.at[index, 'count_gene'] += 1\n",
        "                #break  # Move to the next row in shared_gene_phenotype_table once count is updated\n",
        "\n",
        "    return shared_gene_phenotype_table\n",
        "#Lit_mining1=Lit_mining[Lit_mining['a.abstractText']!='\"\"']\n",
        "shared_gene_phenotype_table1= calculate_count(shared_gene_phenotype_table1, Lit_mining)\n",
        "shared_gene_phenotype_table1['count']=shared_gene_phenotype_table1['count_phenotype'] + shared_gene_phenotype_table1['count_gene']\n",
        "\n",
        "shared_gene_phenotype_table1"
      ],
      "metadata": {
        "id": "50Byy-WZbMRS"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "shared_gene_phenotype_table1['count']=shared_gene_phenotype_table1['count_phenotype'] + shared_gene_phenotype_table1['count_gene']\n",
        "shared_gene_phenotype_table1"
      ],
      "metadata": {
        "id": "hW3hzyCubhhy"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "shared_gene_phenotype_table1.to_csv('shared_gene_phenotype_table.csv', index=False)"
      ],
      "metadata": {
        "id": "NK1nU1ijbmeS"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "# approches"
      ],
      "metadata": {
        "id": "j8JCC3M_e5n3"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "import requests\n",
        "from bs4 import BeautifulSoup\n",
        "\n",
        "url = 'https://www.orpha.net/en/disease/detail/676'\n",
        "response = requests.get(url)\n",
        "soup = BeautifulSoup(response.text, 'html.parser')\n",
        "\n",
        "# Find the element containing the disease name\n",
        "disease_name_element = soup.find('h1', class_='title')\n",
        "\n",
        "if disease_name_element:\n",
        "    disease_name = disease_name_element.text.strip()\n",
        "    print(f\"The name of the disease is: {disease_name}\")\n",
        "else:\n",
        "    print(\"Disease name not found.\")"
      ],
      "metadata": {
        "id": "MlEdNFVFe7dS"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "import requests\n",
        "\n",
        "def fetch_page_content(url):\n",
        "    try:\n",
        "        response = requests.get(url)\n",
        "        if response.status_code == 200:\n",
        "            return response.text\n",
        "        else:\n",
        "            print(\"Failed to fetch page content. Status code:\", response.status_code)\n",
        "            return None\n",
        "    except requests.exceptions.RequestException as e:\n",
        "        print(\"Error:\", e)\n",
        "        return None\n",
        "\n",
        "def main():\n",
        "    url = \"https://www.orpha.net/en/disease/detail/676\"\n",
        "    page_content = fetch_page_content(url)\n",
        "    if page_content:\n",
        "        print(page_content)\n",
        "\n",
        "if __name__ == \"__main__\":\n",
        "    main()"
      ],
      "metadata": {
        "id": "080_21eue7i7"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "import requests\n",
        "from bs4 import BeautifulSoup\n",
        "def get_name(num):\n",
        "  # URL of the page to scrape\n",
        "  url = \"https://www.orpha.net/en/disease/detail/\"+str(num)\n",
        "  # Send a GET request to the URL\n",
        "  response = requests.get(url)\n",
        "  # Parse the HTML content\n",
        "  soup = BeautifulSoup(response.text, 'html.parser')\n",
        "  # Find the title tag and extract its text\n",
        "  title_tag = soup.find('title')\n",
        "  title = title_tag.text if title_tag else \"Title not found\"\n",
        "  # Print the extracted title\n",
        "  return title\n",
        "  #print(\"Title:\", title)"
      ],
      "metadata": {
        "id": "woSMWZ_Cfd8r"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "import requests\n",
        "\n",
        "# URL of the page\n",
        "url = \"https://www.orpha.net/en/disease/detail/746\"\n",
        "\n",
        "# Send a GET request to the URL\n",
        "response = requests.get(url)\n",
        "\n",
        "# Check if the request was successful (status code 200)\n",
        "if response.status_code == 200:\n",
        "    # Print the content of the page\n",
        "    print(response.text)\n",
        "else:\n",
        "    # Print an error message if the request failed\n",
        "    print(\"Failed to fetch the page. Status code:\", response.status_code)"
      ],
      "metadata": {
        "id": "ZKIpL_UjfqYm"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "import requests\n",
        "from bs4 import BeautifulSoup\n",
        "\n",
        "def get_synonyms(num):\n",
        "    # URL of the page to scrape\n",
        "    url = \"https://www.orpha.net/en/disease/detail/\" + str(num)\n",
        "    # Send a GET request to the URL\n",
        "    response = requests.get(url)\n",
        "    # Parse the HTML content\n",
        "    soup = BeautifulSoup(response.text, 'html.parser')\n",
        "    # Find the <strong> tag with the text \"Synonym(s):\"\n",
        "    synonym_tag = soup.find('strong', string=\"Synonym(s): \")\n",
        "    # If the <strong> tag is found\n",
        "    if synonym_tag:\n",
        "        # Find the <ul> tag following the <strong> tag\n",
        "        ul_tag = synonym_tag.find_next('ul')\n",
        "        # Find all <li> tags within the <ul> tag\n",
        "        synonym_tags = ul_tag.find_all('li')\n",
        "        # Extract and return the synonyms\n",
        "        synonyms = [tag.get_text(strip=True) for tag in synonym_tags]\n",
        "        return  \", \".join(synonyms)  #,synonyms\n",
        "    else:\n",
        "        return [\"Synonyms not found\"]\n",
        "\n",
        "# Test the function\n",
        "synonyms = get_synonyms(746)\n",
        "print(synonyms)"
      ],
      "metadata": {
        "id": "MG_3E9PgfwZR"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "def get_disease_definition(num):\n",
        "    # URL of the page to scrape\n",
        "    url = \"https://www.orpha.net/en/disease/detail/\" + str(num)\n",
        "    # Send a GET request to the URL\n",
        "    response = requests.get(url)\n",
        "    # Parse the HTML content\n",
        "    soup = BeautifulSoup(response.text, 'html.parser')\n",
        "    # Find the <strong> tag with the class \"d-block mb-3 service-color\" containing the text \"Disease definition\"\n",
        "    definition_strong = soup.find('strong', class_=\"d-block mb-3 service-color\", string=\"Disease definition\")\n",
        "    # If the <strong> tag is found\n",
        "    if definition_strong:\n",
        "        # Get the next <p> tag which contains the disease definition\n",
        "        definition_p = definition_strong.find_next('p')\n",
        "        # If <p> tag exists\n",
        "        if definition_p:\n",
        "            # Extract and return the disease definition\n",
        "            return definition_p.get_text(strip=True)\n",
        "    # If disease definition is not found\n",
        "    return \"Disease definition not found\"\n",
        "\n",
        "# Test the function\n",
        "disease_definition = get_disease_definition(746)\n",
        "print(\"Disease definition:\", disease_definition)"
      ],
      "metadata": {
        "id": "onGL4yezfzXo"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "import requests\n",
        "from bs4 import BeautifulSoup\n",
        "\n",
        "def get_clinical_description(num):\n",
        "    # URL of the page to scrape\n",
        "    url = \"https://www.orpha.net/en/disease/detail/\" + str(num)\n",
        "    # Send a GET request to the URL\n",
        "    response = requests.get(url)\n",
        "    # Parse the HTML content\n",
        "    soup = BeautifulSoup(response.text, 'html.parser')\n",
        "    # Find the <div> tag with the class \"fs-5 orpha-blue\" containing the text \"Clinical description\"\n",
        "    clinical_div = soup.find('div', class_=\"fs-5 orpha-blue\", string=\"Clinical description\")\n",
        "    # If the <div> tag is found\n",
        "    if clinical_div:\n",
        "        # Get the next <p> tag which contains the clinical description\n",
        "        clinical_p = clinical_div.find_next('p')\n",
        "        # If <p> tag exists\n",
        "        if clinical_p:\n",
        "            # Extract and return the clinical description\n",
        "            return clinical_p.get_text(strip=True)\n",
        "    # If clinical description is not found\n",
        "    return \"Clinical description not found\"\n",
        "\n",
        "# Test the function\n",
        "clinical_description = get_clinical_description(746)\n",
        "print(\"Clinical description:\", clinical_description)"
      ],
      "metadata": {
        "id": "VcsrAVhuf367"
      },
      "execution_count": null,
      "outputs": []
    }
  ]
}